---
title: "Example_Workflow"
output: html_document
date: "2023-07-05"
---

## Documentation 

Before you start, you will need to prepare your raw read files, and fill in the 'metadata.csv' and 

Format of the raw read files is as follows:

Metadata file (metadata.csv) is of the format:

The info on primer sequences and main parameters will be in 'primerinfo_params.csv' file and of the following format: 

Please ensure that you have enough storage to save intermediate files in a temporary directory (default) (or user specified directory) before proceeding.

A few additional notes: 
If you are using the SILVA or UNITE databases, an ordinary personal computer may not have enough memory for the taxonomic assignment steps, even if you have few samples. This issue is not just isolated to this package, but when running the assignTaxonomy function in isolation, using DADA2. 
The test databases present in the package are randomly subsetted, for demonstration purposes only. User will need to upload their own databases to their input data folder that is also where the raw reads are located. 

Prepare and input metadata and primer files according to the templates provided. Using the provided templates, paste sample names in the first column.  In the second column, user will specify the barcode they have used. Currently, user can specify 'its' or 'rps10'. These primer names are case sensitive.

The metadata file should look like this:  


### Load package
For now, package will be loaded by retrieving the package from Github. Eventually, the package will be uploaded to CRAN or Bioconductor. 

```{r installation, message=FALSE, warnings=FALSE}
#temporary installation method
#ADJUST PATHS FOR MOMENT
#devtools::load_all("/Users/masudermann/rps10package")
#devtools::document()

#To pull from Github (Grunwald Lab members will need an auth_token as package is currently private)
devtools:install.github("grunwaldlab/demulticoder", auth_token="")
```

### Reorganize data tables for downstream steps and obtain primer counts across all sample reads  
The sample names, primer sequences and other metadata are reorganized in preparation for running Cutadapt to remove primers. 

```{r create data tables, message=FALSE, warnings=FALSE}
prepare_reads(
  maxN = 0, 
  data_directory = "inst/extdata", 
  output_directory = "~/output_package", 
  tempdir_id = "run2")
```

### Remove primers with Cutadapt and check that no primers still remain  

If primers still remain, you may want to adjust the parameters or manually remove any reads that still have primers (TODO-adjust this instruction later). 

***TODO-include a function to remove any ASV sequences that may still have primers. There likely won't be many-but it might be worth incorporating a function like this***

```{r Trim function, include=TRUE}
cut_trim(
  analysis_setup,
  cutadapt_path="/opt/homebrew/bin/cutadapt",
  verbose = TRUE,
  maxEE = 2,
  truncQ = 5,
  minLen = 200,
  maxLen = 297,
  minCutadaptlength = 50)
```
### Generate ASV abundance matrix.   

Depending on chosen specifications, the minOverlap and maxMismatch can be changed. The default values are the DADA2 default values 

```{r asv abund matrix, include=TRUE}
make_asv_abund_matrix(
  analysis_setup,
  minOverlap = 15,
  maxMismatch = 2,
  verbose = TRUE,
  multithread = TRUE)
```

### Taxonomic identification steps  

After databases are formatted, taxonomic information will be assigned to the ASVs that were inferred previously. The output will be a csv file containing each unique ASV, the abundance of each ASV across each sample, and the taxonomic assignments, the bootstrap support values, as well as a column listing the percent identity of each sequence to the corresponding sequence in the reference database.  

The user must specify which barcode they are using. Either 'rps10', its, or 'rps10_its'

**Databases** Todo-clarify instructions

Within the main directory, user must also have retrieved the rps10 oomycetedb (add link) or the Unite fungal database of choice. The names of the databases must be 'oomycetedb.fasta' and 'fungidb.fasta', or else they won't be recognized. The headers will altered and new versions of the databse will be saved. This will ensure there is consistency across databases. 

***TODO-save renamed db in temporary db or else they will take up too much space.***

```{r assign tax, include=TRUE}
assignTax(
  analysis_setup,
  asv_abund_matrix,
  multithread = TRUE,
  barcode = "rps10_its", 
  retrieve_files=TRUE
)
```

### Convert ASV matrix to Taxmap and Phyloseq objects

While some users will appreciate having a matrix with all ASV sequences, taxonomic assignments and abundances of each ASV per sample, others may prefer to reformat matrices into Phyloseq or Taxmap objects. 

Two wrapper functions are shared. They are derived from Metacoder functions 'parse_tax_data' and 'as_phyloseq'.

```{r Reformat ASV abundance matrix}
asv_matrix_to_taxmap(save_taxmap = TRUE)

taxmap_to_phyloseq(save_phyloseq=TRUE)
```

