---
output: github_document
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  out.width = "100%"
)
```

# Demulticoder R package

*This package is actively under development.  Additional testing, documentation, website, and examples forthcoming. Until this message has been removed, use with caution.*

### Introduction

The ***demulticoder*** package is a Cutadapt and DADA2 wrapper package for metabarcodng analyses. The main commands and outputs are intuitive and comprehensive, which helps to account for the complex and iterative nature of metabarcoding analyses. 

### Key features
- The ability to do analysis on either demultiplexed or pooled amplicons a within sample  
- Multiple datasets can be trimmed of primers, filtered, denoised, merged, and assigned taxonomy in one go (with different parameters for each dataset if desired)  
- The package handles not just 16S or ITS datasets when using default UNITE fungal or Silva 16S databases but also oomycete rps10 analyses using oomycetedb (https://oomycetedb.org), or up to two custom databases (provided they are formatted as described here: https://benjjneb.github.io/dada2/training.html).

### Installation

To install the development version of package: 

```{r, eval=FALSE, echo=TRUE}
devtools::install_github("grunwaldlab/demulticoder")
```

### Dependencies

### Required inputs

In one directory, add the following components:

* Paired-end read files (package tested on Illumina short-read data)

* Your **metadata.csv** file (download template here) (TODO)

* Your **primerinfo_params.csv** file (download template here) (TODO)

### Information on user-defined parameters that can be added to metadata file

TODO

### Citation

Stay tuned

The package was developed by Martha Sudermann, Zachary Foster, Samantha Dawson, Hung Phan, Niklaus Grunwald, Jeff Chang.

