---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# rps10package-rename and add info on package

<!-- badges: start -->
<!-- badges: end -->

Package readme in progress

## Installation

You can install the development version of rps10package from [GitHub](https://github.com/) with:

*TODO-still working on documentation*


```{r echo=T, warning=FALSE}
#temporary installation method
devtools::load_all("~/rps10package")
document()

# install.packages("devtools")
# devtools::install_github("grunwaldlab/rps10_metabarcoding_tool")
```

## Specify directory paths

Before you start, put your read fastq reads into a directory where you would like to do your analysis.
Please note: **xx** files and directories will be created.  

Please ensure that you have enough storage to save intermediate files and reformatted directories on your computer to proceed.  

If you are using the Silva of Unite databases, an ordinary personal computer may not have enough memory.  
***TODO-by default, intermediate files will be saved to a temporary directory. Provide instruction to user on how to override this.***

Prepare and input metadata and primer files according to the templates provided. Using the provided templates, paste sample names in the first column.  In the second column, user will specify the barcode they have used. Currently, user can specify 'its' or 'rps10'. These primer names are case sensitive.

The metadata file should look like this:  

***TODO add screenshot***

You will also need to make a sheet containing info on the primers used and forward and reverse primer sequences.  

***TODO add screenshot***

If barcodes are pooled, user will include each sample name twice, and under primer_name, either its or rps10 barcode.  

***TODO add screenshot of pooled barcode scenario***

Include any other metadata columns you'd like to use during later steps of the analysis. Formatting of names does not matter, but ensure that there are no blank left cells. 

```{r example}
#Change as needed based on where test dataset is located
directory_path<-"~/rps10package/raw_data/rps10_ITS" ##choose a directory for all downstream steps
primer_path <-file.path(directory_path, "primer_info.csv") ##modify .csv name or keep this name
metadata_path <-file.path(directory_path,"metadata.csv") ##modify .csv name or keep this name. The sample_name in the metadata sheet needs to match the first part (before first underscore), of the zipped raw FASTQ files
#Specify the location of where cutadapt is located
cutadapt_path<-"/opt/homebrew/bin/cutadapt"
```


## Reorganize data tables for downstream steps and obtain primer counts across all sample reads  

The sample names, primer sequences and other metadata are reorganized in preparation for running Cutadapt to remove primers. 

```{r create data tables}
data_tables <-
  prepare_reads(
    directory_path,
    primer_path,
    metadata_path,
    maxN = 0,
    multithread = TRUE
  )
```

