---
title: "User_Guide"
output: html_document
date: "2023-04-04"
---

library("devtools")
library("roxygen2")
library("testthat")
library("knitr")
library("dada2")
library(dada2); packageVersion("dada2")
library("metacoder")
library("phyloseq")


```{r load packages, warning=FALSE, message=FALSE, include=FALSE}
devtools::load_all("~/rps10_metabarcoding_tool")
document()
```

Before you start, put your read fastq reads into a directory where you would like to do your analysis.
Please note: xx directories will be created. Please ensure that you have enough storage on your computer to proceed. 
If you are using the Silva of Unite databases, an ordinary personal computer may not have enough memory. 

Step 1-Prepare and input metadata and primer files according to the templates provided. Using the provided templates, input, sample names, the names of corresponding raw read fastq files.
Sample names will belong in the first column. Each sample must be unique. 

#Metadata file just needs sample_name one column, and primer_name in second column (this function is being tweaked-see example)
```{r setup, include=FALSE}
directory_path<-"~/Documents/Postdoc/Jared_microbiome/Rps10/raw_data" ##choose a directory for all downstream steps
primer_path <-file.path(directory_path, "primer_info.csv") ##modify .csv name or keep this name
metadata_path <-file.path(directory_path,"Metadata_final_Mar23_rps10.csv") ##modify .csv name or keep this name. The sample_name in the metadata sheet needs to match the first part (before first underscore), of the zipped raw FASTQ files
cutadapt_path<-"/Users/masudermann/miniconda3/bin/cutadapt"
```

```{r data_tables, include=TRUE}
data_tables<-prepare_reads(directory_path, primer_path, metadata_path, maxN=0, multithread=TRUE)
```

```{r trim, include=TRUE}
#Fix output
cut_trim(data_tables,directory_path, cutadapt_path, verbose=TRUE, maxEE=5, truncQ=5, minLen=200, maxLen=300, minCutadaptlength=50) 

#Figure out why "Error in gzfile(file, "wb") : cannot open the connection" only when running in rmarkdown window
```

```{r asv abund matrix, include=TRUE}
asv_abund_matrix <- make_asv_abund_matrix(data_tables, directory_path, data_tables$cutadapt_data, minOverlap=15, maxMismatch=2, verbose=TRUE, multithread=TRUE) #check when 0 again
```

#Wrapper function to format databases and assign taxonomy
#Change barcode to 'rps10', 'its', or 'rps10_its'
```{r assign tax, include=TRUE}
summary<-assignTax(directory_path, data_tables, asv_abund_matrix, multithread = TRUE, barcode = "rps10_its", database_rps10="oomycetedb.fasta", database_its="sh_general_release_dynamic_22.08.2016.fasta")
```

```{r}
obj_dada<-asvmatrix_to_taxmap(asv_abund_matrix, min_read_depth=10, minimum_bootstrap=75)
phylo_obj2<-taxmap_to_phyloseq(obj_dada)
```
#Exploratory analysis
```{r}
#Heat tree metacoder
```

```{r}
#Alpha, beta diversity examples phyloseq
```

sessioninfo::session_info()
