---
title: "User_Guide"
output: html_document
date: "2023-02-14"
---

Before you start, put your read fastq reads into a directory where you would like to do your analysis.
Please note: xx directories will be created. Please ensure that you have enough storage on your computer to proceed. 
If you are using the Silva of Unite databases, an ordinary personal computer may not have enough memory. 

Step 1-Prepare and input metadata and primer files. Using the provided templates, input, sample names, the names of corresponding raw read fastq files.
Sample names will belong in the first column. Each sample must be unique. 
```{r setup, include=FALSE}
metadata<-read_csv("~/rps10_metabarcoding_tool/data/metadata.csv")
primer_info<-read_csv("~/rps10_metabarcoding_tool/data/primer_info.csv")
cutadapt_path<-"/Users/masudermann/miniconda3/bin/cutadapt"
directory_path<-metadata$directory_path[1]
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
