---
title: "Mixed barcode example"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Mixed barcode example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Demonstration of how to use demulticoder on a dataset that is actually three separate datasets (RPS10, ITS, and 16S) at once

Input metadata and primerinfo_params files are in data folder

TODO show example and add more commentary
```{r setup, include=FALSE}
#TODO-at some point switch to loading github repo version
devtools::load_all("~/demulticoder")
library("demulticoder")
```

### Step 1-Remove N's and create directory structure for downstream steps
```{r prepare reads, fig.height=6, fig.width=6}
outputs<-prepare_reads(
  data_directory = "~/demulticoder_benchmarking/demulticoder_combined_analysis_dads/data", 
  output_directory = "~/output_multiple_datasets", 
  overwrite_existing = TRUE)
```

### Step 2-Run Cutadapt to remove primers and then trim reads with DADA2 filterAndTrim function 
```{r Remove primers and trim reads, fig.height=6, fig.width=6}
cut_trim(
  outputs,
  cutadapt_path="/usr/bin/cutadapt",
  overwrite_existing = TRUE)
```

### Step 3-Core ASV inference step
```{r ASV inference, fig.height=6, fig.width=6}
make_asv_abund_matrix(
  outputs,
  overwrite_existing = TRUE)
```

### Step 4-Assign taxonomy step
```{r assign taxonomy step, fig.height=6, fig.width=6}
assign_tax(
  outputs,
  asv_abund_matrix,
  db_its = "sh_general_release_dynamic_all_25.07.2023.fasta",
  db_rps10 = "oomycetedb.fasta",
  db_16s="silva_nr99_v138.1_wSpecies_train_set.fa",
  retrieve_files=TRUE,
  overwrite_existing=FALSE)
```

### Step 5-convert asv matrix to taxmap and phyloseq objects with one function
```{r convert matrix to other formats, fig.height=6, fig.width=6}
objs<-convert_asv_matrix_to_objs(outputs, save_outputs=TRUE, overwrite_existing = TRUE)
```

```{r}
sessioninfo::session_info()
```