---
title: "Mixed barcode example"
output: 
  rmarkdown::html_vignette:
    fig_path: "man/figures"
vignette: >
  %\VignetteIndexEntry{Mixed barcode example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "./"
)
library("dplyr")
library("ggplot2")
library("metacoder")
library("phyloseq")
library("gridExtra")
```

## Demonstration of how to use demulticoder on a dataset that is actually three separate datasets (RPS10, ITS, and 16S) at once
This is a small dataset comprised 24 samples. In three separate experiments, 16S, ITS, and 16S genes/regions were amplified, following an experiment in which some soil samples were treated with diallyl disulfide (organosulfur biocontrol compound derived from garlic), half were exposed to anaerobic conditions vs. aerobic conditions, and half the samples were one soil type (Tulelake) vs. another (Madras). The goal was to examine the bacterial, fungal, and oomycete communities within these different treatment groups. Reads were obtained following three separate Illumina MiSeq runs. The following *Demulticoder* workflow below picks up after these runs. The read data has already been demultiplexed, and for each sample, there is 16S, ITS, and rps10 reads within three separate FASTQ files per sample. 

The goal of this workflow is to demonstrate that you can analyze simultaneously (but with different parameters), multiple datasets, related or not. Final matrices are separated by barcode, to avoid any violation of assumptions regarding experimental design. If the user so desires, they could combine the final matrices, depending on how the experiment was designed. 

### Step 1-Prepare metadata.csv file
You will see that sample names are unique for each barcoded-set of samples

Here is the CSV file looks like:
```{csv}
sample_name,primer_name,incubation,soil,treatment
1_1_01,r16S,Anaerobic,Madras,untreated
1_2_01,r16S,Anaerobic,Madras,untreated
1_3_01,r16S,Anaerobic,Madras,untreated
2_1_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_2_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_3_01,r16S,Anaerobic,Madras,DADS_1gal_A
3_1_01,r16S,Anaerobic,Tulelake,untreated
3_2_01,r16S,Anaerobic,Tulelake,untreated
3_3_01,r16S,Anaerobic,Tulelake,untreated
4_1_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_2_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_3_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
5_1_01,r16S,Aerobic,Madras,untreated
5_2_01,r16S,Aerobic,Madras,untreated
5_3_01,r16S,Aerobic,Madras,untreated
6_1_01,r16S,Aerobic,Madras,DADS_1gal_A
6_2_01,r16S,Aerobic,Madras,DADS_1gal_A
6_3_01,r16S,Aerobic,Madras,DADS_1gal_A
7_1_01,r16S,Aerobic,Tulelake,untreated
7_2_01,r16S,Aerobic,Tulelake,untreated
7_3_01,r16S,Aerobic,Tulelake,untreated
8_1_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_2_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_3_01,r16S,Aerobic,Tulelake,DADS_1gal_A
Dung1_1_S221_L001_02,its,Anaerobic,Madras,untreated
Dung1_1_S222,rps10,Anaerobic,Madras,untreated
Dung1_2_S222_L001_02,its,Anaerobic,Madras,untreated
Dung1_2_S223,rps10,Anaerobic,Madras,DADS_1gal_A
Dung1_3_S223_L001_02,its,Anaerobic,Madras,DADS_1gal_A
Dung1_3_S224,rps10,Anaerobic,Madras,DADS_1gal_A
Dung2_1_S224_L001_02,its,Anaerobic,Tulelake,untreated
Dung2_1_S225,rps10,Anaerobic,Tulelake,untreated
Dung2_2_S225_L001_02,its,Anaerobic,Tulelake,untreated
Dung2_2_S226,rps10,Anaerobic,Tulelake,DADS_1gal_A
Dung2_3_S226_L001_02,its,Anaerobic,Tulelake,DADS_1gal_A
Dung2_3_S227,rps10,Anaerobic,Tulelake,DADS_1gal_A
Dung3_1_S227_L001_02,its,Aerobic,Madras,untreated
Dung3_1_S228,rps10,Aerobic,Madras,untreated
Dung3_2_S228_L001_02,its,Aerobic,Madras,untreated
Dung3_2_S229,rps10,Aerobic,Madras,DADS_1gal_A
Dung3_3_S229_L001_02,its,Aerobic,Madras,DADS_1gal_A
Dung3_3_S230,rps10,Aerobic,Madras,DADS_1gal_A
Dung4_1_S230_L001_02,its,Aerobic,Tulelake,untreated
Dung4_1_S231,rps10,Aerobic,Tulelake,untreated
Dung4_2_S231_L001_02,its,Aerobic,Tulelake,untreated
Dung4_2_S232,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung4_3_S232_L001_02,its,Aerobic,Tulelake,DADS_1gal_A
Dung4_3_S233,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung5_1_S233_L001_02,its,Anaerobic,Madras,untreated
Dung5_1_S234,rps10,Anaerobic,Madras,untreated
Dung5_2_S234_L001_02,its,Anaerobic,Madras,untreated
Dung5_2_S235,rps10,Anaerobic,Madras,DADS_1gal_A
Dung5_3_S235_L001_02,its,Anaerobic,Madras,DADS_1gal_A
Dung5_3_S236,rps10,Anaerobic,Madras,DADS_1gal_A
Dung6_1_S236_L001_02,its,Anaerobic,Tulelake,untreated
Dung6_1_S237,rps10,Anaerobic,Tulelake,untreated
Dung6_2_S237_L001_02,its,Anaerobic,Tulelake,untreated
Dung6_2_S238,rps10,Anaerobic,Tulelake,DADS_1gal_A
Dung6_3_S238_L001_02,its,Anaerobic,Tulelake,DADS_1gal_A
Dung6_3_S239,rps10,Anaerobic,Tulelake,DADS_1gal_A
Dung7_1_S239_L001_02,its,Aerobic,Madras,untreated
Dung7_1_S240,rps10,Aerobic,Madras,untreated
Dung7_2_S240_L001_02,its,Aerobic,Madras,untreated
Dung7_2_S241,rps10,Aerobic,Madras,DADS_1gal_A
Dung7_3_S241_L001_02,its,Aerobic,Madras,DADS_1gal_A
Dung7_3_S242,rps10,Aerobic,Madras,DADS_1gal_A
Dung8_1_S242_L001_02,its,Aerobic,Tulelake,untreated
Dung8_1_S243,rps10,Aerobic,Tulelake,untreated
Dung8_2_S243_L001_02,its,Aerobic,Tulelake,untreated
Dung8_2_S244,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung8_3_S244_L001_02,its,Aerobic,Tulelake,DADS_1gal_A
Dung8_3_S245,rps10,Aerobic,Tulelake,DADS_1gal_A
```

In tabular format: 
```{r, echo=FALSE}
metadata <- read.csv(text = "
sample_name,primer_name,incubation,soil,treatment
1_1_01,r16S,Anaerobic,Madras,untreated
1_2_01,r16S,Anaerobic,Madras,untreated
1_3_01,r16S,Anaerobic,Madras,untreated
2_1_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_2_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_3_01,r16S,Anaerobic,Madras,DADS_1gal_A
3_1_01,r16S,Anaerobic,Tulelake,untreated
3_2_01,r16S,Anaerobic,Tulelake,untreated
3_3_01,r16S,Anaerobic,Tulelake,untreated
4_1_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_2_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_3_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
5_1_01,r16S,Aerobic,Madras,untreated
5_2_01,r16S,Aerobic,Madras,untreated
5_3_01,r16S,Aerobic,Madras,untreated
6_1_01,r16S,Aerobic,Madras,DADS_1gal_A
6_2_01,r16S,Aerobic,Madras,DADS_1gal_A
6_3_01,r16S,Aerobic,Madras,DADS_1gal_A
7_1_01,r16S,Aerobic,Tulelake,untreated
7_2_01,r16S,Aerobic,Tulelake,untreated
7_3_01,r16S,Aerobic,Tulelake,untreated
8_1_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_2_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_3_01,r16S,Aerobic,Tulelake,DADS_1gal_A
", stringsAsFactors = FALSE)

knitr::kable(head(metadata, 24), row.names = FALSE)
```

### Step 2-Prepare primerinfo_params.csv file
Here is the CSV file looks like:
```{csv}
primer_name,forward,reverse,already_trimmed,minCutadaptlength,multithread,verbose,maxN,maxEE_forward,maxEE_reverse,truncLen_forward,truncLen_reverse,truncQ,minLen,maxLen,minQ,trimLeft,trimRight,rm.lowcomplex,minOverlap,maxMismatch,min_asv_length
rps10,GTTGGTTAGAGYARAAGACT,ATRYYTAGAAAGAYTYGAACT,FALSE,FALSE,50,TRUE,TRUE,0,5,5,0,0,5,50,290,0,0,0,0,15,2,200
its,GTGARTCATCGAATCTTTG,TCCTSCGCTTATTGATATGC,FALSE,FALSE,50,TRUE,TRUE,0,2,2,0,0,5,50,290,0,0,0,0,15,2,50
r16S,GTGYCAGCMGCCGCGGTAA,GGACTACNVGGGTWTCTAAT,TRUE,FALSE,50,TRUE,TRUE,0,2,2,0,0,5,50,Inf,0,0,0,0,15,2,50
```

In tabular format: 
```{r, echo=FALSE}
primerinfo_params <- read.csv(text = "
primer_name,forward,reverse,already_trimmed,minCutadaptlength,multithread,verbose,maxN,maxEE_forward,maxEE_reverse,truncLen_forward,truncLen_reverse,truncQ,minLen,maxLen,minQ,trimLeft,trimRight,rm.lowcomplex,minOverlap,maxMismatch,min_asv_length
rps10,GTTGGTTAGAGYARAAGACT,ATRYYTAGAAAGAYTYGAACT,FALSE,50,TRUE,TRUE,0,5,5,0,0,5,50,290,0,0,0,0,15,2,200
its,GTGARTCATCGAATCTTTG,TCCTSCGCTTATTGATATGC,FALSE,50,TRUE,TRUE,0,2,2,0,0,5,50,290,0,0,0,0,15,2,50
r16S,GTGYCAGCMGCCGCGGTAA,GGACTACNVGGGTWTCTAAT,TRUE,50,TRUE,TRUE,0,2,2,0,0,5,50,Inf,0,0,0,0,15,2,50
", stringsAsFactors = FALSE)

knitr::kable(primerinfo_params, row.names = FALSE)
```

### Step 3-Load packages  
```{r setup, include=FALSE}
#TODO-at some point switch to loading github repo version
devtools::install_github("grunwaldlab/demulticoder", force=TRUE)
library("demulticoder")
```

### Step 4-Remove N's and create directory structure for downstream steps  
```{r prepare reads, fig.height=6, fig.width=6}
outputs<-prepare_reads(
  data_directory = "~/demulticoder_benchmarking_2/demulticoder_combined_analysis_dads/data", 
  output_directory = "~/demulticoder_benchmarking_2/demulticoder_combined_analysis_dads/output2", 
  tempdir_path = "~/demulticoder_benchmarking/temp",
  overwrite_existing = FALSE)
```

### Step 5-Run Cutadapt to remove primers and then trim reads with DADA2 filterAndTrim function  
```{r Remove primers and trim reads, fig.height=6, fig.width=6}
cut_trim(
  outputs,
  cutadapt_path="/usr/bin/cutadapt",
  overwrite_existing = FALSE)
```

### Step 6-Core ASV inference step  
```{r ASV inference, fig.height=6, fig.width=6}
make_asv_abund_matrix(
  outputs,
  overwrite_existing = FALSE)
```

### Step 7-Assign taxonomy step
```{r assign taxonomy step, fig.height=6, fig.width=6}
assign_tax(
  outputs,
  asv_abund_matrix,
  db_its="sh_general_release_dynamic_04.04.2024.fasta",
  db_rps10="oomycetedb.fasta",
  db_16S="silva_nr99_v138.1_wSpecies_train_set.fa",
  retrieve_files=FALSE,
  tryRC = TRUE,
  overwrite_existing = FALSE)
```

### Step 8-Convert ASV matrix to taxmap and phyloseq objects with one function  
```{r convert matrix to other formats, fig.height=6, fig.width=6}
objs<-convert_asv_matrix_to_objs(outputs, save_outputs=TRUE, overwrite_existing = TRUE, minimum_bootstrap = 60)
```

### Step 9-Use phyloseq objects to examine the community composition of each sample  

Here we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the 16S-barcoded samples
```{r Phyloseq demonstration 16S, fig.width=20, fig.height=20, message=FALSE}
data <- objs$phyloseq_r16S %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.005) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 10),
    legend.title = ggplot2::element_text(size = 10),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 10),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```

Now we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the ITS-barcoded samples
```{r Phyloseq demonstration ITS, fig.width=20, fig.height=20, message=FALSE}
data <- objs$phyloseq_its %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.005) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 10),
    legend.title = ggplot2::element_text(size = 10),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 10),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```

Finally, we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the rps10-barcoded samples
```{r Phyloseq demonstration rps10, fig.width=8, fig.height=7, message=FALSE}
data <- objs$phyloseq_rps10 %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>%
  phyloseq::psmelt() %>%
  dplyr::filter(Abundance > 0.0005) %>%
  dplyr::arrange(Genus)

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) +
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 10),
    legend.title = ggplot2::element_text(size = 10),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 10),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```

### Step 10-let's look at ordination plots for the three factors: treatment, soil, and incubation
First we will make ordination plots with 16S outputs
```{r ordination plots 16S, fig.height=10, fig.width=6, message=FALSE}
factors<-c("treatment","soil","incubation")

strat_rel.ps<-transform_sample_counts(objs$phyloseq_r16S, function(x) x / sum(x))

create_plot <- function(ps, factor_name) {
  ps_pcoa <- ordinate(ps, method = "PCoA", distance = "bray")
  
  plot <- plot_ordination(ps, ps_pcoa, color = factor_name) +
    ggtitle(paste("16S PCoA Ordination Plot -", factor_name)) +
    labs(color = "Factor") +
    theme(axis.text = element_text(size = 14),
          axis.title = element_text(size = 14),
          title = element_text(size = 13),
          legend.text = element_text(size = 18)) +
    stat_ellipse(type = "norm", linetype = 2) +
    theme_bw()
  
  return(plot)
}

plot_list <- list()
for (factor in factors) {
  plot_list[[factor]] <- create_plot(strat_rel.ps, factor)
}

combined_plot <- do.call(grid.arrange, c(plot_list, ncol = 1))

print(combined_plot)
```

Second, we will make ordination plots for ITS outputs
```{r ordination plots ITS, fig.height=6, fig.width=6, message=FALSE, echo=FALSE}
factors<-c("treatment","soil","incubation")

strat_rel.ps<-transform_sample_counts(objs$phyloseq_its, function(x) x / sum(x))

create_plot <- function(ps, factor_name) {
  ps_pcoa <- ordinate(ps, method = "PCoA", distance = "bray")
  
  plot <- plot_ordination(ps, ps_pcoa, color = factor_name) +
    ggtitle(paste("ITS PCoA Ordination Plot -", factor_name)) +
    labs(color = "Factor") +
    theme(axis.text = element_text(size = 14),
          axis.title = element_text(size = 14),
          title = element_text(size = 13),
          legend.text = element_text(size = 18)) +
    stat_ellipse(type = "norm", linetype = 2) +
    theme_bw()
  
  return(plot)
}

plot_list <- list()
for (factor in factors) {
  plot_list[[factor]] <- create_plot(strat_rel.ps, factor)
}

combined_plot <- do.call(grid.arrange, c(plot_list, ncol = 1))

print(combined_plot)
```

Third, we will make ordination plots rps10 outputs
```{r ordination plots rps10, fig.height=6, fig.width=6, message=FALSE, echo=FALSE}
factors<-c("treatment","soil","incubation")

strat_rel.ps<-transform_sample_counts(objs$phyloseq_rps10, function(x) x / sum(x))

create_plot <- function(ps, factor_name) {
  ps_pcoa <- ordinate(ps, method = "PCoA", distance = "bray")
  
  plot <- plot_ordination(ps, ps_pcoa, color = factor_name) +
    ggtitle(paste("RPS10 PCoA Ordination Plot -", factor_name)) +
    labs(color = "Factor") +
    theme(axis.text = element_text(size = 14),
          axis.title = element_text(size = 14),
          title = element_text(size = 13),
          legend.text = element_text(size = 18)) +
    stat_ellipse(type = "norm", linetype = 2) +
    theme_bw()
  
  return(plot)
}

plot_list <- list()
for (factor in factors) {
  plot_list[[factor]] <- create_plot(strat_rel.ps, factor)
}

combined_plot <- do.call(grid.arrange, c(plot_list, ncol = 1))

print(combined_plot)
```

### Step 11-Based on the big differences in microbial communities between untreated and treated samples, whether some taxa are more differentially abudnant in untreated vs. treated samples. We will use DESEQ2.

Here is an example with the 16S dataset.  
```{r heat tree 16S, fig.height=6, fig.width=6, message=FALSE, echo=FALSE}
objs$taxmap_r16S$data$tax_abund <- calc_taxon_abund(objs$taxmap_r16S, data = "otu_table", cols = objs$taxmap_r16S$data$sample_data$sample_id)
# Calculate total abundance per sample
asv_total_abundance <- rowSums(objs$taxmap_r16S$data$tax_abund[, -1])


objs$taxmap_r16S <- metacoder::filter_taxa(objs$taxmap_r16S, taxon_names == "Bacteria", subtaxa = TRUE)
objs$taxmap_r16S <- metacoder::filter_taxa(objs$taxmap_r16S, taxon_names != "None", reassign_obs = FALSE)
objs$taxmap_r16S <- metacoder::filter_taxa(objs$taxmap_r16S, taxon_names != "", reassign_obs = FALSE)
objs$taxmap_r16S <- metacoder::filter_taxa(objs$taxmap_r16S, taxon_names != "NA", reassign_obs = FALSE)
objs$taxmap_r16S <- metacoder::filter_taxa(objs$taxmap_r16S, taxon_names != "Unsupported", reassign_obs = FALSE)


objs$taxmap_r16S$data$diff_table_deseq <- metacoder::calc_diff_abund_deseq2(objs$taxmap_r16S,
                                      data = "tax_abund",
                                      cols = objs$taxmap_r16S$data$sample_data$samplename_barcode, # What columns of sample data to
                                      groups = objs$taxmap_r16S$data$sample_data$treatment) # What category each sample is assigned to

objs$taxmap_r16S$data$diff_table_deseq$log2FoldChange[objs$taxmap_r16S$data$diff_table_deseq$padj > 0.05] <- 0

range(objs$taxmap_r16S$objs$taxmap_r16S$data$diff_table_deseq$padj, finite = TRUE) 
range(objs$taxmap_r16S$data$diff_table_deseq$log2FoldChange, finite = TRUE) 


objs$taxmap_r16S%>%
  metacoder::filter_taxa(taxon_ranks == "Species", supertaxa = TRUE, reassign_obs = FALSE) %>%
  # Filter taxa to only include those where the kingdom is "Bacteria"
  heat_tree_matrix(data = "diff_table_deseq",
                   node_label = taxon_names,
                   node_size = n_obs, # number of OTUs
                   node_color = log2FoldChange, # difference between groups
                   node_color_trans = "linear",
                   node_color_interval = c(-11, 11), # symmetric interval
                   edge_color_interval = c(-11, 11), # symmetric interval
                   node_color_range = diverging_palette(), # diverging colors
                   node_size_axis_label = "ASV count",
                   node_color_axis_label = "Log 2 Fold Change",
                   layout = "da", initial_layout = "re",
                   key_size = 0.67, 
                   seed=1)
```


Let's look at what fungal taxa are more abundant in treated vs. untreated samples
```{r heat tree ITS, fig.height=6, fig.width=6, message=FALSE, echo=FALSE}
objs$taxmap_its$data$tax_abund <- calc_taxon_abund(objs$taxmap_its, data = "otu_table", cols = objs$taxmap_its$data$sample_data$sample_id)

# Calculate total abundance per sample
asv_total_abundance <- rowSums(objs$taxmap_its$data$tax_abund[, -1])


objs$taxmap_its <- metacoder::filter_taxa(objs$taxmap_its, taxon_names != "Unsupported", reassign_obs = FALSE)

objs$taxmap_its$data$diff_table_deseq <- metacoder::calc_diff_abund_deseq2(objs$taxmap_its,
                                      data = "tax_abund",
                                      cols = objs$taxmap_its$data$sample_data$samplename_barcode, # What columns of sample data to
                                      groups = objs$taxmap_its$data$sample_data$treatment) # What category each sample is assigned to

objs$taxmap_its$data$diff_table_deseq$log2FoldChange[objs$taxmap_its$data$diff_table_deseq$padj > 0.05] <- 0


range(objs$taxmap_its$objs$taxmap_its$data$diff_table_deseq$padj, finite = TRUE) 
range(objs$taxmap_its$data$diff_table_deseq$log2FoldChange, finite = TRUE) 


set.seed(1)
objs$taxmap_its %>%
  metacoder::filter_taxa(taxon_ranks == "Species", supertaxa = TRUE, reassign_obs = FALSE) %>%
  metacoder::filter_taxa(taxon_names != "Unsupported", reassign_obs = FALSE) %>%
  metacoder::filter_taxa(taxon_names != "", reassign_obs = FALSE) %>%
  heat_tree_matrix(data = "diff_table_deseq",
                   node_label = taxon_names,
                   node_size = n_obs, # number of OTUs
                   node_color = log2FoldChange, # difference between groups
                   node_color_trans = "linear",
                   node_color_interval = c(-23, 23), # symmetric interval
                   edge_color_interval = c(-23, 23), # symmetric interval
                   node_color_range = diverging_palette(), # diverging colors
                   node_size_axis_label = "ASV count",
                   node_color_axis_label = "Log 2 Fold Change",
                   layout = "da", initial_layout = "re",
                   key_size = 0.67,
                   seed = 1)
```

Finally, let's look at which oomycetes are more abudnant in treated vs.treated samples
```{r heat tree RPS10, fig.height=6, fig.width=6, message=FALSE, echo=FALSE}
objs$taxmap_rps10$data$tax_abund <- calc_taxon_abund(objs$taxmap_rps10, data = "otu_table", cols = objs$taxmap_rps10$data$sample_data$sample_id)

# Calculate total abundance per sample
asv_total_abundance <- rowSums(objs$taxmap_rps10$data$tax_abund[, -1])


objs$taxmap_rps10 <- metacoder::filter_taxa(objs$taxmap_rps10, taxon_names != "Unsupported", reassign_obs = FALSE)

objs$taxmap_rps10$data$diff_table_deseq <- metacoder::calc_diff_abund_deseq2(objs$taxmap_rps10,
                                      data = "tax_abund",
                                      cols = objs$taxmap_rps10$data$sample_data$samplename_barcode, # What columns of sample data to
                                      groups = objs$taxmap_rps10$data$sample_data$treatment) # What category each sample is assigned to

objs$taxmap_rps10$data$diff_table_deseq$log2FoldChange[objs$taxmap_rps10$data$diff_table_deseq$padj > 0.05] <- 0


range(objs$taxmap_rps10$objs$taxmap_rps10$data$diff_table_deseq$padj, finite = TRUE) 
range(objs$taxmap_rps10$data$diff_table_deseq$log2FoldChange, finite = TRUE) 


set.seed(1)
objs$taxmap_rps10 %>%
  metacoder::filter_taxa(taxon_ranks == "Species", supertaxa = TRUE, reassign_obs = FALSE) %>%
  metacoder::filter_taxa(taxon_names != "Unsupported", reassign_obs = FALSE) %>%
  metacoder::filter_taxa(taxon_names != "", reassign_obs = FALSE) %>%
  heat_tree_matrix(data = "diff_table_deseq",
                   node_label = taxon_names,
                   node_size = n_obs, # number of OTUs
                   node_color = log2FoldChange, # difference between groups
                   node_color_trans = "linear",
                   node_color_interval = c(-27, 27), # symmetric interval
                   edge_color_interval = c(-27, 27), # symmetric interval
                   node_color_range = diverging_palette(), # diverging colors
                   node_size_axis_label = "ASV count",
                   node_color_axis_label = "Log 2 Fold Change",
                   layout = "da", initial_layout = "re",
                   key_size = 0.67,
                   seed = 1)
```

```{r session info}
sessioninfo::session_info()
```