---
title: "Pooled amplicon example"
output: 
  rmarkdown::html_vignette:
    fig_path: "man/figures"
vignette: >
  %\VignetteIndexEntry{Pooled amplicon example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Demonstration of how to use demulticoder on a dataset that has pooled amplicoins (RPS10 and ITS)

Input metadata and primerinfo_params files are in data folder
```{r setup, include=FALSE}
#TODO-at some point switch to loading github repo version
devtools::load_all("~/demulticoder")
library("demulticoder")
```

### Step 1-Remove N's and create directory structure for downstream steps
```{r prepare reads}
outputs<-prepare_reads(
  data_directory = "~/demulticoder_benchmarking/demulticoder_combined_analysis_rhode/data", 
  output_directory = "~/demulticoder_benchmarking/demulticoder_combined_analysis_rhode/outputs_trunq2_overlap12", 
  tempdir_path = "~/demulticoder_benchmarking/demulticoder_combined_analysis_rhode/temp_rhode2", 
  overwrite_existing = FALSE)
```

### Step 2-Run Cutadapt to remove primers and then trim reads with DADA2 filterAndTrim function 
```{r Remove primers and trim reads}
cut_trim(
  outputs,
  cutadapt_path="/usr/bin/cutadapt",
  overwrite_existing = FALSE)
```

### Step 3-Core ASV inference step
```{r ASV inference}
make_asv_abund_matrix(
  outputs,
  overwrite_existing = FALSE)
```

### Step 4-Assign taxonomy step
```{r assign taxonomy step}
assign_tax(
  outputs,
  asv_abund_matrix,
  db_its = "sh_general_release_dynamic_all_04.04.2024.fasta",
  db_rps10 = "oomycetedb.fasta",
  retrieve_files=TRUE,
  tryRC = TRUE,
  overwrite_existing=FALSE)
```

### Step 5-convert asv matrix to taxmap and phyloseq objects with one function
```{r convert matrix to other formats}
objs<-convert_asv_matrix_to_objs(outputs, save_outputs=TRUE, overwrite_existing = TRUE)
```

### Step 6-Let's now proceed with some exploratory analysis using taxmap objs. The code is addapted from an analysis done by Zach Foster. 

# Based on previous analyses, we will load output matrices directly, combine rps10 and its matrices, and make sure there are no redundant taxa, and then proceed. This demonstration shows that you don't need to rely on the converted phyloseq or taxmap objects if you want more flexibility with being able to combine matrices.
```{r Additional analysis 1}
rps10_matrix<-read.csv("~/demulticoder_benchmarking/demulticoder_combined_analysis_rhode/")
its_matrix<-read.csv("~/demulticoder_benchmarking/demulticoder_combined_analysis_rhode/outputs2/final_asv_abundance_matrix_its.csv")
```

```{r}
sessioninfo::session_info()
```