---
title: "Getting Started"
output: 
  rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getting Started}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "./"
)
```

### Before You Start

* In the following example, we demonstrate key package functionality using a subset of reads from two samples containing pooled ITS1 fungal and *rps10* oomycete amplicons. 

* You can follow along because all test data and associated CSV input files are loaded with the package. Additional examples are also available on the website. 

* Please note, for speed, the test dataset is comprised of randomly subset reads for the samples (S1 and S2). For demonstration purposes only, an abridged version of the full UNITE database is also included, rather than the full one.

### Format of the PE amplicon files

* The package takes forward and reverse Illumina short read sequence data.

* **Format of file names** To avoid errors, the only characters that are acceptable in sample names are letters and numbers. Characters can be separated by underscores, but no other symbols. The files must end with the suffix **R1.fastq.gz** or **R2.fastq.gz**.

### Format of metadata file (*metadata.csv*)

The format of the CSV file is simple. A template is [here](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/metadata.csv).  

The only two necessary columns (with these names) are:

1. sample_name column
2. primer_info column

Any additional metadata should then be pasted after these two columns. They can be referenced later during the analysis steps and save a step of loading metadata later.

S1 and S2 are come from a rhododendron rhizobiome dataset and are a random subset of reads. 

You will notice S1 and S2 are included twice in the 'metadata.csv' sheet. This is because these two samples contain pooled reads (ITS and rps10). To demultiplex and run both analyses in tandem, include the same sample twice under sample_name, and then change the primer_name.

Example using test dataset:
```{r, echo=FALSE}
knitr::kable(data.frame(
  sample_name = c("S1", "S2", "S1", "S2"),
  primer_name = c("rps10", "rps10", "its", "its"),
  organism = c("Cry", "Cin", "Cry", "Cin")
), row.names = FALSE)
```

### Format of primer and parameters file (*primerinfo_parms.csv*)

DADA2 Primer sequence information and user-defined parameters are placed in *primerinfo_params.csv*. 

To simplify how functions are called, user will provide parameters within this input file. We recommend using the template linked [here](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/primerinfo_params.csv). Remember to add any additional optional DADA2 parameters you want to use.

The only required columns user must fill in are:
1.primer_name compatible options are currently (add to cells as written as this is crucial during database formatting step): rps10, its, r16S, other1, other2
2.forward-forward sequence
3.reverse-reverse sequence

Example template for 'primerinfo_params.csv'

```{r, echo=FALSE}
knitr::kable(data.frame(
  primer_name = c("rps10", "its"),
  forward = c("GTTGGTTAGAGYARAAGACT", "CTTGGTCATTTAGAGGAAGTAA"),
  reverse = c("ATRYYTAGAAAGAYTYGAACT", "GCTGCGTTCTTCATCGATGC"),
  already_trimmed = c("FALSE", "FALSE"),
  minCutadaptlength = c(100, 50),
  multithread = c("TRUE", "TRUE"),
  verbose = c("FALSE", "FALSE"),
  maxN = c("1.00E+05", "1.00E+05"),
  maxEE_forward = c(5, 5),
  maxEE_reverse = c(5, 5),
  truncLen_forward = c(0, 0),
  truncLen_reverse = c(0, 0),
  truncQ = c(5, 5),
  minLen = c(150, 50),
  maxLen = c("Inf", "Inf"),
  minQ = c(0, 0),
  trimLeft = c(0, 0),
  trimRight = c(0, 0),
  rm.lowcomplex = c(0, 0),
  minOverlap = c(15, 15),
  maxMismatch = c(0, 0),
  min_asv_length = c(50, 50)
), row.names = FALSE)
```

For more info on parameter specifics, see Documentation tab. 

### Reference Database Format

For now, the package is compatible with the following databases:

* oomycetedb from: [https://grunwaldlab.github.io/OomyceteDB/](https://grunwaldlab.github.io/OomyceteDB/)

* SILVA 16S database with species assignments: [https://www.arb-silva.de/](https://www.arb-silva.de/)

* UNITE fungal database from [https://unite.ut.ee/repository.php](https://unite.ut.ee/repository.php)

A user can select up to two other databases, but will first need to reformat headers exactly like the SILVA database. See more in the 'Documentation' tab. 

Databases will be copied into the user-specified data folder where raw data files and csv files are located. The names will be parameters in the *assignTax* function

### Additional Notes

Computer specifications may be a limiting factor. If you are using the SILVA or UNITE databases for taxonomic assignment steps, an ordinary personal computer (unless it has sufficient RAM) may not have enough memory for the taxonomic assignment steps, even with few samples. The test databases are comprised of randomly subset reads. The following example, should run on a personal computer with at least 16 GB of RAM.

Users need to upload their own databases to their input data folder. If computer crashes during the taxonomic assignment step, you will need to switch to a computer with sufficient memory. 

Please also ensure that you have enough storage to save intermediate files in a temporary directory (default) or user-specified directory before proceeding.

### Loading the Package  

For now, the package will be loaded by retrieving it from GitHub. 
We are submitting package to CRAN. 

```{r installation, echo=TRUE, results='hide', message=FALSE, warning=FALSE}
devtools::install_github("grunwaldlab/demulticoder")
#devtools::load_all("~/demulticoder")
library("demulticoder")
library("metacoder")
library("dplyr")
```

### Reorganize data tables and set-up data directory structure 

The sample names, primer sequences, and other metadata are reorganized in preparation for running Cutadapt to remove primers.
```{r prepare_for_analysis, warning=FALSE, message=FALSE}
analysis_setup<-demulticoder::prepare_reads(
  data_directory = system.file("extdata", package = "demulticoder"),
  output_directory = "~/output_test_dataset", 
  tempdir_path = "~/temp_test_dataset",
  tempdir_id = "test_dataset",
  overwrite_existing = TRUE)
```

### Remove primers with Cutadapt
Before running Cutadapt, please ensure that you have installed it. 
```{r remove_primers_with_cut_trim_function, message=TRUE}
demulticoder::cut_trim(
  analysis_setup,
  cutadapt_path = "/usr/bin/cutadapt",
  #cutadapt_path = "/usr/bin/cutadapt",
  overwrite_existing = TRUE)
```

### ASV inference step
Raw reads will be merged and ASVs will be inferred
```{r ASV_inference, message=TRUE}
make_asv_abund_matrix(
  analysis_setup,
  overwrite_existing = TRUE)
```

### Taxonomic assignment step
Using the core assignTaxonomy function from DADA2, taxonomic assignments will be given to ASVs. 
```{r Assign_taxonomy, message=TRUE}
assign_tax(
  analysis_setup,
  asv_abund_matrix,
  retrieve_files=TRUE,
  overwrite_existing = TRUE)
```

### Reformat ASV matrix as taxmap and phyloseq objects after optional filtering of low abundance ASVs
```{r Reformat_final_matrix, message=TRUE, warning=FALSE}
objs<-convert_asv_matrix_to_objs(analysis_setup, minimum_bootstrap = 0, save_outputs = TRUE)
```

### Format of output matrix

### Objects can now be used for downstream data analysis  
Here we make heattrees using taxmap object.  

First we make a heat tree for our ITS-barcoded samples 
```{r Heatree_demonstration_its, message=FALSE, warning=FALSE}
objs$taxmap_its %>%
  filter_taxa(! grepl(x = taxon_names, "_sp$"), reassign_obs = FALSE) %>%
  filter_taxa(! grepl(x = taxon_names, "incertae_sedis", ignore.case = TRUE), reassign_obs = FALSE) %>%
  filter_taxa(! grepl(x = taxon_names, "NA", ignore.case = TRUE), reassign_obs = FALSE) %>%
  metacoder::heat_tree(node_label = taxon_names,
                       node_size = n_obs,
                       node_color = n_obs,
                       node_color_axis_label = "ASV count",
                       node_size_axis_label = "Total Abundance of Taxa",
                       layout = "da", initial_layout = "re")
```

Now we make a heat tree for our rps10-barcoded samples
```{r Heatree_demonstration_rps10, message=FALSE, warning=FALSE}
objs$taxmap_rps10 %>%
  filter_taxa(! grepl(x = taxon_names, "_sp$"), reassign_obs = FALSE) %>%
  filter_taxa(! grepl(x = taxon_names, "incertae_sedis", ignore.case = TRUE), reassign_obs = FALSE) %>%
  filter_taxa(! grepl(x = taxon_names, "NA", ignore.case = TRUE), reassign_obs = FALSE) %>%
  metacoder::heat_tree(node_label = taxon_names,
                       node_size = n_obs,
                       node_color = n_obs,
                       node_color_axis_label = "ASV count",
                       node_size_axis_label = "Total Abundance of Taxa",
                       layout = "da", initial_layout = "re")
```

We can also do a variety of analyses, if we convert to phyloseq object

Here we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the ITS-barcoded samples
```{r Phyloseq_demonstration_ITS, message=FALSE, warning=FALSE}
data <- objs$phyloseq_its %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.02) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 14),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 14),
    legend.title = ggplot2::element_text(size = 14),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 14),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```

Finally,we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the rps10-barcoded samples
```{r Phyloseq_demonstration_rps10, message=FALSE, warning=FALSE}
data <- objs$phyloseq_rps10 %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.02) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 14),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 14),
    legend.title = ggplot2::element_text(size = 14),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 14),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```
