---
title: "Getting Started"
output: 
  rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getting Started}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "./",
  out.width = "70%",
  fig.retina = 1
)

library("dplyr")
library("kableExtra")
```

### Before You Start

* In the following example, we demonstrate key package functionality using the test dataset that is included in the package
* You can follow along because all test data and associated CSV input files are contained in the package. 
* Additional examples are also available under the [Example vignettes]tab

### Components of the test dataset  

1. **Paired-end short read amplicon data**
  - Files: S1_R1.fastq.gz, S1_R2.fastq.gz, S2_R1.fastq.gz, S2_R1.fastq.gz  
  - The files must be zipped and end in either *R1.fastq.gz* , or *R2.fastq.gz* and each sample must have both R1 and R2 files.
2. [**metadata.csv**](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/metadata.csv)
  - New row for each unique sample
  - Samples entered twice if samples contain two pooled metabarcodes, as in the test data template
3. [**primerinfo_params.csv**](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/primerinfo_params.csv)
  - New row for each unique barcode and associated primer sequence
  - Optional cutadapt and DADA2 parameters
4. **Taxonomy databases**  
  - UNITE fungal database (abridged version)
  - oomyceteDB 

### Format of the paired-end read files  

* The package takes forward and reverse Illumina short read sequence data.
* **Format of file names** To avoid errors, the only characters that are acceptable in sample names are letters and numbers. Characters can be separated by underscores, but no other symbols. The files must end with the suffix **R1.fastq.gz** or **R2.fastq.gz**.

### Format of metadata file   

* The format of the CSV file named **metadata.csv** is simple.  
* A template is [here](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/metadata.csv).  
* The only two **required** columns (with these headers) are:  
  * Column 1. **sample_name**  
  * Column 2. **primer_info**  

**Additional columns**  

* Other columns should be pasted after these two columns. 
* They can be referenced later during the analysis steps and save a step of loading metadata later.

**Notes**  

* S1 and S2 are come from a rhododendron rhizobiome dataset and are a random subset of the full set of reads.  
* S1 and S2 are included twice in the 'metadata.csv' sheet. This is because these two samples contain pooled metabarcodes (ITS1 and *rps10*).  
* To demultiplex and run both analyses in tandem, include the same sample twice under sample_name, and then change the primer_name.  

Here is what the *metadata.csv* looks like for this test dataset:  

```{r, echo=FALSE}
data <- data.frame(
  sample_name = c("S1", "S2", "S1", "S2"),
  primer_name = c("rps10", "rps10", "its", "its"),
  organism = c("Cry", "Cin", "Cry", "Cin"))

knitr::kable(data, row.names = FALSE) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", full_width = F, font_size = 6))
```

### Format of primer and parameters file 

DADA2 Primer sequence information and user-defined parameters are placed in **primerinfo_params.csv**     

* To simplify how functions are called, user will provide parameters within this input file.   
* We recommend using the template linked   [here](https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/primerinfo_params.csv).   
* Remember to add any additional optional DADA2 parameters you want to use.  

* The only three **required** columns (with these headers) are:  
  * Column 1. **sample_name** 
  * Column 2. **forward** (primer sequence)
  * Column 3. **reverse**(primer sequence)  

Example template for **primerinfo_params.csv**  
```{r, echo=FALSE}
primer_info <-data.frame(
  primer_name = c("rps10", "its"),
  forward = c("GTTGGTTAGAGYARAAGACT", "CTTGGTCATTTAGAGGAAGTAA"),
  reverse = c("ATRYYTAGAAAGAYTYGAACT", "GCTGCGTTCTTCATCGATGC"),
  already_trimmed = c("FALSE", "FALSE"),
  minCutadaptlength = c(100, 50),
  multithread = c("TRUE", "TRUE"),
  verbose = c("FALSE", "FALSE"),
  maxN = c("1.00E+05", "1.00E+05"),
  maxEE_forward = c(5, 5),
  maxEE_reverse = c(5, 5),
  truncLen_forward = c(0, 0),
  truncLen_reverse = c(0, 0),
  truncQ = c(5, 5),
  minLen = c(150, 50),
  maxLen = c("Inf", "Inf"),
  minQ = c(0, 0),
  trimLeft = c(0, 0),
  trimRight = c(0, 0),
  rm.lowcomplex = c(0, 0),
  minOverlap = c(15, 15),
  maxMismatch = c(0, 0),
  min_asv_length = c(50, 50))

kable(primer_info, row.names = FALSE) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive", full_width = F, font_size = 12)) %>%
  scroll_box(width = "100%", height = "100px", extra_css = "thead th { white-space: nowrap; }")
```

For more info on parameter specifics, see [Documentation](https://grunwaldlab.github.io/demulticoder/articles/Documentation.html)   

### Reference Database Format  

* For now, the package is compatible with the following databases:  
  * oomycetedb from: [https://grunwaldlab.github.io/OomyceteDB/](https://grunwaldlab.github.io/OomyceteDB/)  
  * SILVA 16S database with species assignments: [https://www.arb-silva.de/](https://www.arb-silva.de/)
  * UNITE fungal database from [https://unite.ut.ee/repository.php](https://unite.ut.ee/repository.php)
  * A user can select up to two other databases, but will first need to reformat headers exactly like the SILVA database headers.  
    * See more in the 'Documentation' tab.  

Databases are downloaded from these sources and then placed in the from the user-specified data folder where raw data files and csv files are located.

### Additional Notes

* Computer specifications may be a limiting factor.  
* If you are using the SILVA or UNITE databases for taxonomic assignment steps, an ordinary personal computer (unless it has sufficient RAM) may not have enough memory for the taxonomic assignment steps, even with few samples.  
* The test databases and reads are subsetted and the following example should run on a personal computer with at least 16 GB of RAM.  
* If computer crashes during the taxonomic assignment step, you will need to switch to a computer with sufficient memory.  
* You must ensure that you have enough storage to save intermediate files in a temporary directory (default) or user-specified directory before proceeding.

### Installation  

**Dependencies**:  
First install **cutadapt** program following the instructions here: https://cutadapt.readthedocs.io/en/stable/installation.html  

Let's locate where the cutadapt executable is.  

You must do this from a **Terminal** window:  
```{sh, eval=FALSE, echo=TRUE}
#If you installed with pip or pipx, or homebrew, run this command from a Terminal window
which cutadapt
cutadapt --version
```

If you followed the cutadapt installation instructions to create a conda environment called cutadapt (change to whatever you named your environment), to install it in. 

Open up a **Terminal** window and type these commands:
```{sh, eval=FALSE, echo=TRUE}
#Run commands from a Terminal window
conda activate cutadapt
which cutadapt
cutadapt --version
```

Second, make sure the following R packages are installed:

* **DADA2** (Latest version is 3.20)  
  * To install, follow these instructions: https://www.bioconductor.org/packages/release/bioc/html/dada2.html  
* **phyloseq**  
  * To install: https://www.bioconductor.org/packages/release/bioc/html/phyloseq.html  
* **metacoder** (Available through CRAN)  

To install the development version of package (while submission to CRAN is in progress):  

### Loading the Package  

For now, the package will be loaded by retrieving it from GitHub. Submission to CRAN is in progress.  
```{r installation_1, eval=FALSE}
#Here we install demulticoder (instructions will be updated once available through CRAN)
devtools::install_github("grunwaldlab/demulticoder")
library("demulticoder")
#Let's make sure other packages are loaded:
library("devtools")
library("dada2")
library("phyloseq")
library("metacoder")
```

```{r installation_2, message=FALSE, warning=FALSE, include=FALSE}
devtools::install_github("grunwaldlab/demulticoder")
library("demulticoder")
library("metacoder")
library("dplyr")
library("metacoder")
```

### Reorganize data tables and set-up data directory structure 

The sample names, primer sequences, and other metadata are reorganized in preparation for running Cutadapt to remove primers.  

```{r prepare_for_analysis, warning=FALSE, message=TRUE, cache=FALSE}
output<-prepare_reads(
  data_directory = system.file("extdata", package = "demulticoder"),
  output_directory = "~/demulticoder_test", #Run commands from a Terminal window", 
  overwrite_existing = TRUE)
```

### Remove primers with Cutadapt  

Before running Cutadapt, **please ensure that you have installed it** 

```{r remove_primers_with_cut_trim_function, warning=FALSE, message=TRUE, cache=FALSE}
cut_trim(
  output,
  cutadapt_path = "/usr/bin/cutadapt", #CHANGE LOCATION TO YOUR LOCAL INSTALLATION
  overwrite_existing = TRUE)
```

### ASV inference step  

Raw reads will be merged and ASVs will be inferred  

```{r ASV_inference, warning=FALSE, message=FALSE, cache=FALSE}
make_asv_abund_matrix(
  output,
  overwrite_existing = TRUE)
```

### Taxonomic assignment step  

Using the core assignTaxonomy function from DADA2, taxonomic assignments will be given to ASVs.  

```{r Assign_taxonomy, warning=FALSE, message=FALSE, cache=FALSE}
assign_tax(
  output,
  asv_abund_matrix,
  retrieve_files=TRUE,
  overwrite_existing = TRUE)
```

### Format of output matrices  

* The default output is a CSV file per metabarcode with the inferred ASVs and sequences, taxonomic assignments, and bootstrap supports provided by DADA2. These data can then be used as input for downstream steps.  
* To view these files, locate the files with the prefix **final_asv_abundance_matrix** in your specified output directory.  
* In this example, it will be in the home directory within **demulticoder_test**. 
* You can then open your matrices using Excel  
* We recommend checking the column with the ASV sequences to make sure there aren't truncated sequences 
*It is important to also review the column showing the taxonomic assignments of each ASV and their associated bootstrap supports. 

We can also load the matrices into our R environment and view then from R Studio.  

We do that here with the *rps10* matrix 
```{r Show_rps10_matrix, warning=FALSE, message=FALSE}
rps10_matrix<-read.csv("~/demulticoder_test/final_asv_abundance_matrix_rps10.csv", header=TRUE)
head(rps10_matrix)
```

We can also inspect the first rows of the *ITS1* matrix
```{r Show_its_matrix, warning=FALSE, message=FALSE}
its_matrix<-read.csv("~/demulticoder_test/final_asv_abundance_matrix_its.csv", header=TRUE)
head(its_matrix)
```

### Reformat ASV matrix as taxmap and phyloseq objects after optional filtering of low abundance ASVs  
```{r Reformat_final_matrix, message=FALSE, warning=FALSE, results='hide'}
objs<-convert_asv_matrix_to_objs(output, minimum_bootstrap = 0, save_outputs = TRUE)
```

### Objects can now be used for downstream data analysis  
Here we make heattrees using taxmap object.  

First we make a heat tree to examine fungal community composition using our **ITS1** data. 
```{r Heatree_demonstration_its, fig.height=8, fig.width=8, warning=FALSE, message=FALSE}
objs$taxmap_its %>%
  filter_taxa(! grepl(x = taxon_names, "NA", ignore.case = TRUE)) %>%
  metacoder::heat_tree(node_label = taxon_names,
                       node_size = n_obs,
                       node_color = n_obs,
                       node_color_axis_label = "ASV count",
                       node_size_axis_label = "Total Abundance of Taxa",
                       layout = "da", initial_layout = "re")
```

Now we make a heat tree to showcase oomycete community composition using our *rps10* data
```{r Heatree_demonstration_rps10, fig.height=8, fig.width=8, warning=FALSE, message=FALSE}
objs$taxmap_rps10 %>%
  filter_taxa(! grepl(x = taxon_names, "NA", ignore.case = TRUE)) %>%
  metacoder::heat_tree(node_label = taxon_names,
                       node_size = n_obs,
                       node_color = n_obs,
                       node_color_axis_label = "ASV count",
                       node_size_axis_label = "Total Abundance of Taxa",
                       layout = "da", initial_layout = "re")
```

We can also do a variety of analyses, if we convert to **phyloseq** object

Here we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the ITS1-barcoded samples
```{r Phyloseq_demonstration_ITS, fig.height=8, fig.width=8, warning=FALSE, message=FALSE}
data <- objs$phyloseq_its %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.02) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 14),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 14),
    legend.title = ggplot2::element_text(size = 14),  
    strip.text = ggplot2::element_text(size = 14),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  
    )
  )

print(abund_plot)
```

Finally, we demonstrate how to make a stacked bar plot of the relative abundance of taxa by sample for the rps10-barcoded samples  
```{r Phyloseq_demonstration_rps10, fig.height=8, fig.width=8, warning=FALSE, message=FALSE}
data <- objs$phyloseq_rps10 %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  phyloseq::psmelt() %>%                                        
  dplyr::filter(Abundance > 0.02) %>%                      
  dplyr::arrange(Genus)                                      

abund_plot <- ggplot2::ggplot(data, ggplot2::aes(x = Sample, y = Abundance, fill = Genus)) + 
  ggplot2::geom_bar(stat = "identity", position = "stack", color = "black", size = 0.2) +
  ggplot2::scale_fill_viridis_d() +
  ggplot2::theme_minimal() +
  ggplot2::labs(
    y = "Relative Abundance",
    title = "Relative abundance of taxa by sample",
    fill = "Genus"
  ) +
  ggplot2::theme(
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 1, vjust = 0.5, size = 14),
    panel.grid.major = ggplot2::element_blank(),
    panel.grid.minor = ggplot2::element_blank(),
    legend.position = "top",
    legend.text = ggplot2::element_text(size = 14),
    legend.title = ggplot2::element_text(size = 14),  # Adjust legend title size
    strip.text = ggplot2::element_text(size = 14),
    strip.background = ggplot2::element_blank()
  ) +
  ggplot2::guides(
    fill = ggplot2::guide_legend(
      reverse = TRUE,
      keywidth = 1,
      keyheight = 1,
      title.position = "top",
      title.hjust = 0.5,  # Center the legend title
      label.theme = ggplot2::element_text(size = 10)  # Adjust the size of the legend labels
    )
  )

print(abund_plot)
```