<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>An R Package for the Integrated Analysis of Multiplex Metabarcodes â€¢ demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="An R Package for the Integrated Analysis of Multiplex Metabarcodes">
<meta name="description" content="The demulticoder package provides a workflow for analyzing multiplex metabarcoding data. It includes functions for filtering, trimming primers, and denoising data using Cutadapt and the dada2 package. The package can handle datasets with pooled amplicons and simple experimental designs.">
<meta property="og:description" content="The demulticoder package provides a workflow for analyzing multiplex metabarcoding data. It includes functions for filtering, trimming primers, and denoising data using Cutadapt and the dada2 package. The package can handle datasets with pooled amplicons and simple experimental designs.">
<meta property="og:image" content="/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="articles/Documentation.html">Documentation</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes">
<li><a class="dropdown-item" href="articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Main Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="demulticoder-r-package">Demulticoder R package<a class="anchor" aria-label="anchor" href="#demulticoder-r-package"></a>
</h1>
</div>
<p><em>This package is actively under development. Until this message has been removed, use with caution. Additional testing, documentation, and examples are in progress.</em></p>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>The <strong><em>demulticoder</em></strong> package is a Cutadapt and DADA2 wrapper package for metabarcodng analyses. The main commands and outputs are intuitive and comprehensive, which helps to account for the complex and iterative nature of metabarcoding analyses.</p>
<p>Here is a brief schematic of the general workflow:</p>
<p><img src="reference/figures/Figure1.svg" width="35%" height="35%" style="display: block; margin: auto auto auto 0;"></p>
</div>
<div class="section level3">
<h3 id="key-features">Key features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h3>
<ul>
<li><p>The ability to do analysis on either demultiplexed or pooled amplicons within samples</p></li>
<li><p>Amplicons from multiple datasets be trimmed of primers, filtered, denoised, merged, and given taxonomic assignments in one go (with different parameters for each dataset if desired)</p></li>
<li><p>The package handles not just 16S or ITS datasets when using default UNITE fungal or Silva 16S databases but also oomycete rps10 analyses using oomycetedb (<a href="https://oomycetedb.org" class="external-link uri">https://oomycetedb.org</a>), or up to two custom databases (provided they are formatted as described here: <a href="https://benjjneb.github.io/dada2/training.html" class="external-link uri">https://benjjneb.github.io/dada2/training.html</a>).</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>To install the development version of package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"grunwaldlab/demulticoder"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h3>
<p><strong>1. Set-up input directory and files</strong></p>
<p>After installing the package, make a data directory and add the following files:<br>
- PE short read amplicon data. The files must end in either *_R1.fastq.gz* , or *_R2.fastq.gz* and each sample must have both R1 and R2 files.</p>
<ul>
<li><p><a href="https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/metadata.csv" class="external-link"><strong>metadata.csv</strong></a> file (there will be a unique row for each sample, and samples will be entered twice if they contain pooled amplicons, as in the example template)</p></li>
<li><p><a href="https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/primerinfo_params.csv" class="external-link"><strong>primerinfo_params.csv</strong></a> file (there will be a new row for each unique barcode and associated primer sequences, and there are also optional Cutadapt, DADA2 or filtering parameters that can be added or adjusted)</p></li>
</ul>
<p><strong>2. Prepare reads</strong></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span><span class="op">&lt;-</span><span class="fu"><a href="reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="st">"&lt;DATADIR&gt;"</span>,</span>
<span>  output_directory <span class="op">=</span> <span class="st">"&lt;OUTDIR&gt;"</span><span class="op">)</span></span></code></pre></div>
<p><strong>3. Cut and trim reads</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"&lt;CUTADAPTPATH&gt;"</span><span class="op">)</span></span></code></pre></div>
<p><strong>4. Make ASV abundance matrix</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span><span class="op">)</span></span></code></pre></div>
<p><strong>5. Assign taxonomy</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span>,</span>
<span>  <span class="va">asv_abund_matrix</span><span class="op">)</span></span></code></pre></div>
<p><strong>6. Convert ASV matrix to taxmap and phyloseq objects</strong></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="check-out-the-associated-github-repository-to-view-source-code">Check out the associated Github repository to view source code<a class="anchor" aria-label="anchor" href="#check-out-the-associated-github-repository-to-view-source-code"></a>
</h3>
<p>For more information, to see source code, or submit issue, check out:<br><a href="https://github.com/grunwaldlab/demulticoder/" class="external-link uri">https://github.com/grunwaldlab/demulticoder/</a></p>
</div>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>The package was developed by Martha Sudermann, Zachary Foster, Samantha Dawson, Hung Phan, Jeff Chang, and Niklaus GrÃ¼nwald</p>
<p>Stay tuned for associated manuscript.</p>
</div>
<div class="section level3">
<h3 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h3>
<p>The demulticoder logo was created with BioRender.com</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/grunwaldlab/demulticoder/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/grunwaldlab/demulticoder/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing demulticoder</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Martha A. Sudermann <br><small class="roles"> Author, maintainer </small>  </li>
<li>Zachary S. L Foster <br><small class="roles"> Author </small>  </li>
<li>Samantha Dawson <br><small class="roles"> Author </small>  </li>
<li>Hung Phan <br><small class="roles"> Author </small>  </li>
<li>Jeff H. Chang <br><small class="roles"> Author </small>  </li>
<li>Niklaus GrÃ¼nwald <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus GrÃ¼nwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
