<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="demulticoder">
<title>Documentation • demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Documentation">
<meta property="og:description" content="demulticoder">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/01_getting_started.html">Get Started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/02_documentation.html">Documentation</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-example-workflows">Example Workflows</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-example-workflows">
    <a class="dropdown-item" href="../articles/03_mixed_barcode_example.html">Mixed Barcode Example</a>
    <a class="dropdown-item" href="../articles/04_pooled_amplicon_example.html">Pooled Amplicon Example</a>
    <a class="dropdown-item" href="../articles/05_custom_database_example.html">Custom Database Example</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Main Functions</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Documentation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/02_documentation.Rmd" class="external-link"><code>vignettes/02_documentation.Rmd</code></a></small>
      <div class="d-none name"><code>02_documentation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This documentation provides comprehensive information on how to use
the <em>Demulticoder</em> R package for processing and analyzing
metabarcode sequencing data. It covers input file requirements,
parameter settings, and key parameters.</p>
</div>
<div class="section level2">
<h2 id="quick-start-guide">Quick Start Guide<a class="anchor" aria-label="anchor" href="#quick-start-guide"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Prepare your input files (metadata.csv, primerinfo_params.csv,
unformatted reference databases, and PE Illumina read files).</li>
<li>Place all input files in a single directory.</li>
<li>Ensure your file names comply with the specified format.</li>
<li>Run the pipeline with default settings or adjust parameters as
needed.</li>
</ol>
</div>
<div class="section level2">
<h2 id="input-files">Input Files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<div class="section level3">
<h3 id="directory-structure">Directory Structure<a class="anchor" aria-label="anchor" href="#directory-structure"></a>
</h3>
<p>Place all your input files into a single directory. The directory
should contain the following files:</p>
<ul>
<li><strong>metadata.csv</strong></li>
<li><strong>primerinfo_params.csv</strong></li>
<li><strong>PE Illumina read files</strong></li>
<li><strong>Unformatted reference databases</strong></li>
</ul>
</div>
<div class="section level3">
<h3 id="file-naming-conventions">File Naming Conventions<a class="anchor" aria-label="anchor" href="#file-naming-conventions"></a>
</h3>
<div class="section level4">
<h4 id="read-name-format">Read Name Format<a class="anchor" aria-label="anchor" href="#read-name-format"></a>
</h4>
<p>To avoid errors, the only characters that are acceptable in sample
names are letters and numbers. Characters can be separated by
underscores, but no other symbols. The files must end with the suffix
<strong>R1.fastq.gz</strong> or <strong>R2.fastq.gz</strong></p>
<p><strong>Examples of permissible sample names are as
follows:</strong></p>
<ul>
<li>Sample1_R1.fastq.gz</li>
<li>Sample1_R2.fastq.gz</li>
</ul>
<p><strong>Other permissible names are:</strong></p>
<ul>
<li>Sample1_001_R1.fastq.gz</li>
<li>Sample1_001_R2.fastq.gz</li>
</ul>
<p><strong>What is not permissible is:</strong></p>
<ul>
<li>Sample1_001_R1_001.fastq.gz</li>
<li>Sample1_001_R2_001.fastq.gz</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="metadata-csv">metadata.csv<a class="anchor" aria-label="anchor" href="#metadata-csv"></a>
</h3>
<p>The metadata.csv file contains information about the samples and
primers (and associated barcodes) used in the experiment. It has the
following two required columns:</p>
<ul>
<li>
<strong>sample_name</strong>: Identifier for each sample (e.g., S1,
S2)</li>
<li>
<strong>primer_name</strong>: Name of the primer used (e.g., rps10,
its, sixteenS, other1, oteher2)</li>
</ul>
<p>Please add your associated metadata to the file after these two
required columns. This can then be used for your downstream exploratory
or diversity analyses, as the sample data will be incorporated into the
final phyloseq and taxmap objects.</p>
<p>Example file (with optional third column):</p>
<pre class="csv"><code>sample_name,primer_name,organism
S1,rps10,Cry
S2,rps10,Cin
S1,its,Cry
S2,its,Cin</code></pre>
</div>
<div class="section level3">
<h3 id="primerinfo_params-csv">primerinfo_params.csv<a class="anchor" aria-label="anchor" href="#primerinfo_params-csv"></a>
</h3>
<p>The primerinfo_params.csv file contains information about the primer
sequences used in the experiment, along with optional additional
parameters that are part of the DADA2 pipeline. If anything is not
specified, the default values will be used. The file has the following
columns:<br><strong>Required columns:</strong></p>
<ul>
<li>
<strong>primer_name</strong>: Name of the primer/barcode (e.g., its,
rps10)</li>
<li>
<strong>forward</strong>: Forward primer sequence</li>
<li>
<strong>reverse</strong>: Reverse primer sequence</li>
</ul>
<p><strong>DADA2 filterAndTrim function parameters:</strong></p>
<ul>
<li>
<strong>already_trimmed</strong>: Boolean indicating if primers are
already trimmed (TRUE/FALSE) (default: FALSE)</li>
<li>
<strong>minCutadaptlength</strong>: Cutadapt parameter-Minimum
length after Cutadapt trimming (default: 0)</li>
<li>
<strong>multithread</strong>: Boolean for multithreading
(TRUE/FALSE) (default: FALSE)</li>
<li>
<strong>verbose</strong>: Boolean for verbose output (TRUE/FALSE)
(default: FALSE)</li>
<li>
<strong>maxN</strong>: Maximum number of N bases allowed (default:
0)</li>
<li>
<strong>maxEE_forward</strong>: Maximum expected errors for forward
reads (default: Inf)</li>
<li>
<strong>maxEE_reverse</strong>: Maximum expected errors for reverse
reads (default: Inf)</li>
<li>
<strong>truncLen_forward</strong>: Truncation length for forward
reads (default: 0)</li>
<li>
<strong>truncLen_reverse</strong>: Truncation length for reverse
reads (default: 0)</li>
<li>
<strong>truncQ</strong>: Truncation quality threshold (default:
2)</li>
<li>
<strong>minLen</strong>: Minimum length of reads after processing
(default: 20)</li>
<li>
<strong>maxLen</strong>: Maximum length of reads after processing
(default: Inf)</li>
<li>
<strong>minQ</strong>: Minimum quality score (default: 0)</li>
<li>
<strong>trimLeft</strong>: Number of bases to trim from the start of
reads (default: 0)</li>
<li>
<strong>trimRight</strong>: Number of bases to trim from the end of
reads (default: 0)</li>
<li>
<strong>rm.lowcomplex</strong>: Boolean for removing low complexity
sequences (default: TRUE)</li>
</ul>
<p><strong>DADA2 learnErrors function parameters:</strong></p>
<ul>
<li>
<strong>nbases</strong>: Number of bases to use for error rate
learning (default: 1e+08)</li>
<li>
<strong>randomize</strong>: Randomize reads for error rate learning
(default: FALSE)</li>
<li>
<strong>MAX_CONSIST</strong>: Maximum number of self-consistency
iterations (default: 10)</li>
<li>
<strong>OMEGA_C</strong>: Convergence threshold for the error rates
(default: 0)</li>
<li>
<strong>qualityType</strong>: Quality score type (“Auto”,
“FastqQuality”, or “ShortRead”) (default: “Auto”)</li>
</ul>
<p><strong>DADA2 plot errors parameters:</strong></p>
<ul>
<li>
<strong>err_out</strong>: Return the error rates used for inference
(default: TRUE)</li>
<li>
<strong>err_in</strong>: Use input error rates instead of learning
them (default: FALSE)</li>
<li>
<strong>nominalQ</strong>: Use nominal Q-scores (default:
FALSE)</li>
<li>
<strong>obs</strong>: Return the observed error rates (default:
TRUE)</li>
</ul>
<p><strong>DADA2 dada function parameters:</strong></p>
<ul>
<li>
<strong>OMP</strong>: Use OpenMP multi-threading if available
(default: TRUE)</li>
<li>
<strong>n</strong>: Number of reads to use for error rate estimation
(default: 1e+05)</li>
<li>
<strong>id.sep</strong>: Character separating sample ID from
sequence name (default: “\s”)</li>
<li>
<strong>orient.fwd</strong>: NULL or TRUE/FALSE to orient sequences
(default: NULL)</li>
<li>
<strong>pool</strong>: Pool samples for error rate estimation
(default: FALSE)</li>
<li>
<strong>selfConsist</strong>: Perform self-consistency iterations
(default: FALSE)</li>
</ul>
<p><strong>DADA2 mergePairs function parameters:</strong></p>
<ul>
<li>
<strong>minOverlap</strong>: Minimum overlap for merging paired-end
reads (default: 12)</li>
<li>
<strong>maxMismatch</strong>: Maximum mismatches allowed in the
overlap region (default: 0)</li>
</ul>
<p><strong>DADA2 removeBimeraDenovo function parameters:</strong></p>
<ul>
<li>
<strong>method</strong>: Method for sample inference (“consensus” or
“pooled”) (default: “consensus”)</li>
</ul>
<p><strong>Other parameters to include in CSV input file:</strong></p>
<ul>
<li>
<strong>min_asv_length</strong>: Minimum length of Amplicon Sequence
Variants (ASVs) after core dada ASV inference steps (default=0)</li>
</ul>
<p>Example file (with select optional columns after forward and reverse
primer sequence columns):</p>
<pre class="csv"><code>primer_name,forward,reverse,already_trimmed,minCutadaptlength,multithread,verbose,maxN,maxEE_forward,maxEE_reverse,truncLen_forward,truncLen_reverse,truncQ,minLen,maxLen,minQ,trimLeft,trimRight,rm.lowcomplex,minOverlap,maxMismatch,min_asv_length
rps10,GTTGGTTAGAGYARAAGACT,ATRYYTAGAAAGAYTYGAACT,FALSE,100,TRUE,FALSE,1.00E+05,5,5,0,0,5,150,Inf,0,0,0,0,15,0,50
its,CTTGGTCATTTAGAGGAAGTAA,GCTGCGTTCTTCATCGATGC,FALSE,50,TRUE,FALSE,1.00E+05,5,5,0,0,5,50,Inf,0,0,0,0,15,0,50</code></pre>
</div>
<div class="section level3">
<h3 id="reference-database">Reference Database<a class="anchor" aria-label="anchor" href="#reference-database"></a>
</h3>
<p>Databases will be copied into the user-specified data folder where
raw data files and csv files are located. The names will be parameters
in the <em>assignTax</em> function.</p>
<p>For now, the package is compatible with the following databases:</p>
<ul>
<li><p>oomycetedb from: <a href="http://www.oomycetedb.org/" class="external-link">http://www.oomycetedb.org/</a></p></li>
<li><p>SILVA 16S database <strong>with</strong> species assignments: <a href="https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1" class="external-link">https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1</a></p></li>
<li><p>UNITE fungal database from <a href="https://unite.ut.ee/repository.php" class="external-link">https://unite.ut.ee/repository.php</a></p></li>
<li><p>Up to two other reference databases. The user will need to
reformat headers exactly as outlined in the <a href="https://benjjneb.github.io/dada2/training.html" class="external-link">DADA2 database
format</a>, and similar to the SILVA database format. The user can then
specify the path to the database in the input file. The database should
be in fasta format.</p></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="faq">FAQ<a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<p>In progress</p>
</div>
<div class="section level2">
<h2 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h2>
<p>In progress</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martha A. Sudermann.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
