<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="demulticoder">
<title>Mixed barcode example • demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mixed barcode example">
<meta property="og:description" content="demulticoder">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/01_getting_started.html">Get Started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/02_documentation.html">Documentation</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-example-workflows">Example Workflows</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-example-workflows">
    <a class="dropdown-item" href="../articles/03_mixed_barcode_example.html">Mixed Barcode Example</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Main Functions</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Mixed barcode example</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/03_mixed_barcode_example.Rmd" class="external-link"><code>vignettes/03_mixed_barcode_example.Rmd</code></a></small>
      <div class="d-none name"><code>03_mixed_barcode_example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="demonstration-of-how-to-use-demulticoder-on-a-dataset-that-is-actually-three-separate-datasets-rps10-its-and-16s-at-once">Demonstration of how to use demulticoder on a dataset that is
actually three separate datasets (RPS10, ITS, and 16S) at once<a class="anchor" aria-label="anchor" href="#demonstration-of-how-to-use-demulticoder-on-a-dataset-that-is-actually-three-separate-datasets-rps10-its-and-16s-at-once"></a>
</h2>
<p>This is a small dataset comprised 24 samples. In three separate
experiments, 16S, ITS, and 16S genes/regions were amplified, following
an experiment in which some soil samples were treated with diallyl
disulfide (organosulfur biocontrol compound derived from garlic), half
were exposed to anaerobic conditions vs. aerobic conditions, and half
the samples were one soil type (Tulelake) vs. another (Madras). The goal
was to examine the bacterial, fungal, and oomycete communities within
these different treatment groups. Reads were obtained following three
separate Illumina MiSeq runs. The following <em>Demulticoder</em>
workflow below picks up after these runs. The read data has already been
demultiplexed, and for each sample, there is 16S, ITS, and rps10 reads
within three separate FASTQ files per sample.</p>
<p>The goal of this workflow is to demonstrate that you can analyze
simultaneously (but with different parameters), multiple datasets,
related or not. Final matrices are separated by barcode, to avoid any
violation of assumptions regarding experimental design. If the user so
desires, they could combine the final matrices, depending on how the
experiment was designed.</p>
<div class="section level3">
<h3 id="step-1-prepare-metadata-csv-file">Step 1-Prepare metadata.csv file<a class="anchor" aria-label="anchor" href="#step-1-prepare-metadata-csv-file"></a>
</h3>
<p>You will see that sample names are unique for Here is the CSV file
looks like:</p>
<pre class="csv"><code>sample_name,primer_name,incubation,soil,treatment
1_1_01,r16S,Anaerobic,Madras,untreated
1_2_01,r16S,Anaerobic,Madras,untreated
1_3_01,r16S,Anaerobic,Madras,untreated
2_1_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_2_01,r16S,Anaerobic,Madras,DADS_1gal_A
2_3_01,r16S,Anaerobic,Madras,DADS_1gal_A
3_1_01,r16S,Anaerobic,Tulelake,untreated
3_2_01,r16S,Anaerobic,Tulelake,untreated
3_3_01,r16S,Anaerobic,Tulelake,untreated
4_1_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_2_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
4_3_01,r16S,Anaerobic,Tulelake,DADS_1gal_A
5_1_01,r16S,Aerobic,Madras,untreated
5_2_01,r16S,Aerobic,Madras,untreated
5_3_01,r16S,Aerobic,Madras,untreated
6_1_01,r16S,Aerobic,Madras,DADS_1gal_A
6_2_01,r16S,Aerobic,Madras,DADS_1gal_A
6_3_01,r16S,Aerobic,Madras,DADS_1gal_A
7_1_01,r16S,Aerobic,Tulelake,untreated
7_2_01,r16S,Aerobic,Tulelake,untreated
7_3_01,r16S,Aerobic,Tulelake,untreated
8_1_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_2_01,r16S,Aerobic,Tulelake,DADS_1gal_A
8_3_01,r16S,Aerobic,Tulelake,DADS_1gal_A
Dung1_1_S221_L001_02,its,Anaerobic ,Madras,untreated
Dung1_1_S222,rps10,Anaerobic ,Madras,untreated
Dung1_2_S222_L001_02,its,Anaerobic ,Madras,untreated
Dung1_2_S223,rps10,Anaerobic ,Madras,DADS_1gal_A
Dung1_3_S223_L001_02,its,Anaerobic ,Madras,DADS_1gal_A
Dung1_3_S224,rps10,Anaerobic ,Madras,DADS_1gal_A
Dung2_1_S224_L001_02,its,Anaerobic ,Tulelake,untreated
Dung2_1_S225,rps10,Anaerobic ,Tulelake,untreated
Dung2_2_S225_L001_02,its,Anaerobic ,Tulelake,untreated
Dung2_2_S226,rps10,Anaerobic ,Tulelake,DADS_1gal_A
Dung2_3_S226_L001_02,its,Anaerobic ,Tulelake,DADS_1gal_A
Dung2_3_S227,rps10,Anaerobic ,Tulelake,DADS_1gal_A
Dung3_1_S227_L001_02,its,Aerobic,Madras,untreated
Dung3_1_S228,rps10,Aerobic,Madras,untreated
Dung3_2_S228_L001_02,its,Aerobic,Madras,untreated
Dung3_2_S229,rps10,Aerobic,Madras,DADS_1gal_A
Dung3_3_S229_L001_02,its,Aerobic,Madras,DADS_1gal_A
Dung3_3_S230,rps10,Aerobic,Madras,DADS_1gal_A
Dung4_1_S230_L001_02,its,Aerobic,Tulelake,untreated
Dung4_1_S231,rps10,Aerobic,Tulelake,untreated
Dung4_2_S231_L001_02,its,Aerobic,Tulelake,untreated
Dung4_2_S232,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung4_3_S232_L001_02,its,Aerobic,Tulelake,DADS_1gal_A
Dung4_3_S233,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung5_1_S233_L001_02,its,Anaerobic ,Madras,untreated
Dung5_1_S234,rps10,Anaerobic ,Madras,untreated
Dung5_2_S234_L001_02,its,Anaerobic ,Madras,untreated
Dung5_2_S235,rps10,Anaerobic ,Madras,DADS_1gal_A
Dung5_3_S235_L001_02,its,Anaerobic ,Madras,DADS_1gal_A
Dung5_3_S236,rps10,Anaerobic ,Madras,DADS_1gal_A
Dung6_1_S236_L001_02,its,Anaerobic ,Tulelake,untreated
Dung6_1_S237,rps10,Anaerobic ,Tulelake,untreated
Dung6_2_S237_L001_02,its,Anaerobic ,Tulelake,untreated
Dung6_2_S238,rps10,Anaerobic ,Tulelake,DADS_1gal_A
Dung6_3_S238_L001_02,its,Anaerobic ,Tulelake,DADS_1gal_A
Dung6_3_S239,rps10,Anaerobic ,Tulelake,DADS_1gal_A
Dung7_1_S239_L001_02,its,Aerobic,Madras,untreated
Dung7_1_S240,rps10,Aerobic,Madras,untreated
Dung7_2_S240_L001_02,its,Aerobic,Madras,untreated
Dung7_2_S241,rps10,Aerobic,Madras,DADS_1gal_A
Dung7_3_S241_L001_02,its,Aerobic,Madras,DADS_1gal_A
Dung7_3_S242,rps10,Aerobic,Madras,DADS_1gal_A
Dung8_1_S242_L001_02,its,Aerobic,Tulelake,untreated
Dung8_1_S243,rps10,Aerobic,Tulelake,untreated
Dung8_2_S243_L001_02,its,Aerobic,Tulelake,untreated
Dung8_2_S244,rps10,Aerobic,Tulelake,DADS_1gal_A
Dung8_3_S244_L001_02,its,Aerobic,Tulelake,DADS_1gal_A
Dung8_3_S245,rps10,Aerobic,Tulelake,DADS_1gal_A</code></pre>
<p>In tabular format:</p>
<table class="table">
<thead><tr class="header">
<th align="left">sample_name</th>
<th align="left">primer_name</th>
<th align="left">incubation</th>
<th align="left">soil</th>
<th align="left">treatment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1_1_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">1_2_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="odd">
<td align="left">1_3_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">2_1_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">2_2_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="even">
<td align="left">2_3_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">3_1_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">3_2_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="odd">
<td align="left">3_3_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">4_1_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">4_2_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="even">
<td align="left">4_3_01</td>
<td align="left">r16S</td>
<td align="left">Anaerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">5_1_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">5_2_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="odd">
<td align="left">5_3_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">6_1_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">6_2_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="even">
<td align="left">6_3_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Madras</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">7_1_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">7_2_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="odd">
<td align="left">7_3_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">untreated</td>
</tr>
<tr class="even">
<td align="left">8_1_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="odd">
<td align="left">8_2_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
<tr class="even">
<td align="left">8_3_01</td>
<td align="left">r16S</td>
<td align="left">Aerobic</td>
<td align="left">Tulelake</td>
<td align="left">DADS_1gal_A</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-2-prepare-primerinfo_params-csv-file">Step 2-Prepare primerinfo_params.csv file<a class="anchor" aria-label="anchor" href="#step-2-prepare-primerinfo_params-csv-file"></a>
</h3>
<p>Here is the CSV file looks like:</p>
<pre class="csv"><code>primer_name,forward,reverse,already_trimmed,minCutadaptlength,multithread,verbose,maxN,maxEE_forward,maxEE_reverse,truncLen_forward,truncLen_reverse,truncQ,minLen,maxLen,minQ,trimLeft,trimRight,rm.lowcomplex,minOverlap,maxMismatch,min_asv_length
rps10,GTTGGTTAGAGYARAAGACT,ATRYYTAGAAAGAYTYGAACT,FALSE,50,TRUE,TRUE,0,5,5,0,0,5,50,Inf,0,0,0,0,15,2,50
its,GTGARTCATCGAATCTTTG,TCCTSCGCTTATTGATATGC,FALSE,50,TRUE,TRUE,0,2,2,0,0,5,50,295,0,0,0,0,15,2,50
r16S,GTGYCAGCMGCCGCGGTAA,GGACTACNVGGGTWTCTAAT,TRUE,50,TRUE,TRUE,0,2,2,0,0,5,50,Inf,0,0,0,0,15,2,50</code></pre>
<p>In tabular format:</p>
<table class="table">
<colgroup>
<col width="4%">
<col width="7%">
<col width="8%">
<col width="5%">
<col width="6%">
<col width="4%">
<col width="2%">
<col width="1%">
<col width="5%">
<col width="5%">
<col width="6%">
<col width="6%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="3%">
<col width="3%">
<col width="5%">
<col width="4%">
<col width="4%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">primer_name</th>
<th align="left">forward</th>
<th align="left">reverse</th>
<th align="left">already_trimmed</th>
<th align="right">minCutadaptlength</th>
<th align="left">multithread</th>
<th align="left">verbose</th>
<th align="right">maxN</th>
<th align="right">maxEE_forward</th>
<th align="right">maxEE_reverse</th>
<th align="right">truncLen_forward</th>
<th align="right">truncLen_reverse</th>
<th align="right">truncQ</th>
<th align="right">minLen</th>
<th align="right">maxLen</th>
<th align="right">minQ</th>
<th align="right">trimLeft</th>
<th align="right">trimRight</th>
<th align="right">rm.lowcomplex</th>
<th align="right">minOverlap</th>
<th align="right">maxMismatch</th>
<th align="right">min_asv_length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rps10</td>
<td align="left">GTTGGTTAGAGYARAAGACT</td>
<td align="left">ATRYYTAGAAAGAYTYGAACT</td>
<td align="left">FALSE</td>
<td align="right">50</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">50</td>
<td align="right">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">its</td>
<td align="left">GTGARTCATCGAATCTTTG</td>
<td align="left">TCCTSCGCTTATTGATATGC</td>
<td align="left">FALSE</td>
<td align="right">50</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">50</td>
<td align="right">295</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">r16S</td>
<td align="left">GTGYCAGCMGCCGCGGTAA</td>
<td align="left">GGACTACNVGGGTWTCTAAT</td>
<td align="left">TRUE</td>
<td align="right">50</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">50</td>
<td align="right">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">50</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-3-remove-ns-and-create-directory-structure-for-downstream-steps">Step 3-Remove N’s and create directory structure for downstream
steps<a class="anchor" aria-label="anchor" href="#step-3-remove-ns-and-create-directory-structure-for-downstream-steps"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outputs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="st">"~/demulticoder_benchmarking/demulticoder_combined_analysis_dads/data"</span>, </span>
<span>  output_directory <span class="op">=</span> <span class="st">"~/demulticoder_benchmarking/package_vignettes/output_multiple_datasets"</span>, </span>
<span>  tempdir_path <span class="op">=</span> <span class="st">"~/demulticoder_benchmarking/temp"</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">3</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">3</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">72</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (5): sample_name, primer_name, incubation, soil, treatment</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; Creating output directory: /home/marthasudermann/demulticoder_benchmarking/temp/demulticoder_run/prefiltered_sequences</span></span></code></pre></div>
<p><img src="prepare%20reads-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="step-4-run-cutadapt-to-remove-primers-and-then-trim-reads-with-dada2-filterandtrim-function">Step 4-Run Cutadapt to remove primers and then trim reads with DADA2
filterAndTrim function<a class="anchor" aria-label="anchor" href="#step-4-run-cutadapt-to-remove-primers-and-then-trim-reads-with-dada2-filterandtrim-function"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running Cutadapt 3.5 for its sequence data </span></span>
<span><span class="co">#&gt; Running Cutadapt 3.5 for rps10 sequence data</span></span></code></pre></div>
<p><img src="Remove%20primers%20and%20trim%20reads-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="step-5-core-asv-inference-step">Step 5-Core ASV inference step<a class="anchor" aria-label="anchor" href="#step-5-core-asv-inference-step"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; 106034910 total bases in 466183 reads from 9 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 .........</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt;    selfConsist step 5</span></span>
<span><span class="co">#&gt; Convergence after  5  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the Forward read of primer pair its</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 78090 reads in 14475 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 45955 reads in 9604 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 42450 reads in 9362 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 47513 reads in 13905 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 37652 reads in 12257 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 50196 reads in 15866 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 33726 reads in 7781 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 48490 reads in 10219 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 82111 reads in 14763 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 65012 reads in 18955 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 50432 reads in 13490 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 44904 reads in 13902 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 35611 reads in 7806 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 29981 reads in 7103 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 35426 reads in 8032 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 40535 reads in 13049 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 37788 reads in 11717 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 25282 reads in 8508 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 24576 reads in 5694 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 31335 reads in 7645 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 21 - 37061 reads in 7258 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 22 - 30407 reads in 9859 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 23 - 42100 reads in 11928 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 24 - 57989 reads in 16385 unique sequences.</span></span>
<span><span class="co">#&gt; 105499828 total bases in 466183 reads from 9 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 .........</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt; Convergence after  4  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the Reverse read of primer pair its</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 78090 reads in 18561 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 45955 reads in 11830 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 42450 reads in 12227 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 47513 reads in 12720 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 37652 reads in 15089 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 50196 reads in 19051 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 33726 reads in 12311 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 48490 reads in 17792 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 82111 reads in 19363 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 65012 reads in 16307 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 50432 reads in 12229 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 44904 reads in 12850 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 35611 reads in 9856 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 29981 reads in 8435 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 35426 reads in 10431 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 40535 reads in 10489 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 37788 reads in 18061 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 25282 reads in 11321 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 24576 reads in 13200 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 31335 reads in 13584 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 21 - 37061 reads in 10363 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 22 - 30407 reads in 9253 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 23 - 42100 reads in 10947 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 24 - 57989 reads in 13634 unique sequences.</span></span>
<span><span class="co">#&gt; 76532 paired-reads (in 235 unique pairings) successfully merged out of 77154 (in 268 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 44826 paired-reads (in 182 unique pairings) successfully merged out of 45262 (in 211 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 41241 paired-reads (in 177 unique pairings) successfully merged out of 41588 (in 200 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 46443 paired-reads (in 134 unique pairings) successfully merged out of 46791 (in 153 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 36763 paired-reads (in 118 unique pairings) successfully merged out of 36988 (in 133 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 49327 paired-reads (in 138 unique pairings) successfully merged out of 49592 (in 153 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 32710 paired-reads (in 136 unique pairings) successfully merged out of 33009 (in 158 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 47112 paired-reads (in 167 unique pairings) successfully merged out of 47654 (in 200 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 80706 paired-reads (in 239 unique pairings) successfully merged out of 81276 (in 280 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 63826 paired-reads (in 177 unique pairings) successfully merged out of 64176 (in 202 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 49621 paired-reads (in 117 unique pairings) successfully merged out of 49737 (in 131 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 43962 paired-reads (in 136 unique pairings) successfully merged out of 44349 (in 154 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 34455 paired-reads (in 154 unique pairings) successfully merged out of 34905 (in 181 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 28736 paired-reads (in 155 unique pairings) successfully merged out of 29390 (in 181 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 33960 paired-reads (in 163 unique pairings) successfully merged out of 34687 (in 188 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 39776 paired-reads (in 129 unique pairings) successfully merged out of 39921 (in 142 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 36836 paired-reads (in 104 unique pairings) successfully merged out of 36988 (in 113 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 24650 paired-reads (in 88 unique pairings) successfully merged out of 24794 (in 96 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 23672 paired-reads (in 102 unique pairings) successfully merged out of 23918 (in 118 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 30099 paired-reads (in 103 unique pairings) successfully merged out of 30492 (in 123 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 35951 paired-reads (in 148 unique pairings) successfully merged out of 36386 (in 166 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 29757 paired-reads (in 117 unique pairings) successfully merged out of 29919 (in 126 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 41280 paired-reads (in 143 unique pairings) successfully merged out of 41588 (in 162 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 56900 paired-reads (in 156 unique pairings) successfully merged out of 57408 (in 180 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Identified 28 bimeras out of 869 input sequences.</span></span></code></pre></div>
<p><img src="ASV%20inference-1.png" width="576"></p>
<pre><code><span><span class="co">#&gt; 126787839 total bases in 451263 reads from 6 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 ......</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt;    selfConsist step 5</span></span>
<span><span class="co">#&gt; Convergence after  5  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the Forward read of primer pair rps10</span></span>
<span><span class="co">#&gt; Sample 1 - 20082 reads in 15404 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 81794 reads in 50235 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 86486 reads in 57261 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 64939 reads in 48091 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 79738 reads in 50805 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 118224 reads in 69385 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 19427 reads in 13201 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 73752 reads in 47699 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 23829 reads in 16766 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 69370 reads in 41438 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 134589 reads in 79909 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 88315 reads in 55387 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 43607 reads in 37646 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 68342 reads in 51601 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 14251 reads in 12014 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 10633 reads in 9740 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 99725 reads in 58227 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 82697 reads in 49568 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 58679 reads in 39009 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 58661 reads in 40824 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 21 - 46627 reads in 39317 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 22 - 60360 reads in 46852 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 23 - 14006 reads in 11814 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 24 - 22369 reads in 18243 unique sequences.</span></span>
<span><span class="co">#&gt; 126347798 total bases in 451263 reads from 6 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 ......</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt;    selfConsist step 5</span></span>
<span><span class="co">#&gt;    selfConsist step 6</span></span>
<span><span class="co">#&gt;    selfConsist step 7</span></span>
<span><span class="co">#&gt; Convergence after  7  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the Reverse read of primer pair rps10</span></span>
<span><span class="co">#&gt; Sample 1 - 20082 reads in 14654 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 81794 reads in 56653 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 86486 reads in 61939 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 64939 reads in 41617 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 79738 reads in 48181 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 118224 reads in 71942 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 19427 reads in 14744 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 73752 reads in 53141 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 23829 reads in 17238 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 69370 reads in 41800 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 134589 reads in 73468 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 88315 reads in 51668 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 43607 reads in 29142 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 68342 reads in 42353 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 14251 reads in 10551 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 10633 reads in 6488 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 99725 reads in 56068 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 82697 reads in 47196 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 58679 reads in 43743 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 58661 reads in 40142 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 21 - 46627 reads in 33507 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 22 - 60360 reads in 31906 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 23 - 14006 reads in 8796 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 24 - 22369 reads in 13421 unique sequences.</span></span>
<span><span class="co">#&gt; 19680 paired-reads (in 36 unique pairings) successfully merged out of 19998 (in 91 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 77741 paired-reads (in 72 unique pairings) successfully merged out of 81624 (in 162 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 81094 paired-reads (in 72 unique pairings) successfully merged out of 86377 (in 191 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 60693 paired-reads (in 192 unique pairings) successfully merged out of 64495 (in 586 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 75768 paired-reads (in 125 unique pairings) successfully merged out of 79396 (in 371 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 113422 paired-reads (in 172 unique pairings) successfully merged out of 117935 (in 471 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 18957 paired-reads (in 42 unique pairings) successfully merged out of 19137 (in 78 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 70795 paired-reads (in 79 unique pairings) successfully merged out of 73418 (in 146 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 23206 paired-reads (in 26 unique pairings) successfully merged out of 23760 (in 54 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 66712 paired-reads (in 81 unique pairings) successfully merged out of 69287 (in 241 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 128424 paired-reads (in 143 unique pairings) successfully merged out of 134179 (in 349 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 85426 paired-reads (in 85 unique pairings) successfully merged out of 88207 (in 264 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 41456 paired-reads (in 37 unique pairings) successfully merged out of 43409 (in 115 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 65184 paired-reads (in 84 unique pairings) successfully merged out of 68075 (in 205 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 13965 paired-reads (in 31 unique pairings) successfully merged out of 14174 (in 66 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 9229 paired-reads (in 16 unique pairings) successfully merged out of 10463 (in 76 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 96759 paired-reads (in 63 unique pairings) successfully merged out of 99648 (in 198 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 78760 paired-reads (in 80 unique pairings) successfully merged out of 82349 (in 244 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 52961 paired-reads (in 106 unique pairings) successfully merged out of 58370 (in 267 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 52492 paired-reads (in 81 unique pairings) successfully merged out of 58507 (in 209 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 44849 paired-reads (in 60 unique pairings) successfully merged out of 45994 (in 190 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 58198 paired-reads (in 55 unique pairings) successfully merged out of 60194 (in 175 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span>
<span><span class="co">#&gt; 13004 paired-reads (in 19 unique pairings) successfully merged out of 13817 (in 65 pairings) input.</span></span>
<span><span class="co">#&gt; 21168 paired-reads (in 38 unique pairings) successfully merged out of 22205 (in 167 pairings) input.</span></span>
<span><span class="co">#&gt; Duplicate sequences in merged output.</span></span></code></pre>
<p><img src="ASV%20inference-2.png" width="576"></p>
<pre><code><span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Identified 548 bimeras out of 732 input sequences.</span></span></code></pre>
<p><img src="ASV%20inference-3.png" width="576"><img src="ASV%20inference-4.png" width="576"></p>
<pre><code><span><span class="co">#&gt; $its</span></span>
<span><span class="co">#&gt; [1] "~/demulticoder_benchmarking/temp/demulticoder_run/asvabund_matrixDADA2_its.RData"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $rps10</span></span>
<span><span class="co">#&gt; [1] "~/demulticoder_benchmarking/temp/demulticoder_run/asvabund_matrixDADA2_rps10.RData"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="step-6-assign-taxonomy-step">Step 6-Assign taxonomy step<a class="anchor" aria-label="anchor" href="#step-6-assign-taxonomy-step"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  db_its <span class="op">=</span> <span class="st">"sh_general_release_dynamic_all_25.07.2023.fasta"</span>,</span>
<span>  db_rps10 <span class="op">=</span> <span class="st">"oomycetedb.fasta"</span>,</span>
<span>  db_16S<span class="op">=</span><span class="st">"silva_nr99_v138.1_wSpecies_train_set.fa"</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>  overwrite_existing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt;          samplename_barcode  input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1  Dung1_1_S221_L001_02_its 107502    78090     77364     77495  76532   76477</span></span>
<span><span class="co">#&gt; 2  Dung1_2_S222_L001_02_its  61061    45955     45480     45404  44826   44775</span></span>
<span><span class="co">#&gt; 3  Dung1_3_S223_L001_02_its  61239    42450     41879     41813  41241   41212</span></span>
<span><span class="co">#&gt; 4  Dung2_1_S224_L001_02_its  63088    47513     47053     46978  46443   46146</span></span>
<span><span class="co">#&gt; 5  Dung2_2_S225_L001_02_its  58055    37652     37262     37094  36763   36571</span></span>
<span><span class="co">#&gt; 6  Dung2_3_S226_L001_02_its  80809    50196     49877     49673  49327   49046</span></span>
<span><span class="co">#&gt; 7  Dung3_1_S227_L001_02_its  52495    33726     33220     33165  32710   32681</span></span>
<span><span class="co">#&gt; 8  Dung3_2_S228_L001_02_its  81501    48490     48003     47784  47112   47069</span></span>
<span><span class="co">#&gt; 9  Dung3_3_S229_L001_02_its 109828    82111     81638     81435  80706   80549</span></span>
<span><span class="co">#&gt; 10 Dung4_1_S230_L001_02_its  89140    65012     64527     64413  63826   63378</span></span>
<span><span class="co">#&gt; 11 Dung4_2_S231_L001_02_its  63345    50432     49931     49931  49621   49267</span></span>
<span><span class="co">#&gt; 12 Dung4_3_S232_L001_02_its  59459    44904     44543     44470  43962   43633</span></span>
<span><span class="co">#&gt; 13 Dung5_1_S233_L001_02_its  46202    35611     35150     35057  34455   34400</span></span>
<span><span class="co">#&gt; 14 Dung5_2_S234_L001_02_its  39533    29981     29579     29519  28736   28722</span></span>
<span><span class="co">#&gt; 15 Dung5_3_S235_L001_02_its  56309    35426     34922     34913  33960   33933</span></span>
<span><span class="co">#&gt; 16 Dung6_1_S236_L001_02_its  54091    40535     40136     40108  39776   39536</span></span>
<span><span class="co">#&gt; 17 Dung6_2_S237_L001_02_its  83307    37788     37413     37100  36836   36644</span></span>
<span><span class="co">#&gt; 18 Dung6_3_S238_L001_02_its  49829    25282     24998     24834  24650   24515</span></span>
<span><span class="co">#&gt; 19 Dung7_1_S239_L001_02_its  57541    24576     24220     23984  23672   23652</span></span>
<span><span class="co">#&gt; 20 Dung7_2_S240_L001_02_its  54654    31335     30784     30639  30099   30043</span></span>
<span><span class="co">#&gt; 21 Dung7_3_S241_L001_02_its  49902    37061     36566     36567  35951   35934</span></span>
<span><span class="co">#&gt; 22 Dung8_1_S242_L001_02_its  42973    30407     30060     30025  29757   29561</span></span>
<span><span class="co">#&gt; 23 Dung8_2_S243_L001_02_its  58646    42100     41795     41701  41280   41090</span></span>
<span><span class="co">#&gt; 24 Dung8_3_S244_L001_02_its  77154    57989     57602     57554  56900   56623</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt; Duplicate sequences detected and merged.</span></span>
<span><span class="co">#&gt;    samplename_barcode  input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1  Dung1_1_S222_rps10  31012    20082     19999     20028  19680   19259</span></span>
<span><span class="co">#&gt; 2  Dung1_2_S223_rps10 117900    81794     81719     81690  77741   75601</span></span>
<span><span class="co">#&gt; 3  Dung1_3_S224_rps10 132855    86486     86438     86422  81094   78814</span></span>
<span><span class="co">#&gt; 4  Dung2_1_S225_rps10  96066    64939     64784     64631  60693   48037</span></span>
<span><span class="co">#&gt; 5  Dung2_2_S226_rps10 111356    79738     79519     79577  75768   64714</span></span>
<span><span class="co">#&gt; 6  Dung2_3_S227_rps10 159936   118224    117976    118170 113422  100408</span></span>
<span><span class="co">#&gt; 7  Dung3_1_S228_rps10  28596    19427     19389     19156  18957   18414</span></span>
<span><span class="co">#&gt; 8  Dung3_2_S229_rps10 110053    73752     73617     73544  70795   66612</span></span>
<span><span class="co">#&gt; 9  Dung3_3_S230_rps10  35693    23829     23784     23793  23206   22878</span></span>
<span><span class="co">#&gt; 10 Dung4_1_S231_rps10  94160    69370     69320     69327  66712   63797</span></span>
<span><span class="co">#&gt; 11 Dung4_2_S232_rps10 181941   134589    134530    134227 128424  114055</span></span>
<span><span class="co">#&gt; 12 Dung4_3_S233_rps10 120075    88315     88238     88236  85426   79149</span></span>
<span><span class="co">#&gt; 13 Dung5_1_S234_rps10  89019    43607     43478     43503  41456   38404</span></span>
<span><span class="co">#&gt; 14 Dung5_2_S235_rps10 104996    68342     68208     68192  65184   58908</span></span>
<span><span class="co">#&gt; 15 Dung5_3_S236_rps10  25501    14251     14203     14187  13965   13760</span></span>
<span><span class="co">#&gt; 16 Dung6_1_S237_rps10  24523    10633     10478     10606   9229    8739</span></span>
<span><span class="co">#&gt; 17 Dung6_2_S238_rps10 147048    99725     99704     99664  96759   94361</span></span>
<span><span class="co">#&gt; 18 Dung6_3_S239_rps10 114844    82697     82620     82402  78760   74217</span></span>
<span><span class="co">#&gt; 19 Dung7_1_S240_rps10  86521    58679     58432     58604  52961   49033</span></span>
<span><span class="co">#&gt; 20 Dung7_2_S241_rps10  86204    58661     58550     58608  52492   48440</span></span>
<span><span class="co">#&gt; 21 Dung7_3_S242_rps10  90932    46627     46444     46136  44849   43436</span></span>
<span><span class="co">#&gt; 22 Dung8_1_S243_rps10  95078    60360     60213     60329  58198   54921</span></span>
<span><span class="co">#&gt; 23 Dung8_2_S244_rps10  23896    14006     13876     13927  13004   12520</span></span>
<span><span class="co">#&gt; 24 Dung8_3_S245_rps10  36871    22369     22225     22326  21168   20016</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-7-convert-asv-matrix-to-taxmap-and-phyloseq-objects-with-one-function">Step 7-convert asv matrix to taxmap and phyloseq objects with one
function<a class="anchor" aria-label="anchor" href="#step-7-convert-asv-matrix-to-taxmap-and-phyloseq-objects-with-one-function"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">outputs</span>, save_outputs<span class="op">=</span><span class="cn">TRUE</span>, overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">841</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">27</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (24): Dung1_1_S221_L001_02_its, Dung1_2_S222_L001_02_its, Dung1_3_S223_L...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For its dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/demulticoder_benchmarking/package_vignettes/output_multiple_datasets/obj_dada_its.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/demulticoder_benchmarking/package_vignettes/output_multiple_datasets/phylo_obj_its.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">184</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">27</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (24): Dung1_1_S222_rps10, Dung1_2_S223_rps10, Dung1_3_S224_rps10, Dung2_...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For rps10 dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/demulticoder_benchmarking/package_vignettes/output_multiple_datasets/obj_dada_rps10.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/demulticoder_benchmarking/package_vignettes/output_multiple_datasets/phylo_obj_rps10.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="use-phyloseq-objects-to-examine-the-community-composition-of-each-sample">Use phyloseq objects to examine the community composition of each
sample<a class="anchor" aria-label="anchor" href="#use-phyloseq-objects-to-examine-the-community-composition-of-each-sample"></a>
</h3>
<p>Here we demonstrate how to make a stacked bar plot of the relative
abundance of taxa by sample for the 16S-barcoded samples</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_its</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                                        </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.005</span><span class="op">)</span> <span class="op">%&gt;%</span>                      </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Phyloseq%20demonstration%2016S-1.png" width="1920"></p>
<p>Here we demonstrate how to make a stacked bar plot of the relative
abundance of taxa by sample for the ITS-barcoded samples</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_its</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                                        </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.005</span><span class="op">)</span> <span class="op">%&gt;%</span>                      </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Phyloseq%20demonstration%20ITS-1.png" width="1920"></p>
<p>Finally,we demonstrate how to make a stacked bar plot of the relative
abundance of taxa by sample for the rps10-barcoded samples</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_rps10</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                                        </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.0005</span><span class="op">)</span> <span class="op">%&gt;%</span>                      </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Phyloseq%20demonstration%20rps10-1.png" width="768"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.1.2 (2021-11-01)</span></span>
<span><span class="co">#&gt;  os       Pop!_OS 22.04 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/Los_Angeles</span></span>
<span><span class="co">#&gt;  date     2024-07-29</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">!</span> <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;    ade4                   1.7-22     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    ape                    5.8        <span style="color: #555555;">2024-04-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    Biobase                2.54.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    BiocGenerics           0.40.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    BiocParallel           1.28.3     <span style="color: #555555;">2021-12-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    biomformat             1.22.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    Biostrings             2.62.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    bit                    4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    bit64                  4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.0.2)</span></span></span>
<span><span class="co">#&gt;    bitops                 1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.1)</span></span></span>
<span><span class="co">#&gt;    bslib                  0.7.0      <span style="color: #555555;">2024-03-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    cachem                 1.1.0      <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    cli                    3.6.3      <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    cluster                2.1.2      <span style="color: #555555;">2021-04-17</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.1)</span></span></span>
<span><span class="co">#&gt;    codetools              0.2-18     <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.0.3)</span></span></span>
<span><span class="co">#&gt;    colorspace             2.1-0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    crayon                 1.5.3      <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    dada2                  1.30.0     <span style="color: #555555;">2024-01-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">bioc_xgit (@ec87892)</span></span></span>
<span><span class="co">#&gt;    data.table             1.15.4     <span style="color: #555555;">2024-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    DBI                    1.2.2      <span style="color: #555555;">2024-02-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    DelayedArray           0.20.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    deldir                 2.0-4      <span style="color: #555555;">2024-02-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> demulticoder         * <span style="color: #BB00BB; font-weight: bold;">0.0.0.9000</span> <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[?]</span> <span style="color: #BB00BB; font-weight: bold;">load_all()</span></span></span>
<span><span class="co">#&gt;    desc                   1.4.3      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    devtools               2.4.3      <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    digest                 0.6.36     <span style="color: #555555;">2024-06-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    dplyr                  1.1.4      <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    ellipsis               0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    evaluate               0.24.0     <span style="color: #555555;">2024-06-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    fansi                  1.0.6      <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    farver                 2.1.2      <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    fastmap                1.2.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    foreach                1.5.2      <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    fs                     1.6.4      <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    furrr                  0.3.1      <span style="color: #555555;">2022-08-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    future                 1.33.2     <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    generics               0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    GenomeInfoDb           1.30.1     <span style="color: #555555;">2022-01-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    GenomeInfoDbData       1.2.7      <span style="color: #555555;">2024-01-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    GenomicAlignments      1.30.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    GenomicRanges          1.46.1     <span style="color: #555555;">2021-11-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    ggplot2                3.5.1      <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    globals                0.16.3     <span style="color: #555555;">2024-03-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    glue                   1.7.0      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    gtable                 0.3.5      <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    highr                  0.11       <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    hms                    1.1.3      <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    htmltools              0.5.8.1    <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    hwriter                1.3.2.1    <span style="color: #555555;">2022-04-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    igraph                 2.0.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    interp                 1.1-6      <span style="color: #555555;">2024-01-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    IRanges                2.28.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    iterators              1.0.14     <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    jpeg                   0.1-10     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    jquerylib              0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    jsonlite               1.8.8      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    knitr                  1.48       <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    labeling               0.4.3      <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    lattice                0.20-45    <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.1)</span></span></span>
<span><span class="co">#&gt;    latticeExtra           0.6-30     <span style="color: #555555;">2022-07-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    lazyeval               0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.0.1)</span></span></span>
<span><span class="co">#&gt;    lifecycle              1.0.4      <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    listenv                0.9.1      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    magrittr               2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    MASS                   7.3-55     <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    Matrix                 1.4-0      <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    MatrixGenerics         1.6.0      <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    matrixStats            1.2.0      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    memoise                2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    metacoder              0.3.7      <span style="color: #555555;">2024-02-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    mgcv                   1.8-39     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    multtest               2.50.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    munsell                0.5.1      <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    nlme                   3.1-155    <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    parallelly             1.37.1     <span style="color: #555555;">2024-02-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    permute                0.9-7      <span style="color: #555555;">2022-01-27</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    phyloseq               1.46.0     <span style="color: #555555;">2024-01-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">bioc_xgit (@7320133)</span></span></span>
<span><span class="co">#&gt;    pillar                 1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    pkgbuild               1.4.4      <span style="color: #555555;">2024-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    pkgconfig              2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    pkgdown                2.0.7      <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    pkgload                1.3.4      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    plyr                   1.8.9      <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    png                    0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    purrr                * 1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    R6                     2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    ragg                   1.2.1      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    RColorBrewer           1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    Rcpp                   1.0.13     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    RcppParallel           5.1.7      <span style="color: #555555;">2023-02-27</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    RCurl                  1.98-1.16  <span style="color: #555555;">2024-07-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    readr                  2.1.5      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    remotes                2.5.0      <span style="color: #555555;">2024-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    reshape2               1.4.4      <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.0.1)</span></span></span>
<span><span class="co">#&gt;    rhdf5                  2.38.1     <span style="color: #555555;">2022-03-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    rhdf5filters           1.6.0      <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    Rhdf5lib               1.16.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    rlang                  1.1.4      <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    rmarkdown              2.27       <span style="color: #555555;">2024-05-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    rprojroot              2.0.4      <span style="color: #555555;">2023-11-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    Rsamtools              2.10.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    rstudioapi             0.16.0     <span style="color: #555555;">2024-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    S4Vectors              0.32.4     <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    sass                   0.4.9      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    scales                 1.3.0      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    sessioninfo            1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    ShortRead              1.60.0     <span style="color: #555555;">2024-01-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">bioc_xgit (@4304db4)</span></span></span>
<span><span class="co">#&gt;    stringi                1.8.4      <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    stringr                1.5.1      <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    SummarizedExperiment   1.24.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    survival               3.2-13     <span style="color: #555555;">2021-08-24</span> <span style="color: #555555;">[5]</span> <span style="color: #555555;">CRAN (R 4.1.1)</span></span></span>
<span><span class="co">#&gt;    systemfonts            1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    textshaping            0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.1)</span></span></span>
<span><span class="co">#&gt;    tibble                 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    tidyr                  1.3.1      <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    tidyselect             1.2.1      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    tzdb                   0.4.0      <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    usethis                2.1.5      <span style="color: #555555;">2021-12-09</span> <span style="color: #555555;">[4]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    utf8                   1.2.4      <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    vctrs                  0.6.5      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    vegan                  2.6-6.1    <span style="color: #555555;">2024-05-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    viridisLite            0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    vroom                  1.6.5      <span style="color: #555555;">2023-12-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    withr                  3.0.0      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    xfun                   0.46       <span style="color: #555555;">2024-07-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    XVector                0.34.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    yaml                   2.3.9      <span style="color: #555555;">2024-07-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span></span>
<span><span class="co">#&gt;    zlibbioc               1.40.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /tmp/Rtmpni2SLp/temp_libpath2824e9279530a5</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [4] /usr/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [5] /usr/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martha A. Sudermann, Zachary S. L Foster, Hung Phan, Samantha Dawson, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
