<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>16S Mothur SOP Validation â€¢ demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="16S Mothur SOP Validation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes">
<li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>16S Mothur SOP Validation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/articles/DADA2_16S_mothur_validation.Rmd" class="external-link"><code>vignettes/articles/DADA2_16S_mothur_validation.Rmd</code></a></small>
      <div class="d-none name"><code>DADA2_16S_mothur_validation.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>This vignette shows how demulticoder was used to analyze the mothur
16S SOP dataset featured in <strong><code>dada2</code></strong>
tutorials</p>
<p>The raw read files and information on the dataset used for this
analysis can be found <a href="https://mothur.org/wiki/miseq_sop/#preparing-for-analysis" class="external-link">here</a></p>
</div>
<div class="section level3">
<h3 id="load-the-demulticoder-r-package-and-necessary-dependencies">Load the demulticoder R package and necessary dependencies<a class="anchor" aria-label="anchor" href="#load-the-demulticoder-r-package-and-necessary-dependencies"></a>
</h3>
<p>For now, the package will be loaded by retrieving it from GitHub. We
are submitting package to CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"grunwaldlab/demulticoder"</span>, force<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://grunwaldlab.github.io/demulticoder/" class="external-link">"demulticoder"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://grunwaldlab.github.io/metacoder_documentation/" class="external-link">"metacoder"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">"phyloseq"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"RCurl"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup-data-directory">Setup data directory<a class="anchor" aria-label="anchor" href="#setup-data-directory"></a>
</h3>
<p>Download read files and associated data in your specified location.
The default is to write to <strong><code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code></strong>.</p>
<p>Ensure your computer has sufficient storage and that you change the
<strong>destination_folder</strong> path to a location where you have
space and can save files to.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">temp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">demulticoder_temp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">temp_dir</span>, <span class="st">"demulticoder_sop_run"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">demulticoder_temp_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Adjust output directory names as you like</span></span>
<span><span class="va">run_folder</span> <span class="op">&lt;-</span> <span class="va">demulticoder_temp_dir</span></span>
<span><span class="va">data_folder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">run_folder</span>, <span class="st">"MiSeq_SOP"</span><span class="op">)</span></span>
<span><span class="va">output_folder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">run_folder</span>, <span class="st">"sop_16S_demulticoder_outputs"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mothur_sop_path</span> <span class="op">&lt;-</span> <span class="st">"https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip"</span></span>
<span><span class="va">downloaded_read_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">run_folder</span>, <span class="st">"miseqsopdata.zip"</span><span class="op">)</span></span>
<span></span>
<span> <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">downloaded_read_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>   <span class="va">bin</span> <span class="op">&lt;-</span> <span class="fu">RCurl</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RCurl/man/getBinaryURL.html" class="external-link">getBinaryURL</a></span><span class="op">(</span><span class="va">mothur_sop_path</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/readBin.html" class="external-link">writeBin</a></span><span class="op">(</span><span class="va">bin</span>, <span class="va">downloaded_read_path</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="va">downloaded_read_path</span>, exdir <span class="op">=</span> <span class="va">run_folder</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Files downloaded:"</span>, <span class="va">downloaded_read_path</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">run_folder</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Print contents of the directory</span></span>
<span> <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Files already exist:"</span>, <span class="va">downloaded_read_path</span>, <span class="st">"! Skipping download.\n"</span><span class="op">)</span></span>
<span> <span class="op">}</span></span>
<span><span class="co">#&gt; Files downloaded: /tmp/Rtmp8rQQx4/demulticoder_sop_run/miseqsopdata.zip </span></span>
<span><span class="co">#&gt; [1] "MiSeq_SOP"        "miseqsopdata.zip"</span></span></code></pre></div>
<p>We now have an unzipped directory in our desired destination called
<strong>MiSeq_SOP</strong></p>
<p>We will also need to rename our files so they are more in line with
the file name conventions specified by demulticoder <a href="https://grunwaldlab.github.io/demulticoder/articles/Documentation.html" class="external-link">see
Documentation</a></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#We will remove the "001 appended at the end of the filename</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">data_folder</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">file</span> <span class="kw">in</span> <span class="va">files</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">new_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_001"</span>, <span class="st">""</span>, <span class="va">file</span><span class="op">)</span></span>
<span>  <span class="va">old_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_folder</span>, <span class="va">file</span><span class="op">)</span></span>
<span>  <span class="va">new_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_folder</span>, <span class="va">new_name</span><span class="op">)</span></span>
<span>  <span class="va">rename_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.rename</a></span><span class="op">(</span><span class="va">old_file_path</span>, <span class="va">new_file_path</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="add-csv-input-files-to-the-data-directory">Add CSV input files to the data directory<a class="anchor" aria-label="anchor" href="#add-csv-input-files-to-the-data-directory"></a>
</h3>
<p>Letâ€™s now create our <strong>metadata.csv</strong> input file</p>
<p>The only required columns are the first with sample names, and the
second with the primer name/barcode used. The subsequent columns (â€˜Dayâ€™
and â€˜Whenâ€™) are necessary for downstream steps with the phyloseq
package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"F3D0_S188_L001"</span>, <span class="st">"F3D1_S189_L001"</span>, <span class="st">"F3D141_S207_L001"</span>, <span class="st">"F3D142_S208_L001"</span>,</span>
<span>  <span class="st">"F3D143_S209_L001"</span>, <span class="st">"F3D144_S210_L001"</span>, <span class="st">"F3D145_S211_L001"</span>, <span class="st">"F3D146_S212_L001"</span>,</span>
<span>  <span class="st">"F3D147_S213_L001"</span>, <span class="st">"F3D148_S214_L001"</span>, <span class="st">"F3D149_S215_L001"</span>, <span class="st">"F3D150_S216_L001"</span>,</span>
<span>  <span class="st">"F3D2_S190_L001"</span>, <span class="st">"F3D3_S191_L001"</span>, <span class="st">"F3D5_S193_L001"</span>, <span class="st">"F3D6_S194_L001"</span>,</span>
<span>  <span class="st">"F3D7_S195_L001"</span>, <span class="st">"F3D8_S196_L001"</span>, <span class="st">"F3D9_S197_L001"</span>, <span class="st">"Mock_S280_L001"</span><span class="op">)</span>,</span>
<span>  primer_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"r16S"</span>, <span class="fl">20</span><span class="op">)</span>,</span>
<span>  Day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">141</span>, <span class="fl">142</span>, <span class="fl">143</span>, <span class="fl">144</span>, <span class="fl">145</span>, <span class="fl">146</span>, <span class="fl">147</span>, <span class="fl">148</span>, <span class="fl">149</span>, <span class="fl">150</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">9</span>, <span class="cn">NA</span><span class="op">)</span>,</span>
<span>  When <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Early"</span>, <span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Late"</span>, <span class="fl">10</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Early"</span>, <span class="fl">7</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>While you can construct your CSV files in Excel, here we construct
and save our metadata file in our data directory directly in R.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Let's define a variable that stores our data directory path</span></span>
<span><span class="va">metadata_dir_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_folder</span>, <span class="st">"metadata.csv"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">metadata</span>, <span class="va">metadata_dir_path</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="co">#&gt;         sample_name primer_name Day  When</span></span>
<span><span class="co">#&gt; 1    F3D0_S188_L001        r16S   0 Early</span></span>
<span><span class="co">#&gt; 2    F3D1_S189_L001        r16S   1 Early</span></span>
<span><span class="co">#&gt; 3  F3D141_S207_L001        r16S 141  Late</span></span>
<span><span class="co">#&gt; 4  F3D142_S208_L001        r16S 142  Late</span></span>
<span><span class="co">#&gt; 5  F3D143_S209_L001        r16S 143  Late</span></span>
<span><span class="co">#&gt; 6  F3D144_S210_L001        r16S 144  Late</span></span>
<span><span class="co">#&gt; 7  F3D145_S211_L001        r16S 145  Late</span></span>
<span><span class="co">#&gt; 8  F3D146_S212_L001        r16S 146  Late</span></span>
<span><span class="co">#&gt; 9  F3D147_S213_L001        r16S 147  Late</span></span>
<span><span class="co">#&gt; 10 F3D148_S214_L001        r16S 148  Late</span></span>
<span><span class="co">#&gt; 11 F3D149_S215_L001        r16S 149  Late</span></span>
<span><span class="co">#&gt; 12 F3D150_S216_L001        r16S 150  Late</span></span>
<span><span class="co">#&gt; 13   F3D2_S190_L001        r16S   2 Early</span></span>
<span><span class="co">#&gt; 14   F3D3_S191_L001        r16S   3 Early</span></span>
<span><span class="co">#&gt; 15   F3D5_S193_L001        r16S   5 Early</span></span>
<span><span class="co">#&gt; 16   F3D6_S194_L001        r16S   6 Early</span></span>
<span><span class="co">#&gt; 17   F3D7_S195_L001        r16S   7 Early</span></span>
<span><span class="co">#&gt; 18   F3D8_S196_L001        r16S   8 Early</span></span>
<span><span class="co">#&gt; 19   F3D9_S197_L001        r16S   9 Early</span></span>
<span><span class="co">#&gt; 20   Mock_S280_L001        r16S  NA  &lt;NA&gt;</span></span></code></pre></div>
<p><strong>metadata.csv</strong></p>
<pre><code><span><span class="co">#&gt;         sample_name primer_name Day  When</span></span>
<span><span class="co">#&gt; 1    F3D0_S188_L001        r16S   0 Early</span></span>
<span><span class="co">#&gt; 2    F3D1_S189_L001        r16S   1 Early</span></span>
<span><span class="co">#&gt; 3  F3D141_S207_L001        r16S 141  Late</span></span>
<span><span class="co">#&gt; 4  F3D142_S208_L001        r16S 142  Late</span></span>
<span><span class="co">#&gt; 5  F3D143_S209_L001        r16S 143  Late</span></span>
<span><span class="co">#&gt; 6  F3D144_S210_L001        r16S 144  Late</span></span>
<span><span class="co">#&gt; 7  F3D145_S211_L001        r16S 145  Late</span></span>
<span><span class="co">#&gt; 8  F3D146_S212_L001        r16S 146  Late</span></span>
<span><span class="co">#&gt; 9  F3D147_S213_L001        r16S 147  Late</span></span>
<span><span class="co">#&gt; 10 F3D148_S214_L001        r16S 148  Late</span></span>
<span><span class="co">#&gt; 11 F3D149_S215_L001        r16S 149  Late</span></span>
<span><span class="co">#&gt; 12 F3D150_S216_L001        r16S 150  Late</span></span>
<span><span class="co">#&gt; 13   F3D2_S190_L001        r16S   2 Early</span></span>
<span><span class="co">#&gt; 14   F3D3_S191_L001        r16S   3 Early</span></span>
<span><span class="co">#&gt; 15   F3D5_S193_L001        r16S   5 Early</span></span>
<span><span class="co">#&gt; 16   F3D6_S194_L001        r16S   6 Early</span></span>
<span><span class="co">#&gt; 17   F3D7_S195_L001        r16S   7 Early</span></span>
<span><span class="co">#&gt; 18   F3D8_S196_L001        r16S   8 Early</span></span>
<span><span class="co">#&gt; 19   F3D9_S197_L001        r16S   9 Early</span></span>
<span><span class="co">#&gt; 20   Mock_S280_L001        r16S  NA  &lt;NA&gt;</span></span></code></pre>
<p>Now, make our <strong>primerinfo_params.csv</strong> input file</p>
<p>We then make and save the second CSV input file with the name of the
metabarcode used, primer sequences, and the optional
<strong><code>dada2</code></strong> parameter options.</p>
<p>We referenced the DADA2 <a href="https://benjjneb.github.io/dada2/tutorial.html" class="external-link">tutorial</a> to
select the same parameter options.</p>
<p><em>Note, primers were already trimmed from reads during the
demultiplexing step that occurs after sequencing, but just to be
certain, we included the Earth Microbiome primers sequences described <a href="https://earthmicrobiome.org/protocols-and-standards/16s/" class="external-link">here</a>,
which will be searched for across reads.</em></p>
<p><strong>primerinfo_params.csv</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create the data frame</span></span>
<span><span class="va">primer_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  primer_name <span class="op">=</span> <span class="st">"r16S"</span>,</span>
<span>  forward <span class="op">=</span> <span class="st">"GTGYCAGCMGCCGCGGTAA"</span>,</span>
<span>  reverse <span class="op">=</span> <span class="st">"GGACTACNVGGGTWTCTAAT"</span>,</span>
<span>  already_trimmed <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  minCutadaptlength <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  multithread <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  maxN <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  maxEE_forward <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  maxEE_reverse <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  truncLen_forward <span class="op">=</span> <span class="fl">240</span>,</span>
<span>  truncLen_reverse <span class="op">=</span> <span class="fl">160</span>,</span>
<span>  truncQ <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  minLen <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  maxLen <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  minQ <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  trimLeft <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  trimRight <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  rm.lowcomplex <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  minOverlap <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  maxMismatch <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  min_asv_length <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We construct and save our <strong>primerinfo_params</strong> file in
our data directory</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">primer_params_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_folder</span>, <span class="st">"primerinfo_params.csv"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">primer_info</span>, <span class="va">primer_params_path</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   primer_name             forward              reverse already_trimmed</span></span>
<span><span class="co">#&gt; 1        r16S GTGYCAGCMGCCGCGGTAA GGACTACNVGGGTWTCTAAT            TRUE</span></span>
<span><span class="co">#&gt;   minCutadaptlength multithread verbose maxN maxEE_forward maxEE_reverse</span></span>
<span><span class="co">#&gt; 1                 0        TRUE    TRUE    0             2             2</span></span>
<span><span class="co">#&gt;   truncLen_forward truncLen_reverse truncQ minLen maxLen minQ trimLeft</span></span>
<span><span class="co">#&gt; 1              240              160      2     20    Inf    0        0</span></span>
<span><span class="co">#&gt;   trimRight rm.lowcomplex minOverlap maxMismatch min_asv_length seed</span></span>
<span><span class="co">#&gt; 1         0             0         12           0             50    1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="download-silva-reference-database">Download SILVA reference database<a class="anchor" aria-label="anchor" href="#download-silva-reference-database"></a>
</h3>
<p>We now retrieve the SILVA database (v.138.2) (to the taxonomic level
of species).</p>
<p>More information is <a href="https://zenodo.org/records/14169026" class="external-link">here</a></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">silva_path</span> <span class="op">&lt;-</span> <span class="st">"https://zenodo.org/records/14169026/files/silva_nr99_v138.2_toSpecies_trainset.fa.gz?download=1"</span></span>
<span><span class="va">silva_download_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_folder</span>, <span class="st">"silva_nr99_v138.2_toSpecies_trainset.fa.gz"</span><span class="op">)</span></span>
<span></span>
<span> <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">silva_download_file_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>   <span class="va">bin2</span> <span class="op">&lt;-</span> <span class="fu">RCurl</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RCurl/man/getBinaryURL.html" class="external-link">getBinaryURL</a></span><span class="op">(</span><span class="va">silva_path</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/readBin.html" class="external-link">writeBin</a></span><span class="op">(</span><span class="va">bin2</span>, <span class="va">silva_download_file_path</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"File downloaded:"</span>, <span class="va">silva_download_file_path</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span> <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"File already exists:"</span>, <span class="va">silva_download_file_path</span>, <span class="st">"! Skipping download.\n"</span><span class="op">)</span></span>
<span> <span class="op">}</span></span>
<span><span class="co">#&gt; File downloaded: /tmp/Rtmp8rQQx4/demulticoder_sop_run/MiSeq_SOP/silva_nr99_v138.2_toSpecies_trainset.fa.gz</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-reads-for-dada2-analyses">Prepare reads for DADA2 analyses<a class="anchor" aria-label="anchor" href="#prepare-reads-for-dada2-analyses"></a>
</h3>
<p>Remove Nâ€™s and create directory structure for downstream steps<br>
Note-intermediate files are saved in a temporary folder automatically.
If you prefer to define a temp directory path, refer to documentation
for the <strong>prepare_reads</strong> function</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The file names will first need to be renamed to go through the demulticoder workflow, since it is looking for files that have suffixes like R1.fastq.gz or R2.fastq.gz</span></span>
<span><span class="va">output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="va">data_folder</span>, <span class="co">#Use data_directory path defined above </span></span>
<span>  output_directory <span class="op">=</span> <span class="va">output_folder</span>, <span class="co"># Uses output_directory path defined above</span></span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (17): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (17): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">20</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (3): sample_name, primer_name, When</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (1): Day</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; Creating output directory: /tmp/Rtmp8rQQx4/demulticoder_run/prefiltered_sequences</span></span></code></pre></div>
<p><img src="prepare_reads-1.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="remove-primers-and-trim-reads">Remove primers and trim reads<a class="anchor" aria-label="anchor" href="#remove-primers-and-trim-reads"></a>
</h3>
<p>Run Cutadapt to remove primers and then trim reads with
<strong><code>dada2</code></strong> filterAndTrim function</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running cutadapt 3.5 for r16S sequence data</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D0_S188_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D0_S188_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D1_S189_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D1_S189_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D141_S207_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D141_S207_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D142_S208_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D142_S208_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D143_S209_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D143_S209_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D144_S210_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D144_S210_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D145_S211_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D145_S211_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D146_S212_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D146_S212_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D147_S213_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D147_S213_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D148_S214_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D148_S214_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D149_S215_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D149_S215_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D150_S216_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D150_S216_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D2_S190_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D2_S190_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D3_S191_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D3_S191_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D5_S193_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D5_S193_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D6_S194_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D6_S194_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D7_S195_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D7_S195_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D8_S196_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D8_S196_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D9_S197_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/F3D9_S197_L001_R2_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed forward reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/Mock_S280_L001_R1_r16S.fastq.gz</span></span>
<span><span class="co">#&gt; Already trimmed reverse reads were appended to trimmed read directory, and they are located here:/tmp/Rtmp8rQQx4/demulticoder_run/trimmed_sequences/Mock_S280_L001_R2_r16S.fastq.gz</span></span></code></pre></div>
<p><img src="Remove_primers_and_trim_reads-1.png" width="70%"></p>
<p>We can now visualize the outputs from the primer removal and trimming
steps. A CSV files is output showing which samples still have primer
sequences and the barplot above summarizes the outputs.</p>
<p>There are circumstances where a few primer sequences may still
remain. If so, any ASVs with any residual primer sequences will be
filtered at the end.</p>
</div>
<div class="section level3">
<h3 id="core-asv-inference-step">Core ASV inference step<a class="anchor" aria-label="anchor" href="#core-asv-inference-step"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; 33513360 total bases in 139639 reads from 20 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 ....................</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt;    selfConsist step 5</span></span>
<span><span class="co">#&gt; Convergence after  5  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the  Forward  read of primer pair  r16S</span></span>
<span><span class="co">#&gt; Sample 1 - 7112 reads in 1978 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 5299 reads in 1639 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 5463 reads in 1477 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 2914 reads in 904 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 2941 reads in 939 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 4312 reads in 1267 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 6741 reads in 1756 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 4560 reads in 1438 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 15636 reads in 3589 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 11412 reads in 2761 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 12017 reads in 3021 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 5032 reads in 1566 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 18075 reads in 3707 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 6250 reads in 1479 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 4052 reads in 1195 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 7369 reads in 1832 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 4765 reads in 1183 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 4871 reads in 1382 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 6504 reads in 1709 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 4314 reads in 897 unique sequences.</span></span>
<span><span class="co">#&gt; 22342240 total bases in 139639 reads from 20 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">#&gt; selfConsist step 1 ....................</span></span>
<span><span class="co">#&gt;    selfConsist step 2</span></span>
<span><span class="co">#&gt;    selfConsist step 3</span></span>
<span><span class="co">#&gt;    selfConsist step 4</span></span>
<span><span class="co">#&gt;    selfConsist step 5</span></span>
<span><span class="co">#&gt;    selfConsist step 6</span></span>
<span><span class="co">#&gt;    selfConsist step 7</span></span>
<span><span class="co">#&gt; Convergence after  7  rounds.</span></span>
<span><span class="co">#&gt; Error rate plot for the  Reverse  read of primer pair  r16S</span></span>
<span><span class="co">#&gt; Sample 1 - 7112 reads in 1659 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 5299 reads in 1349 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 3 - 5463 reads in 1335 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 4 - 2914 reads in 853 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 5 - 2941 reads in 880 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 6 - 4312 reads in 1286 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 7 - 6741 reads in 1803 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 8 - 4560 reads in 1265 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 9 - 15636 reads in 3413 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 10 - 11412 reads in 2522 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 11 - 12017 reads in 2771 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 12 - 5032 reads in 1415 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 13 - 18075 reads in 3290 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 14 - 6250 reads in 1390 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 15 - 4052 reads in 1134 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 16 - 7369 reads in 1635 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 17 - 4765 reads in 1084 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 18 - 4871 reads in 1161 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 19 - 6504 reads in 1502 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 20 - 4314 reads in 732 unique sequences.</span></span>
<span><span class="co">#&gt; 6540 paired-reads (in 107 unique pairings) successfully merged out of 6891 (in 197 pairings) input.</span></span>
<span><span class="co">#&gt; 5027 paired-reads (in 101 unique pairings) successfully merged out of 5189 (in 157 pairings) input.</span></span>
<span><span class="co">#&gt; 4986 paired-reads (in 81 unique pairings) successfully merged out of 5267 (in 166 pairings) input.</span></span>
<span><span class="co">#&gt; 2595 paired-reads (in 52 unique pairings) successfully merged out of 2754 (in 108 pairings) input.</span></span>
<span><span class="co">#&gt; 2553 paired-reads (in 60 unique pairings) successfully merged out of 2785 (in 119 pairings) input.</span></span>
<span><span class="co">#&gt; 3646 paired-reads (in 55 unique pairings) successfully merged out of 4109 (in 157 pairings) input.</span></span>
<span><span class="co">#&gt; 6079 paired-reads (in 81 unique pairings) successfully merged out of 6514 (in 198 pairings) input.</span></span>
<span><span class="co">#&gt; 3968 paired-reads (in 91 unique pairings) successfully merged out of 4388 (in 187 pairings) input.</span></span>
<span><span class="co">#&gt; 14233 paired-reads (in 143 unique pairings) successfully merged out of 15355 (in 352 pairings) input.</span></span>
<span><span class="co">#&gt; 10529 paired-reads (in 120 unique pairings) successfully merged out of 11165 (in 277 pairings) input.</span></span>
<span><span class="co">#&gt; 11154 paired-reads (in 137 unique pairings) successfully merged out of 11797 (in 298 pairings) input.</span></span>
<span><span class="co">#&gt; 4349 paired-reads (in 85 unique pairings) successfully merged out of 4802 (in 179 pairings) input.</span></span>
<span><span class="co">#&gt; 17431 paired-reads (in 153 unique pairings) successfully merged out of 17812 (in 272 pairings) input.</span></span>
<span><span class="co">#&gt; 5850 paired-reads (in 81 unique pairings) successfully merged out of 6095 (in 159 pairings) input.</span></span>
<span><span class="co">#&gt; 3713 paired-reads (in 86 unique pairings) successfully merged out of 3891 (in 147 pairings) input.</span></span>
<span><span class="co">#&gt; 6865 paired-reads (in 99 unique pairings) successfully merged out of 7191 (in 187 pairings) input.</span></span>
<span><span class="co">#&gt; 4428 paired-reads (in 68 unique pairings) successfully merged out of 4605 (in 128 pairings) input.</span></span>
<span><span class="co">#&gt; 4576 paired-reads (in 101 unique pairings) successfully merged out of 4739 (in 174 pairings) input.</span></span>
<span><span class="co">#&gt; 6092 paired-reads (in 109 unique pairings) successfully merged out of 6315 (in 173 pairings) input.</span></span>
<span><span class="co">#&gt; 4269 paired-reads (in 20 unique pairings) successfully merged out of 4281 (in 28 pairings) input.</span></span></code></pre></div>
<p><img src="ASV_inference-1.png" width="70%"></p>
<pre><code><span><span class="co">#&gt; Identified 61 bimeras out of 293 input sequences.</span></span></code></pre>
<p><img src="ASV_inference-2.png" width="70%"></p>
<pre><code><span><span class="co">#&gt; $r16S</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmp8rQQx4/demulticoder_run/asvabund_matrixDADA2_r16S.RData"</span></span></code></pre>
<p>We can now visualize the outputs from the ASV inference step.</p>
<p>The first plot shows the how reads were merged in terms of mismatches
and indels.</p>
<p>The second plot shows the overlap lengths across the inferred
ASVs.</p>
<p>We can also look at the distribution of ASV lengths</p>
</div>
<div class="section level3">
<h3 id="assign-taxonomy-step">Assign taxonomy step<a class="anchor" aria-label="anchor" href="#assign-taxonomy-step"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">output</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  db_16S<span class="op">=</span><span class="st">"silva_nr99_v138.2_toSpecies_trainset.fa.gz"</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tracking read counts:</span></span>
<span><span class="co">#&gt;       samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1    F3D0_S188_L001_r16S  7733     7112      6976      6979   6540    6528</span></span>
<span><span class="co">#&gt; 2    F3D1_S189_L001_r16S  5829     5299      5226      5239   5027    5016</span></span>
<span><span class="co">#&gt; 3  F3D141_S207_L001_r16S  5926     5463      5331      5357   4986    4863</span></span>
<span><span class="co">#&gt; 4  F3D142_S208_L001_r16S  3158     2914      2799      2830   2595    2521</span></span>
<span><span class="co">#&gt; 5  F3D143_S209_L001_r16S  3164     2941      2822      2868   2553    2519</span></span>
<span><span class="co">#&gt; 6  F3D144_S210_L001_r16S  4798     4312      4150      4228   3646    3507</span></span>
<span><span class="co">#&gt; 7  F3D145_S211_L001_r16S  7331     6741      6592      6627   6079    5820</span></span>
<span><span class="co">#&gt; 8  F3D146_S212_L001_r16S  4993     4560      4450      4470   3968    3879</span></span>
<span><span class="co">#&gt; 9  F3D147_S213_L001_r16S 16956    15636     15433     15505  14233   13006</span></span>
<span><span class="co">#&gt; 10 F3D148_S214_L001_r16S 12332    11412     11250     11267  10529    9935</span></span>
<span><span class="co">#&gt; 11 F3D149_S215_L001_r16S 13006    12017     11857     11898  11154   10653</span></span>
<span><span class="co">#&gt; 12 F3D150_S216_L001_r16S  5474     5032      4879      4925   4349    4240</span></span>
<span><span class="co">#&gt; 13   F3D2_S190_L001_r16S 19489    18075     17907     17939  17431   16835</span></span>
<span><span class="co">#&gt; 14   F3D3_S191_L001_r16S  6726     6250      6145      6176   5850    5486</span></span>
<span><span class="co">#&gt; 15   F3D5_S193_L001_r16S  4418     4052      3930      3991   3713    3713</span></span>
<span><span class="co">#&gt; 16   F3D6_S194_L001_r16S  7933     7369      7231      7294   6865    6678</span></span>
<span><span class="co">#&gt; 17   F3D7_S195_L001_r16S  5103     4765      4646      4673   4428    4217</span></span>
<span><span class="co">#&gt; 18   F3D8_S196_L001_r16S  5274     4871      4786      4802   4576    4547</span></span>
<span><span class="co">#&gt; 19   F3D9_S197_L001_r16S  7023     6504      6341      6442   6092    6015</span></span>
<span><span class="co">#&gt; 20   Mock_S280_L001_r16S  4748     4314      4287      4286   4269    4269</span></span></code></pre></div>
<p>As a check we can take a look at read counts across the workflow. If
there are sudden drops, we should reconsider our adjusting certain
<strong><code>dada2</code></strong> parameters and re-running the
analysis.</p>
</div>
<div class="section level3">
<h3 id="convert-asv-matrix-to-taxmap-and-phyloseq-objects">Convert ASV matrix to <strong><code>taxmap</code></strong> and
<strong><code>phyloseq</code></strong> objects<a class="anchor" aria-label="anchor" href="#convert-asv-matrix-to-taxmap-and-phyloseq-objects"></a>
</h3>
<p>Convert ASV matrix to <strong><code>taxmap</code></strong> and
<strong><code>phyloseq</code></strong> objects with one function</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">232</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (20): F3D0_S188_L001_r16S, F3D1_S189_L001_r16S, F3D141_S207_L001_r16S, F...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For r16S dataset </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Taxmap object saved in: /tmp/Rtmp8rQQx4/demulticoder_sop_run/sop_16S_demulticoder_outputs/taxmap_obj_r16S.RData </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: /tmp/Rtmp8rQQx4/demulticoder_sop_run/sop_16S_demulticoder_outputs/phylo_obj_r16S.RData </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="examine-accuracy-relative-to-a-mock-community">Examine accuracy relative to a mock community<a class="anchor" aria-label="anchor" href="#examine-accuracy-relative-to-a-mock-community"></a>
</h3>
<p>Evaluate accuracy using mock community, as shown in
<strong><code>dada2</code></strong> tutorial</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">matrix_filepath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">output</span><span class="op">$</span><span class="va">directory_paths</span><span class="op">$</span><span class="va">output_directory</span>, <span class="st">"final_asv_abundance_matrix_r16S.csv"</span><span class="op">)</span></span>
<span><span class="va">tax_matrix</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">matrix_filepath</span><span class="op">)</span></span>
<span></span>
<span><span class="va">unqs.mock</span> <span class="op">&lt;-</span> <span class="va">tax_matrix</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tax_matrix</span><span class="op">)</span> <span class="op">==</span> <span class="st">"Mock_S280_L001_r16S"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">unqs.mock</span> <span class="op">&lt;-</span> <span class="va">unqs.mock</span><span class="op">[</span><span class="va">unqs.mock</span><span class="op">$</span><span class="va">Mock_S280_L001_r16S</span> <span class="op">!=</span> <span class="fl">0</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"DADA2 inferred"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">unqs.mock</span><span class="op">)</span>, <span class="st">"sample sequences present in the Mock community.\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; DADA2 inferred 20 sample sequences present in the Mock community.</span></span></code></pre></div>
<p>When looking at mock community sample, we were able to extract 20
bacterial sequences that also matched with what was present in the mock
community.</p>
</div>
<div class="section level3">
<h3 id="alpha-diversity-analysis">Alpha diversity analysis<a class="anchor" aria-label="anchor" href="#alpha-diversity-analysis"></a>
</h3>
<p>Follow-up work using phyloseq to do side-by-side comparison with
DADA2 workflow example and to examine alpha diversity results</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html" class="external-link">prune_samples</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html" class="external-link">sample_names</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"Mock_S280_L001_r16S"</span>, <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span> <span class="co"># Remove mock sample</span></span>
<span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_richness.html" class="external-link">plot_richness</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, x<span class="op">=</span><span class="st">"Day"</span>, measures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Shannon"</span>, <span class="st">"Simpson"</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"When"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_alpha_diversity_analysis-1.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="beta-diversity-analysis">Beta diversity analysis<a class="anchor" aria-label="anchor" href="#beta-diversity-analysis"></a>
</h3>
<p>Examine ordination plots as additional point of comparison with the
DADA2 tutorial</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform data to proportions as appropriate for Bray-Curtis distances</span></span>
<span><span class="va">ps.prop</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, <span class="kw">function</span><span class="op">(</span><span class="va">otu</span><span class="op">)</span> <span class="va">otu</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">otu</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ord.nmds.bray</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html" class="external-link">ordinate</a></span><span class="op">(</span><span class="va">ps.prop</span>, method<span class="op">=</span><span class="st">"NMDS"</span>, distance<span class="op">=</span><span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Run 0 stress 0.0808378 </span></span>
<span><span class="co">#&gt; Run 1 stress 0.08096389 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.009177604  max resid 0.02810685 </span></span>
<span><span class="co">#&gt; Run 2 stress 0.1231378 </span></span>
<span><span class="co">#&gt; Run 3 stress 0.121153 </span></span>
<span><span class="co">#&gt; Run 4 stress 0.08635462 </span></span>
<span><span class="co">#&gt; Run 5 stress 0.0808378 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 5.812944e-06  max resid 1.554258e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 6 stress 0.1231378 </span></span>
<span><span class="co">#&gt; Run 7 stress 0.09061366 </span></span>
<span><span class="co">#&gt; Run 8 stress 0.08096389 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.009176341  max resid 0.02810282 </span></span>
<span><span class="co">#&gt; Run 9 stress 0.1433231 </span></span>
<span><span class="co">#&gt; Run 10 stress 0.0808378 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 5.646214e-06  max resid 1.499963e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 11 stress 0.1231378 </span></span>
<span><span class="co">#&gt; Run 12 stress 0.0809639 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.009215858  max resid 0.02823353 </span></span>
<span><span class="co">#&gt; Run 13 stress 0.121153 </span></span>
<span><span class="co">#&gt; Run 14 stress 0.1320722 </span></span>
<span><span class="co">#&gt; Run 15 stress 0.08635462 </span></span>
<span><span class="co">#&gt; Run 16 stress 0.121153 </span></span>
<span><span class="co">#&gt; Run 17 stress 0.08096389 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.009186834  max resid 0.02813753 </span></span>
<span><span class="co">#&gt; Run 18 stress 0.08635462 </span></span>
<span><span class="co">#&gt; Run 19 stress 0.08635462 </span></span>
<span><span class="co">#&gt; Run 20 stress 0.09466617 </span></span>
<span><span class="co">#&gt; *** Best solution repeated 2 times</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html" class="external-link">plot_ordination</a></span><span class="op">(</span><span class="va">ps.prop</span>, <span class="va">ord.nmds.bray</span>, color<span class="op">=</span><span class="st">"When"</span>, title<span class="op">=</span><span class="st">"Bray NMDS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_beta_diversity_analysis-1.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="top-taxa-analysis">Top taxa analysis<a class="anchor" aria-label="anchor" href="#top-taxa-analysis"></a>
</h3>
<p>Letâ€™s look at what the top 20 taxa are in the early vs.Â late samples
time points, as shown in the DADA2 tutorial</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top20</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html" class="external-link">taxa_sums</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span><span class="va">ps.top20</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, <span class="kw">function</span><span class="op">(</span><span class="va">OTU</span><span class="op">)</span> <span class="va">OTU</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">OTU</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ps.top20</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html" class="external-link">prune_taxa</a></span><span class="op">(</span><span class="va">top20</span>, <span class="va">ps.top20</span><span class="op">)</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_bar.html" class="external-link">plot_bar</a></span><span class="op">(</span><span class="va">ps.top20</span>, x<span class="op">=</span><span class="st">"Day"</span>, fill<span class="op">=</span><span class="st">"Family"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">When</span>, scales<span class="op">=</span><span class="st">"free_x"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_community_analysis-1.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p><strong>Information on the 16S SOP Mothur dataset can be found
here:</strong> <a href="https://mothur.org/wiki/miseq_sop/" class="external-link uri">https://mothur.org/wiki/miseq_sop/</a></p>
<p>Kozich, J. J., Westcott, S. L., Baxter, N. T., Highlander, S. K., and
Schloss, P. D. 2013. Development of a dual-index sequencing strategy and
curation pipeline for analyzing amplicon sequence data on the MiSeq
Illumina sequencing platform. Applied and Environmental Microbiology
79:5112â€“5120. <a href="https://doi.org/10.1128/AEM.01043-13" class="external-link uri">https://doi.org/10.1128/AEM.01043-13</a>.</p>
<p><strong>Information on the DADA2 16S tutorial and associated
manuscript can be found here:</strong> <a href="https://benjjneb.github.io/dada2/tutorial.html" class="external-link uri">https://benjjneb.github.io/dada2/tutorial.html</a></p>
<p>Callahan, B. J., McMurdie, P. J., Rosen, M. J., Han, A. W., Johnson,
A. J. A., and Holmes, S. P. 2016. DADA2: high-resolution sample
inference from Illumina amplicon data. Nature Methods 13:581. <a href="https://doi.org/10.1038/nmeth.3869" class="external-link uri">https://doi.org/10.1038/nmeth.3869</a>.</p>
</div>
<div class="section level3">
<h3 id="software-and-packages">Software and packages<a class="anchor" aria-label="anchor" href="#software-and-packages"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.1.2 (2021-11-01)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Pop!_OS 22.04 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] purrr_1.0.4        future_1.40.0      RCurl_1.98-1.17    dplyr_1.1.4       </span></span>
<span><span class="co">#&gt; [5] phyloseq_1.38.0    metacoder_0.3.8    demulticoder_0.1.2 kableExtra_1.4.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] systemfonts_1.0.4           plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;   [3] igraph_2.1.4                lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;   [5] splines_4.1.2               listenv_0.9.1              </span></span>
<span><span class="co">#&gt;   [7] BiocParallel_1.28.3         usethis_3.1.0              </span></span>
<span><span class="co">#&gt;   [9] GenomeInfoDb_1.30.1         ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt;  [11] digest_0.6.37               foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.8.1           magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [15] memoise_2.0.1               cluster_2.1.2              </span></span>
<span><span class="co">#&gt;  [17] tzdb_0.5.0                  remotes_2.5.0              </span></span>
<span><span class="co">#&gt;  [19] globals_0.17.0              Biostrings_2.62.0          </span></span>
<span><span class="co">#&gt;  [21] readr_2.1.5                 RcppParallel_5.1.10        </span></span>
<span><span class="co">#&gt;  [23] matrixStats_1.5.0           vroom_1.6.5                </span></span>
<span><span class="co">#&gt;  [25] svglite_2.1.3               pkgdown_2.1.1              </span></span>
<span><span class="co">#&gt;  [27] jpeg_0.1-10                 colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [29] textshaping_0.3.6           xfun_0.51                  </span></span>
<span><span class="co">#&gt;  [31] callr_3.7.6                 crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.9.1              dada2_1.30.0               </span></span>
<span><span class="co">#&gt;  [35] survival_3.2-13             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [37] ape_5.8-1                   glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [39] gtable_0.3.6                zlibbioc_1.40.0            </span></span>
<span><span class="co">#&gt;  [41] XVector_0.34.0              DelayedArray_0.20.0        </span></span>
<span><span class="co">#&gt;  [43] pkgbuild_1.4.6              Rhdf5lib_1.16.0            </span></span>
<span><span class="co">#&gt;  [45] BiocGenerics_0.40.0         scales_1.4.0               </span></span>
<span><span class="co">#&gt;  [47] DBI_1.2.3                   miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [49] Rcpp_1.0.14                 viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt;  [51] xtable_1.8-4                bit_4.6.0                  </span></span>
<span><span class="co">#&gt;  [53] stats4_4.1.2                profvis_0.4.0              </span></span>
<span><span class="co">#&gt;  [55] htmlwidgets_1.6.4           RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [57] ellipsis_0.3.2              urlchecker_1.0.1           </span></span>
<span><span class="co">#&gt;  [59] pkgconfig_2.0.3             farver_2.1.2               </span></span>
<span><span class="co">#&gt;  [61] sass_0.4.9                  deldir_2.0-4               </span></span>
<span><span class="co">#&gt;  [63] labeling_0.4.3              tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [65] rlang_1.1.6                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [67] later_1.4.1                 tools_4.1.2                </span></span>
<span><span class="co">#&gt;  [69] cachem_1.1.0                cli_3.6.5                  </span></span>
<span><span class="co">#&gt;  [71] generics_0.1.3              ade4_1.7-23                </span></span>
<span><span class="co">#&gt;  [73] devtools_2.4.5              evaluate_1.0.3             </span></span>
<span><span class="co">#&gt;  [75] biomformat_1.22.0           stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [77] fastmap_1.2.0               yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [79] ragg_1.2.1                  processx_3.8.6             </span></span>
<span><span class="co">#&gt;  [81] knitr_1.50                  bit64_4.6.0-1              </span></span>
<span><span class="co">#&gt;  [83] fs_1.6.5                    nlme_3.1-155               </span></span>
<span><span class="co">#&gt;  [85] mime_0.12                   xml2_1.3.7                 </span></span>
<span><span class="co">#&gt;  [87] compiler_4.1.2              rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [89] curl_6.2.1                  png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [91] tibble_3.2.1                bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [93] stringi_1.8.7               ps_1.9.0                   </span></span>
<span><span class="co">#&gt;  [95] desc_1.4.3                  lattice_0.20-45            </span></span>
<span><span class="co">#&gt;  [97] Matrix_1.4-0                vegan_2.6-10               </span></span>
<span><span class="co">#&gt;  [99] permute_0.9-7               multtest_2.50.0            </span></span>
<span><span class="co">#&gt; [101] vctrs_0.6.5                 furrr_0.3.1                </span></span>
<span><span class="co">#&gt; [103] pillar_1.10.2               lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [105] rhdf5filters_1.6.0          jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [107] data.table_1.17.0           bitops_1.0-7               </span></span>
<span><span class="co">#&gt; [109] httpuv_1.6.15               GenomicRanges_1.46.1       </span></span>
<span><span class="co">#&gt; [111] R6_2.6.1                    latticeExtra_0.6-30        </span></span>
<span><span class="co">#&gt; [113] hwriter_1.3.2.1             promises_1.3.2             </span></span>
<span><span class="co">#&gt; [115] ShortRead_1.60.0            parallelly_1.43.0          </span></span>
<span><span class="co">#&gt; [117] IRanges_2.28.0              sessioninfo_1.2.3          </span></span>
<span><span class="co">#&gt; [119] codetools_0.2-18            MASS_7.3-55                </span></span>
<span><span class="co">#&gt; [121] pkgload_1.4.0               rhdf5_2.38.1               </span></span>
<span><span class="co">#&gt; [123] SummarizedExperiment_1.24.0 withr_3.0.2                </span></span>
<span><span class="co">#&gt; [125] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span></span>
<span><span class="co">#&gt; [127] S4Vectors_0.32.4            GenomeInfoDbData_1.2.7     </span></span>
<span><span class="co">#&gt; [129] mgcv_1.8-39                 parallel_4.1.2             </span></span>
<span><span class="co">#&gt; [131] hms_1.1.3                   grid_4.1.2                 </span></span>
<span><span class="co">#&gt; [133] tidyr_1.3.1                 rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [135] MatrixGenerics_1.6.0        Biobase_2.54.0             </span></span>
<span><span class="co">#&gt; [137] shiny_1.10.0                interp_1.1-6</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus GrÃ¼nwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
