<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>16S Mothur SOP Validation • demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="16S Mothur SOP Validation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes">
<li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="DADA2_16S_mothur_validation_files/kePrint-0.0.1/kePrint.js"></script><link href="DADA2_16S_mothur_validation_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>16S Mothur SOP Validation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/DADA2_16S_mothur_validation.Rmd" class="external-link"><code>vignettes/DADA2_16S_mothur_validation.Rmd</code></a></small>
      <div class="d-none name"><code>DADA2_16S_mothur_validation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="this-vignette-shows-how-demulticoder-was-used-to-analyze-the-mothur-16s-sop-dataset-featured-in-dada2-tutorials">This vignette shows how demulticoder was used to analyze the mothur
16S SOP dataset featured in DADA2 tutorials<a class="anchor" aria-label="anchor" href="#this-vignette-shows-how-demulticoder-was-used-to-analyze-the-mothur-16s-sop-dataset-featured-in-dada2-tutorials"></a>
</h2>
<p>First, make sure input metadata and primerinfo_params files are in
data folder</p>
<p>The only required columns are the first with sample names, and the
second with the primer name/barcode used. The subsequent columns are
user-specific columns for downstream steps <em>metadata.csv
file</em></p>
<table class="table">
<thead><tr class="header">
<th align="left">sample_name</th>
<th align="left">primer_name</th>
<th align="right">Day</th>
<th align="left">When</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">F3D0</td>
<td align="left">r16S</td>
<td align="right">0</td>
<td align="left">Early</td>
</tr>
<tr class="even">
<td align="left">F3D1</td>
<td align="left">r16S</td>
<td align="right">1</td>
<td align="left">Early</td>
</tr>
<tr class="odd">
<td align="left">F3D141</td>
<td align="left">r16S</td>
<td align="right">141</td>
<td align="left">Late</td>
</tr>
<tr class="even">
<td align="left">F3D142</td>
<td align="left">r16S</td>
<td align="right">142</td>
<td align="left">Late</td>
</tr>
<tr class="odd">
<td align="left">F3D143</td>
<td align="left">r16S</td>
<td align="right">143</td>
<td align="left">Late</td>
</tr>
<tr class="even">
<td align="left">F3D144</td>
<td align="left">r16S</td>
<td align="right">144</td>
<td align="left">Late</td>
</tr>
<tr class="odd">
<td align="left">F3D145</td>
<td align="left">r16S</td>
<td align="right">145</td>
<td align="left">Late</td>
</tr>
<tr class="even">
<td align="left">F3D146</td>
<td align="left">r16S</td>
<td align="right">146</td>
<td align="left">Late</td>
</tr>
<tr class="odd">
<td align="left">F3D147</td>
<td align="left">r16S</td>
<td align="right">147</td>
<td align="left">Late</td>
</tr>
<tr class="even">
<td align="left">F3D148</td>
<td align="left">r16S</td>
<td align="right">148</td>
<td align="left">Late</td>
</tr>
<tr class="odd">
<td align="left">F3D149</td>
<td align="left">r16S</td>
<td align="right">149</td>
<td align="left">Late</td>
</tr>
<tr class="even">
<td align="left">F3D150</td>
<td align="left">r16S</td>
<td align="right">150</td>
<td align="left">Late</td>
</tr>
<tr class="odd">
<td align="left">F3D2</td>
<td align="left">r16S</td>
<td align="right">2</td>
<td align="left">Early</td>
</tr>
<tr class="even">
<td align="left">F3D3</td>
<td align="left">r16S</td>
<td align="right">3</td>
<td align="left">Early</td>
</tr>
<tr class="odd">
<td align="left">F3D5</td>
<td align="left">r16S</td>
<td align="right">5</td>
<td align="left">Early</td>
</tr>
<tr class="even">
<td align="left">F3D6</td>
<td align="left">r16S</td>
<td align="right">6</td>
<td align="left">Early</td>
</tr>
<tr class="odd">
<td align="left">F3D7</td>
<td align="left">r16S</td>
<td align="right">7</td>
<td align="left">Early</td>
</tr>
<tr class="even">
<td align="left">F3D8</td>
<td align="left">r16S</td>
<td align="right">8</td>
<td align="left">Early</td>
</tr>
<tr class="odd">
<td align="left">F3D9</td>
<td align="left">r16S</td>
<td align="right">9</td>
<td align="left">Early</td>
</tr>
<tr class="even">
<td align="left">Mock</td>
<td align="left">r16S</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<p>I then included the necessary second file with the name of the
barcode selected, primer sequences, and the optional DADA2 parameter
options. I referenced the DADA2 <a href="https://benjjneb.github.io/dada2/tutorial.html" class="external-link">tutorial</a> to
select the proper parameter options.</p>
Note, primers were already trimmed from reads, but just to be certain, I
included the Earth Microbiome primers described <a href="https://earthmicrobiome.org/protocols-and-standards/16s/" class="external-link">here</a>,
and a few primer sequences were still found within a small number of
reads. <em>primerinfo_params.csv</em>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:100px; overflow-x: scroll; width:100%; ">
<table class="table table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
primer_name
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
forward
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
reverse
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
already_trimmed
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minCutadaptlength
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
multithread
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
verbose
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxN
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxEE_forward
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxEE_reverse
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncLen_forward
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncLen_reverse
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncQ
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minLen
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxLen
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minQ
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
trimLeft
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
trimRight
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rm.lowcomplex
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minOverlap
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxMismatch
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
min_asv_length
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
r16S
</td>
<td style="text-align:left;">
GTGYCAGCMGCCGCGGTAA
</td>
<td style="text-align:left;">
GGACTACNVGGGTWTCTAAT
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
240
</td>
<td style="text-align:right;">
160
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
50
</td>
</tr></tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-1">Step 1<a class="anchor" aria-label="anchor" href="#step-1"></a>
</h3>
<p>Remove N’s and create directory structure for downstream steps</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outputs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/mothur_16S_sop/data"</span>, </span>
<span>  output_directory <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs"</span>,</span>
<span>  tempdir_path <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs_temp"</span>,</span>
<span>  tempdir_id <span class="op">=</span> <span class="st">"temp_files"</span><span class="op">)</span></span></code></pre></div>
<p><img src="prepare_reads-1.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="step-2">Step 2<a class="anchor" aria-label="anchor" href="#step-2"></a>
</h3>
<p>Run Cutadapt to remove primers and then trim reads with DADA2
filterAndTrim function</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/opt/homebrew/bin/cutadapt"</span><span class="op">)</span></span></code></pre></div>
<p>We can now visualize the outputs from the primer removal and trimming
steps. A CSV files is output showing which samples still have primer
sequences and the barplot below summarizes the outputs.</p>
<p>There are circumstances where a few primer sequences may still
remain. If so, any ASVs with any residual primer sequences will be
filtered at the end.</p>
<p><img src="posttrim_primer_plot_16S.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="step-3">Step 3<a class="anchor" aria-label="anchor" href="#step-3"></a>
</h3>
<p>Core ASV inference step</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span><span class="op">)</span></span></code></pre></div>
<p>We can now visualize the outputs from the ASV inference step.</p>
<p>The first plot shows the how reads were merged in terms of mismatches
and indels.</p>
<p>The plot to the right shows the overlap lengths across the inferred
ASVs.</p>
<p><img src="read_merging_info_r16S.png" width="50%"></p>
<p>We can also look at the distribution of ASV lengths</p>
<p><img src="asv_seqlength_plot_r16S.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="step-4">Step 4<a class="anchor" aria-label="anchor" href="#step-4"></a>
</h3>
<p>Assign taxonomy step</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  db_16S<span class="op">=</span><span class="st">"silva_nr99_v138.2_toSpecies_trainset.fa.gz"</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>As a check we can take a look at read counts across the workflow. If
there are sudden drops, we should reconsider our adjusting certain DADA2
parameters and re-running the analysis.</p>
<pre><code><span><span class="co">#&gt; Tables showing read counts throughout the demulticoder DADA2 workflow</span></span>
<span><span class="co">#&gt;    samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1           F3D0_r16S  7733     7112      6976      6979   6540    6528</span></span>
<span><span class="co">#&gt; 2           F3D1_r16S  5829     5299      5226      5239   5027    5016</span></span>
<span><span class="co">#&gt; 3         F3D141_r16S  5926     5463      5331      5357   4986    4863</span></span>
<span><span class="co">#&gt; 4         F3D142_r16S  3158     2914      2799      2830   2595    2521</span></span>
<span><span class="co">#&gt; 5         F3D143_r16S  3164     2941      2822      2868   2553    2519</span></span>
<span><span class="co">#&gt; 6         F3D144_r16S  4798     4312      4150      4228   3646    3507</span></span>
<span><span class="co">#&gt; 7         F3D145_r16S  7331     6741      6592      6627   6079    5820</span></span>
<span><span class="co">#&gt; 8         F3D146_r16S  4993     4560      4450      4470   3968    3879</span></span>
<span><span class="co">#&gt; 9         F3D147_r16S 16956    15636     15433     15505  14233   13006</span></span>
<span><span class="co">#&gt; 10        F3D148_r16S 12332    11412     11250     11267  10529    9935</span></span>
<span><span class="co">#&gt; 11        F3D149_r16S 13006    12017     11857     11898  11154   10653</span></span>
<span><span class="co">#&gt; 12        F3D150_r16S  5474     5032      4879      4925   4349    4240</span></span>
<span><span class="co">#&gt; 13          F3D2_r16S 19489    18075     17907     17939  17431   16835</span></span>
<span><span class="co">#&gt; 14          F3D3_r16S  6726     6250      6145      6176   5850    5486</span></span>
<span><span class="co">#&gt; 15          F3D5_r16S  4418     4052      3930      3991   3713    3713</span></span>
<span><span class="co">#&gt; 16          F3D6_r16S  7933     7369      7231      7294   6865    6678</span></span>
<span><span class="co">#&gt; 17          F3D7_r16S  5103     4765      4646      4673   4428    4217</span></span>
<span><span class="co">#&gt; 18          F3D8_r16S  5274     4871      4786      4802   4576    4547</span></span>
<span><span class="co">#&gt; 19          F3D9_r16S  7023     6504      6341      6442   6092    6015</span></span>
<span><span class="co">#&gt; 20          Mock_r16S  4748     4314      4287      4286   4269    4269</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="step-5">Step 5<a class="anchor" aria-label="anchor" href="#step-5"></a>
</h3>
<p>Convert asv matrix to taxmap and phyloseq objects with one
function</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">outputs</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">232</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>  (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (20): F3D0_r16S, F3D1_r16S, F3D141_r16S, F3D142_r16S, F3D143_r16S, F3D14...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For r16S dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs/taxmap_obj_r16S.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs/phylo_obj_r16S.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-6">Step 6<a class="anchor" aria-label="anchor" href="#step-6"></a>
</h3>
<p>Evaluate accuracy using mock community, as shown in dada2
tutorial</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">track_reads_demulticoder</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs/track_reads_r16S.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">track_reads_demulticoder</span><span class="op">)</span></span>
<span><span class="co">#&gt;      input          filtered       denoisedF       denoisedR    </span></span>
<span><span class="co">#&gt;  Min.   : 3158   Min.   : 2914   Min.   : 2799   Min.   : 2830  </span></span>
<span><span class="co">#&gt;  1st Qu.: 4944   1st Qu.: 4498   1st Qu.: 4409   1st Qu.: 4424  </span></span>
<span><span class="co">#&gt;  Median : 5878   Median : 5381   Median : 5278   Median : 5298  </span></span>
<span><span class="co">#&gt;  Mean   : 7571   Mean   : 6982   Mean   : 6852   Mean   : 6890  </span></span>
<span><span class="co">#&gt;  3rd Qu.: 7783   3rd Qu.: 7176   3rd Qu.: 7040   3rd Qu.: 7058  </span></span>
<span><span class="co">#&gt;  Max.   :19489   Max.   :18075   Max.   :17907   Max.   :17939  </span></span>
<span><span class="co">#&gt;      merged         nonchim     </span></span>
<span><span class="co">#&gt;  Min.   : 2553   Min.   : 2519  </span></span>
<span><span class="co">#&gt;  1st Qu.: 4194   1st Qu.: 4132  </span></span>
<span><span class="co">#&gt;  Median : 5006   Median : 4940  </span></span>
<span><span class="co">#&gt;  Mean   : 6444   Mean   : 6212  </span></span>
<span><span class="co">#&gt;  3rd Qu.: 6621   3rd Qu.: 6566  </span></span>
<span><span class="co">#&gt;  Max.   :17431   Max.   :16835</span></span>
<span></span>
<span><span class="va">tax_matrix</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"~/benchmark_demulticoder/mothur_16S_sop/vignette_outputs/final_asv_abundance_matrix_r16S.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">unqs.mock</span> <span class="op">&lt;-</span> <span class="va">tax_matrix</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tax_matrix</span><span class="op">)</span> <span class="op">==</span> <span class="st">"Mock_r16S"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">unqs.mock</span> <span class="op">&lt;-</span> <span class="va">unqs.mock</span><span class="op">[</span><span class="va">unqs.mock</span><span class="op">$</span><span class="va">Mock_r16S</span> <span class="op">!=</span> <span class="fl">0</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"DADA2 inferred"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">unqs.mock</span><span class="op">)</span>, <span class="st">"sample sequences present in the Mock community.\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; DADA2 inferred 20 sample sequences present in the Mock community.</span></span>
<span></span>
<span><span class="va">mock.ref</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/getSequences.html" class="external-link">getSequences</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~/benchmark_demulticoder/mothur_16S_sop/data"</span>, <span class="st">"HMP_MOCK.v35.fasta"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">match.ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">unqs.mock</span><span class="op">$</span><span class="va">sequence</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">mock.ref</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Of those,"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">match.ref</span><span class="op">)</span>, <span class="st">"were exact matches to the expected reference sequences.\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Of those, 20 were exact matches to the expected reference sequences.</span></span></code></pre></div>
<p>When looking at mock community sample, we were able to extract 20
bacterial sequences with 0% mismatch, and this matched what was
described previously.</p>
</div>
<div class="section level3">
<h3 id="step-7">Step 7<a class="anchor" aria-label="anchor" href="#step-7"></a>
</h3>
<p>Follow-up work using phyloseq to do side-by-side comparison with
dada2 example and to examine alpha diversity results</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html" class="external-link">prune_samples</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html" class="external-link">sample_names</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"Mock_r16S"</span>, <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span> <span class="co"># Remove mock sample</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_richness.html" class="external-link">plot_richness</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, x<span class="op">=</span><span class="st">"Day"</span>, measures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Shannon"</span>, <span class="st">"Simpson"</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"When"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_alpha_diversity_analysis-1.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="step-8">Step 8<a class="anchor" aria-label="anchor" href="#step-8"></a>
</h3>
<p>Examine ordination plots as additional point of comparison with DADA2
tutorial</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform data to proportions as appropriate for Bray-Curtis distances</span></span>
<span><span class="va">ps.prop</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, <span class="kw">function</span><span class="op">(</span><span class="va">otu</span><span class="op">)</span> <span class="va">otu</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">otu</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ord.nmds.bray</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html" class="external-link">ordinate</a></span><span class="op">(</span><span class="va">ps.prop</span>, method<span class="op">=</span><span class="st">"NMDS"</span>, distance<span class="op">=</span><span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html" class="external-link">plot_ordination</a></span><span class="op">(</span><span class="va">ps.prop</span>, <span class="va">ord.nmds.bray</span>, color<span class="op">=</span><span class="st">"When"</span>, title<span class="op">=</span><span class="st">"Bray NMDS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_beta_diversity_analysis-1.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="step-9">Step 9<a class="anchor" aria-label="anchor" href="#step-9"></a>
</h3>
<p>Let’s look at what the top 20 taxa are in the early vs. late samples
time points, as shown in the dada2 tutorial</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top20</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html" class="external-link">taxa_sums</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span><span class="op">)</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span><span class="va">ps.top20</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_r16S</span>, <span class="kw">function</span><span class="op">(</span><span class="va">OTU</span><span class="op">)</span> <span class="va">OTU</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">OTU</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ps.top20</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html" class="external-link">prune_taxa</a></span><span class="op">(</span><span class="va">top20</span>, <span class="va">ps.top20</span><span class="op">)</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_bar.html" class="external-link">plot_bar</a></span><span class="op">(</span><span class="va">ps.top20</span>, x<span class="op">=</span><span class="st">"Day"</span>, fill<span class="op">=</span><span class="st">"Family"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">When</span>, scales<span class="op">=</span><span class="st">"free_x"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_community_analysis-1.png" width="50%"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
