<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Documentation • demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Documentation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes">
<li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Documentation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/articles/Documentation.Rmd" class="external-link"><code>vignettes/articles/Documentation.Rmd</code></a></small>
      <div class="d-none name"><code>Documentation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This documentation page provides additional information on to use the
<strong><code>demulticoder</code></strong> R package for processing and
analyzing metabarcode sequencing data. Specifically, it provides more
detail on input directory and file requirements, and key parameters.</p>
</div>
<div class="section level2">
<h2 id="package-workflow">Package workflow<a class="anchor" aria-label="anchor" href="#package-workflow"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Prepare your input files (
<strong><code>metadata.csv</code></strong>,
<strong><code>primerinfo_params.csv</code></strong>, unformatted
reference databases, and PE Illumina read files).</li>
<li>Place all input files in a single directory.</li>
<li>Ensure your file names comply with the specified format.</li>
<li>Run the four steps/functions of pipeline with default settings or
adjust parameters as needed.</li>
</ol>
</div>
<div class="section level2">
<h2 id="data-directory-structure">Data directory structure<a class="anchor" aria-label="anchor" href="#data-directory-structure"></a>
</h2>
<p>Place all your input files into a single directory. The directory
should contain the following files:</p>
<ul>
<li><strong>PE Illumina read files</strong></li>
<li><strong><code>metadata.csv</code></strong></li>
<li><strong><code>primerinfo_params.csv</code></strong></li>
<li><strong>Reference databases</strong></li>
</ul>
<div class="section level3">
<h3 id="read-name-format">Read Name Format<a class="anchor" aria-label="anchor" href="#read-name-format"></a>
</h3>
<p>To avoid errors, the only characters that are acceptable in sample
names are letters and numbers. Characters can be separated by
underscores, but no other symbols. The files must end with the suffix
<strong><code>R1.fastq.gz</code></strong> or
<strong><code>R2.fastq.gz</code></strong></p>
<p><strong>Examples of permissible sample names are as
follows:</strong></p>
<ul>
<li>Sample1_R1.fastq.gz</li>
<li>Sample1_R2.fastq.gz</li>
</ul>
<p><strong>Other permissible names are:</strong></p>
<ul>
<li>Sample1_001_R1.fastq.gz</li>
<li>Sample1_001_R2.fastq.gz</li>
</ul>
<p><strong>What is not permissible is:</strong></p>
<ul>
<li>Sample1_001_R1_001.fastq.gz</li>
<li>Sample1_001_R2_001.fastq.gz</li>
</ul>
</div>
<div class="section level3">
<h3 id="metadata-file-components">Metadata file components<a class="anchor" aria-label="anchor" href="#metadata-file-components"></a>
</h3>
<p>The <strong>metadata.csv</strong> file contains information about the
samples and primers (and associated metabarcodes) used in the
experiment. It has the following two required columns:</p>
<ul>
<li>
<strong>sample_name</strong>: Identifier for each sample (e.g., S1,
S2)</li>
<li>
<strong>primer_name</strong>: Name of the primer used (applicable
options: <code>rps10</code>, <code>its</code>, <code>r16S</code>,
<code>other1</code>, <code>other2</code>)</li>
</ul>
<p>Please add your associated metadata to the file after these two
required columns. This can then be used for your downstream exploratory
or diversity analyses, as the sample data will be incorporated into the
final <strong><code>phyloseq</code></strong> and
<strong><code>taxmap</code></strong> objects.</p>
<p>Example file (with optional third column):</p>
<pre class="csv"><code>sample_name,primer_name,organism
S1,rps10,Cry
S2,rps10,Cin
S1,its,Cry
S2,its,Cin</code></pre>
</div>
<div class="section level3">
<h3 id="primer-and-parameter-file-components">Primer and parameter file components<a class="anchor" aria-label="anchor" href="#primer-and-parameter-file-components"></a>
</h3>
<p>The <strong>primerinfo_params.csv</strong> file contains information
about the primer sequences used in the experiment, along with optional
additional parameters that are part of the
<strong><code>DADA2</code></strong> pipeline. If anything is not
specified, the default values will be used.</p>
<p><strong>Required columns:</strong></p>
<ul>
<li>
<strong><code>primer_name</code></strong>: Name of the
primer/barcode (e.g., <code>its</code>, <code>rps10</code>)</li>
<li>
<strong><code>forward</code></strong>: Forward primer sequence</li>
<li>
<strong><code>reverse</code></strong>: Reverse primer sequence</li>
</ul>
<p>Below are the parameters that can be input into the
<strong>primerinfo_params.csv</strong> file along with the defaults.
Refer to the <strong><code>DADA2</code></strong> documentation and <a href="https://www.bioconductor.org/packages/release/bioc/manuals/dada2/man/dada2.pdf" class="external-link">manual</a>
for additional information.</p>
<p><strong><code>DADA2</code> <code>filterAndTrim</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>already_trimmed</code></strong>: Boolean indicating if
primers are already trimmed (<code>TRUE/FALSE</code>) (default:
<code>FALSE</code>)</li>
<li>
<strong><code>minCutadaptlength</code></strong>:
<strong><code>Cutadapt</code></strong> parameter-Filter out processed
reads that are shorter than specified length (default:
<code>0</code>)</li>
<li>
<strong><code>multithread</code></strong>: Boolean for
multithreading (<code>TRUE/FALSE</code>) (default:
<code>FALSE</code>)</li>
<li>
<strong><code>verbose</code></strong>: Boolean for verbose output
(<code>TRUE/FALSE</code>) (default: <code>FALSE</code>)</li>
<li>
<strong><code>maxN</code></strong>: Maximum number of N bases
allowed (default: <code>0</code>)</li>
<li>
<strong><code>maxEE_forward</code></strong>: Maximum expected errors
for forward reads (default: <code>Inf</code>)</li>
<li>
<strong><code>maxEE_reverse</code></strong>: Maximum expected errors
for reverse reads (default: <code>Inf</code>)</li>
<li>
<strong><code>truncLen_forward</code></strong>: Truncation length
for forward reads (default: <code>0</code>)</li>
<li>
<strong><code>truncLen_reverse</code></strong>: Truncation length
for reverse reads (default: <code>0</code>)</li>
<li>
<strong><code>truncQ</code></strong>: Truncation quality threshold
(default: <code>2</code>)</li>
<li>
<strong><code>minLen</code></strong>: Minimum length of reads after
processing (default: <code>20</code>)</li>
<li>
<strong><code>maxLen</code></strong>: Maximum length of reads after
processing (default: <code>Inf</code>)</li>
<li>
<strong><code>minQ</code></strong>: Minimum quality score (default:
<code>0</code>)</li>
<li>
<strong><code>trimLeft</code></strong>: Number of bases to trim from
the start of reads (default: <code>0</code>)</li>
<li>
<strong><code>trimRight</code></strong>: Number of bases to trim
from the end of reads (default: <code>0</code>)</li>
<li>
<strong><code>rm.lowcomplex</code></strong>: Boolean for removing
low complexity sequences (default: <code>TRUE</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>learnErrors</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>nbases</code></strong>: Number of bases to use for
error rate learning (default: <code>1e+08</code>)</li>
<li>
<strong><code>randomize</code></strong>: Randomize reads for error
rate learning (default: <code>FALSE</code>)</li>
<li>
<strong><code>MAX_CONSIST</code></strong>: Maximum number of
self-consistency iterations (default: <code>10</code>)</li>
<li>
<strong><code>OMEGA_C</code></strong>: Convergence threshold for the
error rates (default: <code>0</code>)</li>
<li>
<strong><code>qualityType</code></strong>: Quality score type
(<code>"Auto"</code>, <code>"FastqQuality",</code> or
<code>"ShortRead"</code>) (default: <code>"Auto"</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>plotErrors</code>
parameters:</strong></p>
<ul>
<li>
<strong><code>err_out</code></strong>: Return the error rates used
for inference (default: <code>TRUE</code>)</li>
<li>
<strong><code>err_in</code></strong>: Use input error rates instead
of learning them (default: <code>FALSE</code>)</li>
<li>
<strong><code>nominalQ</code></strong>: Use nominal Q-scores
(default: <code>FALSE</code>)</li>
<li>
<strong><code>obs</code></strong>: Return the observed error rates
(default: <code>TRUE</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>dada</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>OMP</code></strong>: Use OpenMP multi-threading if
available (default: <code>TRUE</code>)</li>
<li>
<strong><code>n</code></strong>: Number of reads to use for error
rate estimation (default: <code>1e+05</code>)</li>
<li>
<strong><code>id.sep</code></strong>: Character separating sample ID
from sequence name (default: <code>"\\s"</code>)</li>
<li>
<strong><code>orient.fwd</code></strong>: NULL or TRUE/FALSE to
orient sequences (default: <code>NULL</code>)</li>
<li>
<strong><code>pool</code></strong>: Pool samples for error rate
estimation (default: <code>FALSE</code>)</li>
<li>
<strong><code>selfConsist</code></strong>: Perform self-consistency
iterations (default: <code>FALSE</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>mergePairs</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>minOverlap</code></strong>: Minimum overlap for
merging paired-end reads (default: <code>12</code>)</li>
<li>
<strong><code>maxMismatch</code></strong>: Maximum mismatches
allowed in the overlap region (default: <code>0</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>removeBimeraDenovo</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>method</code></strong>: Method for sample inference
(<code>"consensus"</code> or <code>"pooled"</code>) (default:
<code>"consensus"</code>)</li>
</ul>
<p><strong><code>DADA2</code> <code>assignTaxonomy</code> function
parameters:</strong></p>
<ul>
<li>
<strong><code>minBoot</code></strong>: The minimum bootstrap
confidence for assigning a taxonomic level (default:
<code>0</code>)<br>
</li>
<li>
<strong><code>tryRC</code></strong>: If TRUE, the reverse-complement
of each sequences will be used for classification if it is a better
match to the reference sequences than the forward sequence (default:
<code>FALSE</code>)</li>
</ul>
<p><strong>Other parameters to include in CSV input file:</strong></p>
<ul>
<li>
<strong><code>min_asv_length</code></strong>: Minimum length of
Amplicon Sequence Variants (ASVs) after core dada ASV inference steps
(default = <code>0</code>)<br>
</li>
<li>
<strong><code>seed</code></strong>: For greater reproducibility,
user can specify an integer to set as a seed to use when the following
<strong><code>DADA2</code></strong> functions are run:
<strong><code>plotQualityProfile</code></strong>,
<strong><code>learnErrors</code></strong>,
<strong><code>dada</code></strong>,
<strong><code>makeSequenceTable</code></strong>, and
<strong><code>assignTaxonomy</code></strong> (default:
<code>NULL</code>)</li>
</ul>
<p>Example file (with select optional columns after forward and reverse
primer sequence columns):</p>
<pre class="csv"><code>primer_name,forward,reverse,already_trimmed,minCutadaptlength,multithread,verbose,maxN,maxEE_forward,maxEE_reverse,truncLen_forward,truncLen_reverse,truncQ,minLen,maxLen,minQ,trimLeft,trimRight,rm.lowcomplex,minOverlap,maxMismatch,min_asv_length
rps10,GTTGGTTAGAGYARAAGACT,ATRYYTAGAAAGAYTYGAACT,FALSE,100,TRUE,FALSE,1.00E+05,5,5,0,0,5,150,Inf,0,0,0,0,15,0,50
its,CTTGGTCATTTAGAGGAAGTAA,GCTGCGTTCTTCATCGATGC,FALSE,50,TRUE,FALSE,1.00E+05,5,5,0,0,5,50,Inf,0,0,0,0,15,0,50</code></pre>
</div>
<div class="section level3">
<h3 id="reference-databases">Reference Databases<a class="anchor" aria-label="anchor" href="#reference-databases"></a>
</h3>
<p>Databases will be copied into the user-specified data folder where
raw data files and csv files are located. The names will be parameters
in the <em>assignTax</em> function.</p>
<p>For now, the package is compatible with the following databases:</p>
<ul>
<li><p><strong><code>oomyceteDB</code></strong> from: <a href="https://grunwaldlab.github.io/OomyceteDB/" class="external-link">https://grunwaldlab.github.io/OomyceteDB/</a></p></li>
<li>
<p><strong><code>SILVA 16S database</code></strong>
<strong>with</strong> species assignments: <a href="https://www.arb-silva.de/" class="external-link">https://www.arb-silva.de/</a></p>
<ul>
<li>An easily accessible download is found here: <a href="https://zenodo.org/records/14169026" class="external-link">https://zenodo.org/records/14169026</a>
</li>
</ul>
</li>
<li><p><strong><code>UNITE fungal database</code></strong> from <a href="https://zenodo.org/records/14169026" class="external-link">https://zenodo.org/records/14169026</a></p></li>
<li><p>Up to two other reference databases. The user will need to
reformat headers exactly as outlined <a href="https://benjjneb.github.io/dada2/training.html" class="external-link">here</a>. The user
can then specify the path to the database in the input file. The
database should be in FASTA format.</p></li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
