<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Package Workflow and Quick Start • demulticoder</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Package Workflow and Quick Start">
<meta property="og:description" content="demulticoder">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">demulticoder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Package_Workflow.html">Package Workflow and Quick Start</a>
    </li>
    <li>
      <a href="../articles/example_large_dataset.html">Example workflow and proof of concept analysis of pooled RPS10 and ITS amplicons rhodendron rhizobiome dataset </a>
    </li>
    <li>
      <a href="../articles/validation_16S_mothur_SOP.html">mothur_SOP_16S_validation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Package Workflow and Quick Start</h1>
            
      
      
      <div class="hidden name"><code>Package_Workflow.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>The <em>demulticoder</em> package is a Cutadapt and DADA2 wrapper
package for metabarcoding analyses. The package is designed for users
with varying experience doing metabarcoding analysis and a strength is
the ability to to analyses on multiple datasets at once, when amplicons
have been pooled and contain different barcodes, or when samples are
already demultiplexed.</p>
<p>The package is capable of streamlined analysis of ITS or 16S
metabarcoding data, as well as oomycete rps10 amplicons. There is also
the option to input up to two custom databases, so long as databases are
formatted according to DADA2 specifications (<a href="https://benjjneb.github.io/dada2/training.html" class="external-link uri">https://benjjneb.github.io/dada2/training.html</a>), and has
no more than 7 taxonomic levels from kingdom down to species.</p>
<p>The package was developed by Martha Sudermann, Zachary Foster,
Niklaus Grunwald, Jeff Chang.</p>
</div>
<div class="section level3">
<h3 id="before-you-start">Before You Start<a class="anchor" aria-label="anchor" href="#before-you-start"></a>
</h3>
<p>You will need to prepare your raw read files and fill in the
‘metadata.csv’ and ‘primerinfo_params.csv’ templates.</p>
<p>A strength of the package is that you can run multiple different
datasets at once, which is shown in test dataset below. If samples
contain pooled barcodes, you can also use the package to demultiplex and
run analyses in tandem. S1 and S2 samples below illustrate this. Both
samples contain rps10 and ITS pooled barcodes.</p>
</div>
<div class="section level3">
<h3 id="format-of-the-raw-read-files">Format of the Raw Read Files<a class="anchor" aria-label="anchor" href="#format-of-the-raw-read-files"></a>
</h3>
<p>The package takes foward and reverse Illumina short read sequence
data.</p>
<p>To avoid errors, the only characters that are acceptable in sample
names are letters and numbers. Characters can be separated by
underscores, but no other symbols. The final characters before the
‘.fastq.gz’ suffix MUST be ’_R1’ or ’_R2’.</p>
<p>Examples are permissible sample names are as follows:</p>
<ul>
<li>Sample1_R1.fastq.gz</li>
<li>Sample1_R2.fastq.gz</li>
</ul>
<p>Other permissible names are:</p>
<ul>
<li>Sample1_001_R1.fastq.gz</li>
<li>Sample1_001_R2.fastq.gz</li>
</ul>
<p>What is not permissible and will have to be renamed is:</p>
<ul>
<li>Sample1_001_R1_001.fastq.gz</li>
<li>Sample1_001_R2_001.fastq.gz</li>
</ul>
<p>The parsing functions will error out because the R1 and R2 are not
directly preceeding the ‘.fastq.gz’ suffix.</p>
</div>
<div class="section level3">
<h3 id="format-of-metadata-file-metadata-csv">Format of metadata file (<em>metadata.csv</em>)<a class="anchor" aria-label="anchor" href="#format-of-metadata-file-metadata-csv"></a>
</h3>
<p>The format of the CSV file is simple. The only two necessary columns
(with these names) are:</p>
<ol style="list-style-type: decimal">
<li>sample_name column</li>
<li>primer_info column</li>
</ol>
<p>Any additional metadata should then be pasted after these two
columns. They can be referenced later during the analysis steps and save
a step of loading metadata later.</p>
<p>S1 and S2 are come from a rhododendron rhizobiome dataset and are a
random subset of reads.</p>
<p>You will notice S1 and S2 are included twice in the ‘metadata.csv’
sheet. This is because these two samples contain pooled reads (ITS and
rps10). To demultiplex and run both analyses in tandem, include the same
sample twice under sample_name, and then change the primer_name.</p>
<p>Example using test dataset:</p>
<table class="table">
<thead><tr class="header">
<th align="left">sample_name</th>
<th align="left">primer_name</th>
<th align="left">well</th>
<th align="left">organism</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">S1</td>
<td align="left">rps10</td>
<td align="left">A01</td>
<td align="left">Cry</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">rps10</td>
<td align="left">B01</td>
<td align="left">Cin</td>
</tr>
<tr class="odd">
<td align="left">S1</td>
<td align="left">its</td>
<td align="left">A01</td>
<td align="left">Cry</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">its</td>
<td align="left">B01</td>
<td align="left">Cin</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="primer-sequence-information-and-user-defined-parameters-are-placed-in-primerinfo_params-csv">Primer sequence information and user-defined parameters are placed
in <em>primerinfo_params.csv</em><a class="anchor" aria-label="anchor" href="#primer-sequence-information-and-user-defined-parameters-are-placed-in-primerinfo_params-csv"></a>
</h3>
<p>To simplify how functions are called, user will provide parameters
within this input file. We recommend using the template provided in
documentation.</p>
<p>The only required columns user must fill in are: 1.primer_name
(rps10, its, or other for another barcode) 2.forward-forward sequence
3.reverse-reverse sequence 4.already_trimmed (TRUE/FALSE)</p>
<p>(Most datasets will require that primers first be removed from reads.
However, for some 16S datasets, if protocols like the Earth Microbiome
Project were followed, primers are mostly removed during the
demultiplexing for barcoded samples following an Illumina run). It may
be that a few primers still remain. If already_trimmed flag is
specified, remaining primers will be removed and then reads for which no
primers were located will be copied to the trimmed folder for subsequent
analyses.</p>
<p>If a user doesn’t add any info to the subsequent columns, the default
DADA2 parameters will be used.</p>
<p>TODO-provide info on the different parameter descriptions and which
functions they are associated with.</p>
<p>Example template for ‘primerinfo_params.csv’</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="4%">
<col width="8%">
<col width="7%">
<col width="5%">
<col width="6%">
<col width="4%">
<col width="2%">
<col width="3%">
<col width="5%">
<col width="5%">
<col width="6%">
<col width="6%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="3%">
<col width="3%">
<col width="5%">
<col width="3%">
<col width="4%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">primer_name</th>
<th align="left">forward</th>
<th align="left">reverse</th>
<th align="left">already_trimmed</th>
<th align="right">minCutadaptlength</th>
<th align="left">multithread</th>
<th align="left">verbose</th>
<th align="left">maxN</th>
<th align="right">maxEE_forward</th>
<th align="right">maxEE_reverse</th>
<th align="right">truncLen_forward</th>
<th align="right">truncLen_reverse</th>
<th align="right">truncQ</th>
<th align="right">minLen</th>
<th align="left">maxLen</th>
<th align="right">minQ</th>
<th align="right">trimLeft</th>
<th align="right">trimRight</th>
<th align="right">rm.lowcomplex</th>
<th align="right">minOverlap</th>
<th align="right">maxMismatch</th>
<th align="right">min_asv_length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rps10</td>
<td align="left">GTTGGTTAGAGYARAAGACT</td>
<td align="left">ATRYYTAGAAAGAYTYGAACT</td>
<td align="left">FALSE</td>
<td align="right">100</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">1.00E+05</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">150</td>
<td align="left">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">its</td>
<td align="left">CTTGGTCATTTAGAGGAAGTAA</td>
<td align="left">GCTGCGTTCTTCATCGATGC</td>
<td align="left">FALSE</td>
<td align="right">50</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">1.00E+05</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">50</td>
<td align="left">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">50</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="reference-database-format">Reference Database Format<a class="anchor" aria-label="anchor" href="#reference-database-format"></a>
</h3>
<p>For now, the package is compatible with the following databases:</p>
<ul>
<li><p>oomycetedb from: <a href="http://www.oomycetedb.org/" class="external-link">http://www.oomycetedb.org/</a></p></li>
<li><p>SILVA 16S database with species assignments: <a href="https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1" class="external-link">https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1</a></p></li>
<li><p>UNITE fungal database from <a href="https://unite.ut.ee/repository.php" class="external-link">https://unite.ut.ee/repository.php</a></p></li>
</ul>
<p>A user can select up to one other database (for now), but will first
need to reformat headers exactly like the UNITE fungal database
specifications.</p>
<p>Databases will be copied into the user-specified data folder where
raw data files and csv files are located. The names will be parameters
in the <em>assignTax</em> function</p>
</div>
<div class="section level3">
<h3 id="additional-notes">Additional Notes<a class="anchor" aria-label="anchor" href="#additional-notes"></a>
</h3>
<p>Computer specifications may be a limiting factor– If you are using
the SILVA or UNITE databases for taxonomic assignment steps, an ordinary
personal computer (unless it has sufficient RAM) may not have enough
memory for the taxonomic assignment steps, even with few samples. The
test databases in the package are randomly subsetted for demonstration
purposes.</p>
<p>Users need to upload their own databases to their input data folder.
If computer crashes during the taxonomic assignment step, please switch
to a computing cluster.</p>
<p>Please also ensure that you have enough storage to save intermediate
files in a temporary directory (default) or user-specified directory
before proceeding.</p>
</div>
<div class="section level3">
<h3 id="loading-the-package">Loading the Package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<p>For now, the package will be loaded by retrieving it from GitHub.
Eventually, the package will be uploaded to CRAN or Bioconductor.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"grunwaldlab/demulticoder"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## knitr (1.45  -&gt; 1.46 ) [CRAN]</span></span>
<span><span class="co">## crul  (1.4.0 -&gt; 1.4.2) [CRAN]</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The downloaded binary packages are in</span></span>
<span><span class="co">##  /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/downloaded_packages</span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="color: #00BBBB;">R CMD build</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## * checking for file ‘/private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpEDjTAY/remotes7b126979f636/grunwaldlab-demulticoder-2684e6b/DESCRIPTION’ ... OK</span></span>
<span><span class="co">## * preparing ‘demulticoder’:</span></span>
<span><span class="co">## * checking DESCRIPTION meta-information ... OK</span></span>
<span><span class="co">## * checking for LF line-endings in source and make files and shell scripts</span></span>
<span><span class="co">## * checking for empty or unneeded directories</span></span>
<span><span class="co">## Omitted ‘LazyData’ from DESCRIPTION</span></span>
<span><span class="co">## * building ‘demulticoder_0.0.0.9000.tar.gz’</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://grunwaldlab.github.io/demulticoder/" class="external-link">demulticoder</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reorganize-data-tables">Reorganize Data Tables<a class="anchor" aria-label="anchor" href="#reorganize-data-tables"></a>
</h3>
<p>The sample names, primer sequences, and other metadata are
reorganized in preparation for running Cutadapt to remove primers.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">demulticoder</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>,</span>
<span>  output_directory <span class="op">=</span> <span class="st">"~/output_test_dataset"</span>, </span>
<span>  tempdir_id <span class="op">=</span> <span class="st">"test_dataset"</span>,</span>
<span>  overwrite_existing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Package_Workflow_files/figure-html/prepare_reads%20function-1.png" width="700"></p>
<pre><code><span><span class="co">##   sample_name samplename_barcode primer_name well organism</span></span>
<span><span class="co">## 1          S1             S1_its         its  A01      Cry</span></span>
<span><span class="co">## 2          S1             S1_its         its  A01      Cry</span></span>
<span><span class="co">## 3          S1           S1_rps10       rps10  A01      Cry</span></span>
<span><span class="co">## 4          S1           S1_rps10       rps10  A01      Cry</span></span>
<span><span class="co">## 5          S2             S2_its         its  B01      Cin</span></span>
<span><span class="co">## 6          S2             S2_its         its  B01      Cin</span></span>
<span><span class="co">## 7          S2           S2_rps10       rps10  B01      Cin</span></span>
<span><span class="co">## 8          S2           S2_rps10       rps10  B01      Cin</span></span>
<span><span class="co">##                  forward                f_compt                  f_rev</span></span>
<span><span class="co">## 1 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 2 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 3   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">## 4   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">## 5 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 6 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 7   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">## 8   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">##                     f_rc               reverse               r_compt</span></span>
<span><span class="co">## 1 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 2 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 3   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">## 4   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">## 5 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 6 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 7   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">## 8   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">##                   r_rev                  r_rc file_id direction</span></span>
<span><span class="co">## 1  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S1_R1   Forward</span></span>
<span><span class="co">## 2  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S1_R2   Reverse</span></span>
<span><span class="co">## 3 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S1_R1   Forward</span></span>
<span><span class="co">## 4 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S1_R2   Reverse</span></span>
<span><span class="co">## 5  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S2_R1   Forward</span></span>
<span><span class="co">## 6  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S2_R2   Reverse</span></span>
<span><span class="co">## 7 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S2_R1   Forward</span></span>
<span><span class="co">## 8 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S2_R2   Reverse</span></span>
<span><span class="co">##                                                                                                                directory_data_path</span></span>
<span><span class="co">## 1 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R1.fastq.gz</span></span>
<span><span class="co">## 4 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R2.fastq.gz</span></span>
<span><span class="co">## 5 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R1.fastq.gz</span></span>
<span><span class="co">## 6 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R2.fastq.gz</span></span>
<span><span class="co">## 7 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R1.fastq.gz</span></span>
<span><span class="co">## 8 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R2.fastq.gz</span></span>
<span><span class="co">##                                                                             temp_data_path</span></span>
<span><span class="co">## 1 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R1.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R2.fastq.gz</span></span>
<span><span class="co">## 5 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R1.fastq.gz</span></span>
<span><span class="co">## 6 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R2.fastq.gz</span></span>
<span><span class="co">## 7 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R1.fastq.gz</span></span>
<span><span class="co">## 8 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R2.fastq.gz</span></span>
<span><span class="co">##                                                                                                 prefiltered_path</span></span>
<span><span class="co">## 1 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R1.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R2.fastq.gz</span></span>
<span><span class="co">## 5 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R1.fastq.gz</span></span>
<span><span class="co">## 6 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R2.fastq.gz</span></span>
<span><span class="co">## 7 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R1.fastq.gz</span></span>
<span><span class="co">## 8 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R2.fastq.gz</span></span>
<span><span class="co">##                                                                                                       trimmed_path</span></span>
<span><span class="co">## 1   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S1_R1_its.fastq.gz</span></span>
<span><span class="co">## 2   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S1_R2_its.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S1_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S1_R2_rps10.fastq.gz</span></span>
<span><span class="co">## 5   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S2_R1_its.fastq.gz</span></span>
<span><span class="co">## 6   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S2_R2_its.fastq.gz</span></span>
<span><span class="co">## 7 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S2_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 8 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/trimmed_sequences/S2_R2_rps10.fastq.gz</span></span>
<span><span class="co">##                                                                                                       untrimmed_path</span></span>
<span><span class="co">## 1   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S1_R1_its.fastq.gz</span></span>
<span><span class="co">## 2   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S1_R2_its.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S1_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S1_R2_rps10.fastq.gz</span></span>
<span><span class="co">## 5   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S2_R1_its.fastq.gz</span></span>
<span><span class="co">## 6   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S2_R2_its.fastq.gz</span></span>
<span><span class="co">## 7 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S2_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 8 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/untrimmed_sequences/S2_R2_rps10.fastq.gz</span></span>
<span><span class="co">##                                                                                                       filtered_path</span></span>
<span><span class="co">## 1   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S1_R1_its.fastq.gz</span></span>
<span><span class="co">## 2   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S1_R2_its.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S1_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S1_R2_rps10.fastq.gz</span></span>
<span><span class="co">## 5   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S2_R1_its.fastq.gz</span></span>
<span><span class="co">## 6   /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S2_R2_its.fastq.gz</span></span>
<span><span class="co">## 7 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S2_R1_rps10.fastq.gz</span></span>
<span><span class="co">## 8 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/filtered_sequences/S2_R2_rps10.fastq.gz</span></span>
<span><span class="co">##   primer_name                forward                f_compt</span></span>
<span><span class="co">## 1         its CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT</span></span>
<span><span class="co">## 2       rps10   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA</span></span>
<span><span class="co">##                    f_rev                   f_rc               reverse</span></span>
<span><span class="co">## 1 AATGAAGGAGATTTACTGGTTC TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC</span></span>
<span><span class="co">## 2   TCAGAARAYGAGATTGGTTG   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT</span></span>
<span><span class="co">##                 r_compt                 r_rev                  r_rc</span></span>
<span><span class="co">## 1  CGACGCAAGAAGTAGCTACG  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span></span>
<span><span class="co">## 2 TAYRRATCTTTCTRARCTTGA TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span></span>
<span><span class="co">##   file_id sample_name direction</span></span>
<span><span class="co">## 1   S1_R1          S1   Forward</span></span>
<span><span class="co">## 2   S1_R2          S1   Reverse</span></span>
<span><span class="co">## 3   S2_R1          S2   Forward</span></span>
<span><span class="co">## 4   S2_R2          S2   Reverse</span></span>
<span><span class="co">##                                                                                                                directory_data_path</span></span>
<span><span class="co">## 1 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R1.fastq.gz</span></span>
<span><span class="co">## 4 /private/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T/RtmpinwAIg/temp_libpathdd3c11e47307/demulticoder/extdata/S2_R2.fastq.gz</span></span>
<span><span class="co">##                                                                             temp_data_path</span></span>
<span><span class="co">## 1 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R1.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/S2_R2.fastq.gz</span></span>
<span><span class="co">##                                                                                                 prefiltered_path</span></span>
<span><span class="co">## 1 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R1.fastq.gz</span></span>
<span><span class="co">## 2 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S1_R2.fastq.gz</span></span>
<span><span class="co">## 3 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R1.fastq.gz</span></span>
<span><span class="co">## 4 /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/prefiltered_sequences/S2_R2.fastq.gz</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2 × 20</span></span></span>
<span><span class="co">##   primer_name already_trimmed minCutadaptlength multithread verbose  maxN</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> rps10       FALSE                         150 TRUE        TRUE        0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> its         FALSE                          50 TRUE        TRUE        0</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 14 more variables: maxEE_forward &lt;dbl&gt;, maxEE_reverse &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   truncLen_forward &lt;dbl&gt;, truncLen_reverse &lt;dbl&gt;, truncQ &lt;dbl&gt;, minLen &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   maxLen &lt;dbl&gt;, minQ &lt;dbl&gt;, trimLeft &lt;dbl&gt;, trimRight &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   rm.lowcomplex &lt;dbl&gt;, minOverlap &lt;dbl&gt;, maxMismatch &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   min_asv_length &lt;dbl&gt;</span></span></span>
<span><span class="co">##   samplename_barcode primer_name sample_name well organism</span></span>
<span><span class="co">## 1             S1_its         its          S1  A01      Cry</span></span>
<span><span class="co">## 3           S1_rps10       rps10          S1  A01      Cry</span></span>
<span><span class="co">## 2             S2_its         its          S2  B01      Cin</span></span>
<span><span class="co">## 4           S2_rps10       rps10          S2  B01      Cin</span></span>
<span><span class="co">##                  forward                f_compt                  f_rev</span></span>
<span><span class="co">## 1 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 3   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">## 2 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span></span>
<span><span class="co">## 4   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span></span>
<span><span class="co">##                     f_rc               reverse               r_compt</span></span>
<span><span class="co">## 1 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 3   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">## 2 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span></span>
<span><span class="co">## 4   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span></span>
<span><span class="co">##                   r_rev                  r_rc</span></span>
<span><span class="co">## 1  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span></span>
<span><span class="co">## 3 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span></span>
<span><span class="co">## 2  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span></span>
<span><span class="co">## 4 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="remove-primers-with-cutadapt">Remove Primers with Cutadapt<a class="anchor" aria-label="anchor" href="#remove-primers-with-cutadapt"></a>
</h3>
<p>Before running Cutadapt, please ensure that you have installed
it.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">demulticoder</span><span class="fu">::</span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/opt/homebrew/bin/cutadapt"</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running Cutadapt 4.1 for its sequence data </span></span>
<span><span class="co">## Running Cutadapt 4.1 for rps10 sequence data</span></span></code></pre>
<p><img src="Package_Workflow_files/figure-html/remove%20primers%20with%20cut_trim%20function-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="asv-inference-step">ASV inference step<a class="anchor" aria-label="anchor" href="#asv-inference-step"></a>
</h3>
<p>Raw reads will be merged and ASVs will be inferred</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1658423 total bases in 6294 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">## Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">## selfConsist step 1 ..</span></span>
<span><span class="co">##    selfConsist step 2</span></span>
<span><span class="co">##    selfConsist step 3</span></span>
<span><span class="co">## Convergence after  3  rounds.</span></span>
<span><span class="co">## Error rate plot for the Forward read of primer pair its</span></span></code></pre>
<pre><code><span><span class="co">## Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">## infinite values.</span></span></code></pre>
<pre><code><span><span class="co">## Sample 1 - 3285 reads in 1348 unique sequences.</span></span>
<span><span class="co">## Sample 2 - 3009 reads in 1363 unique sequences.</span></span>
<span><span class="co">## 1689445 total bases in 6294 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">## Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">## selfConsist step 1 ..</span></span>
<span><span class="co">##    selfConsist step 2</span></span>
<span><span class="co">##    selfConsist step 3</span></span>
<span><span class="co">##    selfConsist step 4</span></span>
<span><span class="co">## Convergence after  4  rounds.</span></span>
<span><span class="co">## Error rate plot for the Reverse read of primer pair its</span></span></code></pre>
<pre><code><span><span class="co">## Sample 1 - 3285 reads in 2157 unique sequences.</span></span>
<span><span class="co">## Sample 2 - 3009 reads in 1915 unique sequences.</span></span></code></pre>
<p><img src="Package_Workflow_files/figure-html/ASV%20inference-1.png" width="700"></p>
<pre><code><span><span class="co">## 1918751 total bases in 6828 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">## Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">## selfConsist step 1 ..</span></span>
<span><span class="co">##    selfConsist step 2</span></span>
<span><span class="co">## Convergence after  2  rounds.</span></span>
<span><span class="co">## Error rate plot for the Forward read of primer pair rps10</span></span></code></pre>
<pre><code><span><span class="co">## Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">## infinite values.</span></span></code></pre>
<pre><code><span><span class="co">## Sample 1 - 3245 reads in 1996 unique sequences.</span></span>
<span><span class="co">## Sample 2 - 3583 reads in 2250 unique sequences.</span></span>
<span><span class="co">## 1911983 total bases in 6828 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">## Initializing error rates to maximum possible estimate.</span></span>
<span><span class="co">## selfConsist step 1 ..</span></span>
<span><span class="co">##    selfConsist step 2</span></span>
<span><span class="co">##    selfConsist step 3</span></span>
<span><span class="co">## Convergence after  3  rounds.</span></span>
<span><span class="co">## Error rate plot for the Reverse read of primer pair rps10</span></span></code></pre>
<pre><code><span><span class="co">## Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">## infinite values.</span></span></code></pre>
<pre><code><span><span class="co">## Sample 1 - 3245 reads in 2254 unique sequences.</span></span>
<span><span class="co">## Sample 2 - 3583 reads in 2951 unique sequences.</span></span></code></pre>
<p><img src="Package_Workflow_files/figure-html/ASV%20inference-2.png" width="700"><img src="Package_Workflow_files/figure-html/ASV%20inference-3.png" width="700"><img src="Package_Workflow_files/figure-html/ASV%20inference-4.png" width="700"></p>
<pre><code><span><span class="co">## $its</span></span>
<span><span class="co">## [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/asvabund_matrixDADA2_its.RData"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $rps10</span></span>
<span><span class="co">## [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpEDjTAY/test_dataset/asvabund_matrixDADA2_rps10.RData"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="taxonomic-assignment-step">Taxonomic assignment step<a class="anchor" aria-label="anchor" href="#taxonomic-assignment-step"></a>
</h3>
<p>Using the core assignTaxonomy function from DADA2, taxonomic
assignments will be given to ASVs.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>  overwrite_existing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">## 1             S1_its  5678     3285      3202      3205   2976    2976</span></span>
<span><span class="co">## 2             S2_its  4827     3009      2911      2864   2742    2742</span></span>
<span><span class="co">##   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">## 1           S1_rps10  4119     3245      3243      3223   3215    3215</span></span>
<span><span class="co">## 2           S2_rps10  4985     3583      3579      3580   3186    3186</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="reformat-asv-matrix-as-taxmap-and-phyloseq-objects-after-optional-filtering-of-low-abundance-asvs">Reformat ASV matrix as taxmap and phyloseq objects after optional
filtering of low abundance ASVs<a class="anchor" aria-label="anchor" href="#reformat-asv-matrix-as-taxmap-and-phyloseq-objects-after-optional-filtering-of-low-abundance-asvs"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span>save_outputs<span class="op">=</span><span class="cn">TRUE</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## For its dataset </span></span>
<span><span class="co">## Taxmap object saved in: ~/output_test_dataset/obj_dada_its.RData </span></span>
<span><span class="co">## Phyloseq object saved in: ~/output_test_dataset/phylo_obj_its.RData </span></span>
<span><span class="co">## ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">## For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to NA </span></span>
<span><span class="co">## ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co">## For rps10 dataset </span></span>
<span><span class="co">## Taxmap object saved in: ~/output_test_dataset/obj_dada_rps10.RData </span></span>
<span><span class="co">## Phyloseq object saved in: ~/output_test_dataset/phylo_obj_rps10.RData </span></span>
<span><span class="co">## ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">## For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to NA </span></span>
<span><span class="co">## ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded samples<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples"></a>
</h3>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples-1">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded samples<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples-1"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span><span class="va">obj_dada_its</span>,</span>
<span>          node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>          node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>          node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>          layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Package_Workflow_files/figure-html/Heatree%20demonstration%20its-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-sample">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded sample<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-sample"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span><span class="va">obj_dada_rps10</span>,</span>
<span>          node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>          node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>          node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>          layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Package_Workflow_files/figure-html/Heatree%20demonstration%20rps10-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="we-can-also-do-a-variety-of-analyses-if-we-convert-to-phyloseq-object">We can also do a variety of analyses, if we convert to phyloseq
object<a class="anchor" aria-label="anchor" href="#we-can-also-do-a-variety-of-analyses-if-we-convert-to-phyloseq-object"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">phylo_obj_its</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom</a></span><span class="op">(</span>taxrank <span class="op">=</span> <span class="st">"Genus"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>                     </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>                                        </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>                      </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">2</span>, <span class="st">"lines"</span><span class="op">)</span>,  <span class="co"># Adjust the size of the legend color key</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Package_Workflow_files/figure-html/Phyloseq%20demonstration%20ITS-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">phylo_obj_rps10</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co">#tax_glom(taxrank = "Genus") %&gt;%                     </span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>                                        </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>                      </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">2</span>, <span class="st">"lines"</span><span class="op">)</span>,  <span class="co"># Adjust the size of the legend color key</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Package_Workflow_files/figure-html/Phyloseq%20demonstration%20rps10-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Martha Sudermann.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
