<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="demulticoder">
<title>Package Workflow • demulticoder</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package Workflow">
<meta property="og:description" content="demulticoder">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/package_workflow.html">Package Workflow</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Package Workflow</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/package_workflow.Rmd" class="external-link"><code>vignettes/package_workflow.Rmd</code></a></small>
      <div class="d-none name"><code>package_workflow.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <em>demulticoder</em> package is a Cutadapt and DADA2 wrapper
package for metabarcoding analyses. The package is designed for users
with varying experience doing metabarcoding analysis. Outputs are
intuitive and comprehensive to account for the iterative nature of
metabarcodeing analyses.</p>
<div class="section level3">
<h3 id="key-features">Key features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h3>
<ul>
<li>The ability to do analysis either demultiplexed or pooled amplicons
within sample<br>
</li>
<li>Multiple datasets can be trimmed of primers, filtered, denoised,
merged, and assigned taxonomy in one go<br>
</li>
<li>The package handles not just 16S or ITS datasets when using default
UNITE fungal or Silva 16S databases but also oomycete rps10 analyses
using oomycetedb (oomycetedb.org), or up to two custom databases
(provided they are formatted as described here: <a href="https://benjjneb.github.io/dada2/training.html" class="external-link uri">https://benjjneb.github.io/dada2/training.html</a>).</li>
</ul>
<p>The package was developed by Martha Sudermann, Zachary Foster,
Niklaus Grunwald, Jeff Chang.</p>
</div>
<div class="section level3">
<h3 id="before-you-start">Before You Start<a class="anchor" aria-label="anchor" href="#before-you-start"></a>
</h3>
<p>In the following example, we demonstrate key package functionality
using a subset of reads from two samples containing pooled ITS1 fungal
and rps10 oomycete amplicons.</p>
<p>You can follow along because all test data and associated CSV input
files are loaded with the package. Additional examples are also
available on the website.</p>
<p><em>Please note, for speed, the test dataset is comprised of randomly
subset reads for the samples (S1 and S2), and due to database size, the
full UNITE database is included in the package, but is also a smaller
subset of the larger database.</em></p>
<p>You will need to prepare your raw read files and fill in the <a href="https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/metadata.csv" class="external-link"><strong>metadata.csv</strong></a>
and <a href="https://github.com/grunwaldlab/demulticoder/blob/main/inst/extdata/primerinfo_params.csv" class="external-link"><strong>primerinfo_params.csv</strong></a>
templates.</p>
</div>
<div class="section level3">
<h3 id="format-of-the-raw-read-files">Format of the Raw Read Files<a class="anchor" aria-label="anchor" href="#format-of-the-raw-read-files"></a>
</h3>
<p>The package takes foward and reverse Illumina short read sequence
data.</p>
<p>To avoid errors, the only characters that are acceptable in sample
names are letters and numbers. Characters can be separated by
underscores, but no other symbols. The final characters before the
<strong>.fastq.gz</strong> suffix MUST be **_R1** or **_R2**.</p>
<p>Examples are permissible sample names are as follows:</p>
<ul>
<li>Sample1_R1.fastq.gz</li>
<li>Sample1_R2.fastq.gz</li>
</ul>
<p>Other permissible names are:</p>
<ul>
<li>Sample1_001_R1.fastq.gz</li>
<li>Sample1_001_R2.fastq.gz</li>
</ul>
<p>What is not permissible and will have to be renamed is:</p>
<ul>
<li>Sample1_001_R1_001.fastq.gz</li>
<li>Sample1_001_R2_001.fastq.gz</li>
</ul>
<p>The parsing functions will error out because the R1 and R2 are not
directly preceeding the ‘.fastq.gz’ suffix.</p>
</div>
<div class="section level3">
<h3 id="format-of-metadata-file-metadata-csv">Format of metadata file (<em>metadata.csv</em>)<a class="anchor" aria-label="anchor" href="#format-of-metadata-file-metadata-csv"></a>
</h3>
<p>The format of the CSV file is simple. The only two necessary columns
(with these names) are:</p>
<ol style="list-style-type: decimal">
<li>sample_name column</li>
<li>primer_info column</li>
</ol>
<p>Any additional metadata should then be pasted after these two
columns. They can be referenced later during the analysis steps and save
a step of loading metadata later.</p>
<p>S1 and S2 are come from a rhododendron rhizobiome dataset and are a
random subset of reads.</p>
<p>You will notice S1 and S2 are included twice in the ‘metadata.csv’
sheet. This is because these two samples contain pooled reads (ITS and
rps10). To demultiplex and run both analyses in tandem, include the same
sample twice under sample_name, and then change the primer_name.</p>
<p>Example using test dataset:</p>
<table class="table">
<thead><tr class="header">
<th align="left">sample_name</th>
<th align="left">primer_name</th>
<th align="left">well</th>
<th align="left">organism</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">S1</td>
<td align="left">rps10</td>
<td align="left">A01</td>
<td align="left">Cry</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">rps10</td>
<td align="left">B01</td>
<td align="left">Cin</td>
</tr>
<tr class="odd">
<td align="left">S1</td>
<td align="left">its</td>
<td align="left">A01</td>
<td align="left">Cry</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">its</td>
<td align="left">B01</td>
<td align="left">Cin</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="primer-sequence-information-and-user-defined-parameters-are-placed-in-primerinfo_params-csv">Primer sequence information and user-defined parameters are placed
in <em>primerinfo_params.csv</em><a class="anchor" aria-label="anchor" href="#primer-sequence-information-and-user-defined-parameters-are-placed-in-primerinfo_params-csv"></a>
</h3>
<p>To simplify how functions are called, user will provide parameters
within this input file. We recommend using the template provided in
documentation.</p>
<p>The only required columns user must fill in are: 1.primer_name
(rps10, its, or other for another barcode) 2.forward-forward sequence
3.reverse-reverse sequence 4.already_trimmed (TRUE/FALSE)</p>
<p>(Most datasets will require that primers first be removed from reads.
However, for some 16S datasets, if protocols like the Earth Microbiome
Project were followed, primers are mostly removed during the
demultiplexing for barcoded samples following an Illumina run). It may
be that a few primers still remain. If already_trimmed flag is
specified, remaining primers will be removed and then reads for which no
primers were located will be copied to the trimmed folder for subsequent
analyses.</p>
<p>If a user doesn’t add any info to the subsequent columns, the default
DADA2 parameters will be used.</p>
<p>TODO-provide info on the different parameter descriptions and which
functions they are associated with.</p>
<p>Example template for ‘primerinfo_params.csv’</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="4%">
<col width="8%">
<col width="7%">
<col width="5%">
<col width="6%">
<col width="4%">
<col width="2%">
<col width="3%">
<col width="5%">
<col width="5%">
<col width="6%">
<col width="6%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="3%">
<col width="3%">
<col width="5%">
<col width="3%">
<col width="4%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">primer_name</th>
<th align="left">forward</th>
<th align="left">reverse</th>
<th align="left">already_trimmed</th>
<th align="right">minCutadaptlength</th>
<th align="left">multithread</th>
<th align="left">verbose</th>
<th align="left">maxN</th>
<th align="right">maxEE_forward</th>
<th align="right">maxEE_reverse</th>
<th align="right">truncLen_forward</th>
<th align="right">truncLen_reverse</th>
<th align="right">truncQ</th>
<th align="right">minLen</th>
<th align="left">maxLen</th>
<th align="right">minQ</th>
<th align="right">trimLeft</th>
<th align="right">trimRight</th>
<th align="right">rm.lowcomplex</th>
<th align="right">minOverlap</th>
<th align="right">maxMismatch</th>
<th align="right">min_asv_length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rps10</td>
<td align="left">GTTGGTTAGAGYARAAGACT</td>
<td align="left">ATRYYTAGAAAGAYTYGAACT</td>
<td align="left">FALSE</td>
<td align="right">100</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">1.00E+05</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">150</td>
<td align="left">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">its</td>
<td align="left">CTTGGTCATTTAGAGGAAGTAA</td>
<td align="left">GCTGCGTTCTTCATCGATGC</td>
<td align="left">FALSE</td>
<td align="right">50</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">1.00E+05</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">50</td>
<td align="left">Inf</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">50</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="reference-database-format">Reference Database Format<a class="anchor" aria-label="anchor" href="#reference-database-format"></a>
</h3>
<p>For now, the package is compatible with the following databases:</p>
<ul>
<li><p>oomycetedb from: <a href="http://www.oomycetedb.org/" class="external-link">http://www.oomycetedb.org/</a></p></li>
<li><p>SILVA 16S database with species assignments: <a href="https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1" class="external-link">https://zenodo.org/records/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1</a></p></li>
<li><p>UNITE fungal database from <a href="https://unite.ut.ee/repository.php" class="external-link">https://unite.ut.ee/repository.php</a></p></li>
</ul>
<p>A user can select up to one other database (for now), but will first
need to reformat headers exactly like the UNITE fungal database
specifications.</p>
<p>Databases will be copied into the user-specified data folder where
raw data files and csv files are located. The names will be parameters
in the <em>assignTax</em> function</p>
</div>
<div class="section level3">
<h3 id="additional-notes">Additional Notes<a class="anchor" aria-label="anchor" href="#additional-notes"></a>
</h3>
<p>Computer specifications may be a limiting factor– If you are using
the SILVA or UNITE databases for taxonomic assignment steps, an ordinary
personal computer (unless it has sufficient RAM) may not have enough
memory for the taxonomic assignment steps, even with few samples. The
test databases in the package are randomly subsetted for demonstration
purposes.</p>
<p>Users need to upload their own databases to their input data folder.
If computer crashes during the taxonomic assignment step, please switch
to a computing cluster.</p>
<p>Please also ensure that you have enough storage to save intermediate
files in a temporary directory (default) or user-specified directory
before proceeding.</p>
</div>
<div class="section level3">
<h3 id="loading-the-package">Loading the Package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<p>For now, the package will be loaded by retrieving it from GitHub.
Eventually, the package will be uploaded to CRAN or Bioconductor.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#devtools::install_github("grunwaldlab/demulticoder")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://grunwaldlab.github.io/demulticoder/" class="external-link">demulticoder</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reorganize-data-tables">Reorganize Data Tables<a class="anchor" aria-label="anchor" href="#reorganize-data-tables"></a>
</h3>
<p>The sample names, primer sequences, and other metadata are
reorganized in preparation for running Cutadapt to remove primers.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analysis_setup</span><span class="op">&lt;-</span><span class="fu">demulticoder</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>,</span>
<span>  output_directory <span class="op">=</span> <span class="st">"~/output_test_dataset"</span>, </span>
<span>  tempdir_id <span class="op">=</span> <span class="st">"test_dataset"</span>,</span>
<span>  overwrite_existing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/prepare_reads%20function-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="remove-primers-with-cutadapt">Remove Primers with Cutadapt<a class="anchor" aria-label="anchor" href="#remove-primers-with-cutadapt"></a>
</h3>
<p>Before running Cutadapt, please ensure that you have installed
it.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">demulticoder</span><span class="fu">::</span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/opt/homebrew/bin/cutadapt"</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running Cutadapt 4.1 for its sequence data </span></span>
<span><span class="co">#&gt; Running Cutadapt 4.1 for rps10 sequence data</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/remove%20primers%20with%20cut_trim%20function-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="asv-inference-step">ASV inference step<a class="anchor" aria-label="anchor" href="#asv-inference-step"></a>
</h3>
<p>Raw reads will be merged and ASVs will be inferred</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; 710847 total bases in 2694 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Error rate plot for the Forward read of primer pair its</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 1479 reads in 654 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 1215 reads in 610 unique sequences.</span></span>
<span><span class="co">#&gt; 724232 total bases in 2694 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Error rate plot for the Reverse read of primer pair its</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 1479 reads in 1019 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 1215 reads in 814 unique sequences.</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/ASV%20inference-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; 824778 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Error rate plot for the Forward read of primer pair rps10</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 1429 reads in 933 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 1506 reads in 1018 unique sequences.</span></span>
<span><span class="co">#&gt; 821853 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; Error rate plot for the Reverse read of primer pair rps10</span></span>
<span><span class="co">#&gt; Warning in scale_y_log10(): <span style="color: #00BB00;">log-10</span> transformation introduced</span></span>
<span><span class="co">#&gt; infinite values.</span></span>
<span><span class="co">#&gt; Sample 1 - 1429 reads in 1044 unique sequences.</span></span>
<span><span class="co">#&gt; Sample 2 - 1506 reads in 1284 unique sequences.</span></span></code></pre>
<p><img src="package_workflow_files/figure-html/ASV%20inference-2.png" width="700"><img src="package_workflow_files/figure-html/ASV%20inference-3.png" width="700"><img src="package_workflow_files/figure-html/ASV%20inference-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; $its</span></span>
<span><span class="co">#&gt; [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//Rtmpw3urBK/test_dataset/asvabund_matrixDADA2_its.RData"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $rps10</span></span>
<span><span class="co">#&gt; [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//Rtmpw3urBK/test_dataset/asvabund_matrixDADA2_rps10.RData"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="taxonomic-assignment-step">Taxonomic assignment step<a class="anchor" aria-label="anchor" href="#taxonomic-assignment-step"></a>
</h3>
<p>Using the core assignTaxonomy function from DADA2, taxonomic
assignments will be given to ASVs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>  overwrite_existing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1             S1_its  2564     1479      1425      1431   1311    1311</span></span>
<span><span class="co">#&gt; 2             S2_its  1996     1215      1143      1122   1023    1023</span></span>
<span><span class="co">#&gt;   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1           S1_rps10  1830     1429      1429      1422   1420    1420</span></span>
<span><span class="co">#&gt; 2           S2_rps10  2090     1506      1505      1505    753     753</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reformat-asv-matrix-as-taxmap-and-phyloseq-objects-after-optional-filtering-of-low-abundance-asvs">Reformat ASV matrix as taxmap and phyloseq objects after optional
filtering of low abundance ASVs<a class="anchor" aria-label="anchor" href="#reformat-asv-matrix-as-taxmap-and-phyloseq-objects-after-optional-filtering-of-low-abundance-asvs"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">analysis_setup</span>, save_outputs<span class="op">=</span><span class="cn">TRUE</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; For its dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/output_test_dataset/obj_dada_its.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/output_test_dataset/phylo_obj_its.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to NA </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co">#&gt; For rps10 dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/output_test_dataset/obj_dada_rps10.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/output_test_dataset/phylo_obj_rps10.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to NA </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded samples<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples"></a>
</h3>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples-1">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded samples<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-samples-1"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metacoder</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_its</span>,</span>
<span>          node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>          node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>          node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>          layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/Heatree%20demonstration%20its-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-sample">Now we demonstrate how to make a heattree using taxmap object. First
we make a heat tree for our ITS-barcoded sample<a class="anchor" aria-label="anchor" href="#now-we-demonstrate-how-to-make-a-heattree-using-taxmap-object--first-we-make-a-heat-tree-for-our-its-barcoded-sample"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metacoder</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_rps10</span>,</span>
<span>          node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>          node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>          node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>          node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>          layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/Heatree%20demonstration%20rps10-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="we-can-also-do-a-variety-of-analyses-if-we-convert-to-phyloseq-object">We can also do a variety of analyses, if we convert to phyloseq
object<a class="anchor" aria-label="anchor" href="#we-can-also-do-a-variety-of-analyses-if-we-convert-to-phyloseq-object"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_its</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                                        </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">%&gt;%</span>                      </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/Phyloseq%20demonstration%20ITS-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">phyloseq_rps10</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                                        </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&gt;</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">%&gt;%</span>                      </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span>                                      </span>
<span></span>
<span><span class="va">abund_plot</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Sample</span>, y <span class="op">=</span> <span class="va">Abundance</span>, fill <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Relative Abundance"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Relative abundance of taxa by sample"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Genus"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Adjust legend title size</span></span>
<span>    strip.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span></span>
<span>      reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      keywidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      keyheight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>      title.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>      title.hjust <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Center the legend title</span></span>
<span>      label.theme <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>  <span class="co"># Adjust the size of the legend labels</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">abund_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="package_workflow_files/figure-html/Phyloseq%20demonstration%20rps10-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha Sudermann.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
