<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pooled barcode example • demulticoder</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pooled barcode example">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes">
<li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
    <li><a class="dropdown-item" href="../articles/Pooled_barcode_example.html">Pooled barcode example</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="Pooled_barcode_example_files/kePrint-0.0.1/kePrint.js"></script><link href="Pooled_barcode_example_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pooled barcode example</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/vignettes/Pooled_barcode_example.Rmd" class="external-link"><code>vignettes/Pooled_barcode_example.Rmd</code></a></small>
      <div class="d-none name"><code>Pooled_barcode_example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="this-vignette-shows-how-demulticoder-can-be-used-when-you-have-pooled-rps10-and-its1-amplicons-within-each-sample-">This vignette shows how demulticoder can be used when you have
pooled rps10 and ITS1 amplicons within each sample.<a class="anchor" aria-label="anchor" href="#this-vignette-shows-how-demulticoder-can-be-used-when-you-have-pooled-rps10-and-its1-amplicons-within-each-sample-"></a>
</h2>
<p>This is example shows how an analysis is done with the full
rhododendron dataset showcased in the <a href="https://grunwaldlab.github.io/demulticoder/articles/Getting_started.html" class="external-link">Getting
Started</a> vignette. Reads and databases are not sub-sampled as they
were for that test dataset.</p>
<div class="section level3">
<h3 id="prepare-input-files">Prepare input files<a class="anchor" aria-label="anchor" href="#prepare-input-files"></a>
</h3>
<p>First, make sure input metadata and primerinfo_params files are in
data folder</p>
<p>The only required columns are the first with sample names, and the
second with the primer name/barcode used. The subsequent columns are
user-specific columns for downstream steps</p>
<strong><em>metadata.csv</em> file</strong>
<table class="table table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
sample_name
</th>
<th style="text-align:left;">
primer_name
</th>
<th style="text-align:right;">
plate
</th>
<th style="text-align:left;">
well
</th>
<th style="text-align:left;">
organism
</th>
<th style="text-align:left;">
flooded
</th>
<th style="text-align:right;">
path_conc
</th>
<th style="text-align:right;">
experiment
</th>
<th style="text-align:left;">
sample_type
</th>
<th style="text-align:left;">
is_ambiguous
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
rps10
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
A01
</td>
<td style="text-align:left;">
Cry
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
100
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
its
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
A01
</td>
<td style="text-align:left;">
Cry
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
100
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
S10
</td>
<td style="text-align:left;">
rps10
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
B02
</td>
<td style="text-align:left;">
Cry
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
S10
</td>
<td style="text-align:left;">
its
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
B02
</td>
<td style="text-align:left;">
Cry
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
S100
</td>
<td style="text-align:left;">
rps10
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
B02
</td>
<td style="text-align:left;">
Cin
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
S100
</td>
<td style="text-align:left;">
its
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
B02
</td>
<td style="text-align:left;">
Cin
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Sample
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
</tbody>
</table>
<p><em>For the purposes of this example, I will just show the first few
rows of metadata.csv</em> file</p>
<p>I then included the necessary second file with the name of the
barcode selected, primer sequences, and the optional DADA2 parameter
options. I referenced the DADA2 <a href="https://benjjneb.github.io/dada2/tutorial.html" class="external-link">tutorial</a> to
select the proper parameter options.</p>
<em>primerinfo_params.csv</em>
<div style="border: 1px solid #ddd; padding: 0px; thead th { white-space: nowrap; }overflow-y: scroll; height:100px; overflow-x: scroll; width:100%; ">
<table class="table table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
primer_name
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
forward
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
reverse
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
already_trimmed
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minCutadaptlength
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
multithread
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
verbose
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxN
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxEE_forward
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxEE_reverse
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncLen_forward
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncLen_reverse
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
truncQ
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minLen
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxLen
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minQ
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
trimLeft
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
trimRight
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rm.lowcomplex
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
minOverlap
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
maxMismatch
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
min_asv_length
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
rps10
</td>
<td style="text-align:left;">
CTTGGTCATTTAGAGGAAGTAA
</td>
<td style="text-align:left;">
GCTGCGTTCTTCATCGATGC
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
150
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
150
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
50
</td>
</tr>
<tr>
<td style="text-align:left;">
its
</td>
<td style="text-align:left;">
GTTGGTTAGAGYARAAGACT
</td>
<td style="text-align:left;">
ATRYYTAGAAAGAYTYGAACT
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
50
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="loading-the-package">Loading the Package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<p>For now, the package will be loaded by retrieving it from GitHub. We
are submitting package to CRAN</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"grunwaldlab/demulticoder"</span>, force<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#devtools::load_all("~/demulticoder")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://grunwaldlab.github.io/demulticoder/" class="external-link">"demulticoder"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-1">Step 1<a class="anchor" aria-label="anchor" href="#step-1"></a>
</h3>
<p>Remove N’s and create directory structure for downstream steps</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outputs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span>
<span>  </span>
<span>  data_directory <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/demulticoder/data"</span>, </span>
<span>  output_directory <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/demulticoder/vignette_outputs"</span>,</span>
<span>  tempdir_path <span class="op">=</span> <span class="st">"~/benchmark_demulticoder/demulticoder/vignette_temp"</span>,</span>
<span>  tempdir_id <span class="op">=</span> <span class="st">"temp_files"</span><span class="op">)</span></span></code></pre></div>
<p><img src="prepare_reads-1.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="step-2">Step 2<a class="anchor" aria-label="anchor" href="#step-2"></a>
</h3>
<p>Run Cutadapt to remove primers and then trim reads with DADA2
filterAndTrim function</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cut_trim.html">cut_trim</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span><span class="op">)</span></span></code></pre></div>
<p>We can now visualize the outputs from the primer removal and trimming
steps. A CSV files is output showing which samples still have primer
sequences and the barplot below summarizes the outputs.</p>
<p>There are circumstances where a few primer sequences may still
remain. If so, any ASVs with any residual primer sequences will be
filtered at the end.</p>
<p><img src="posttrim_primer_plot_pooled.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="step-3">Step 3<a class="anchor" aria-label="anchor" href="#step-3"></a>
</h3>
<p>Core ASV inference step</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>We can now visualize the outputs from the ASV inference step.</p>
<p>The first plot shows the how reads were merged in terms of mismatches
and indels.</p>
<p>The plot to the right shows the overlap lengths across the inferred
ASVs.</p>
<p>First we see the <em>rps10</em> plots</p>
<p><img src="read_merging_info_rps10.png" width="70%"></p>
<p>Second, we see the ITS plots</p>
<p><img src="read_merging_info_its.png" width="70%"></p>
<p>We can also look at the distribution of ASV lengths</p>
<p>First we see the <em>rps10</em> plots</p>
<p><img src="asv_seqlength_plot_rps10.png" width="70%"></p>
<p>Then we see the ITS plots</p>
<p><img src="asv_seqlength_plot_its.png" width="70%"></p>
</div>
<div class="section level3">
<h3 id="step-4">Step 4<a class="anchor" aria-label="anchor" href="#step-4"></a>
</h3>
<p>Assign taxonomy step</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assign_tax.html">assign_tax</a></span><span class="op">(</span></span>
<span>  <span class="va">outputs</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  db_its <span class="op">=</span> <span class="st">"sh_general_release_dynamic_18.07.2023.fasta"</span>,</span>
<span>  db_rps10 <span class="op">=</span> <span class="st">"oomycetedb_release2.fasta"</span>,</span>
<span>  retrieve_files<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>As a check we can take a look at read counts across the workflow. If
there are sudden drops, we should reconsider our adjusting certain DADA2
parameters and re-running the analysis.</p>
<p>First we track the <em>rps10</em> reads through the demulticoder
DADA2 workflow</p>
<pre><code><span><span class="co">#&gt;   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1           S1_rps10 36752    29365     29351     29346  29310   29310</span></span>
<span><span class="co">#&gt; 2          S10_rps10 27709    22996     22926     22985  22910   22903</span></span>
<span><span class="co">#&gt; 3         S100_rps10 52392    40666     40640     40617  38989   38989</span></span>
<span><span class="co">#&gt; 4         S101_rps10 41784    33442     33374     33386  33324   33161</span></span>
<span><span class="co">#&gt; 5         S102_rps10 90519    67515     67489     67466  67216   67157</span></span>
<span><span class="co">#&gt; 6         S103_rps10 71282    51653     51436     51637  51372   51372</span></span></code></pre>
<p>Then we track <em>ITS1</em> reads</p>
<pre><code><span><span class="co">#&gt; Tables showing read counts throughout the demulticoder DADA2 workflow</span></span>
<span><span class="co">#&gt;   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span></span>
<span><span class="co">#&gt; 1             S1_its 50668    28885     28662     28564  27695   27695</span></span>
<span><span class="co">#&gt; 2            S10_its 26971    13293     13083     13038  12214   12214</span></span>
<span><span class="co">#&gt; 3           S100_its 19379    11748     11559     11484  11121   11121</span></span>
<span><span class="co">#&gt; 4           S101_its 33203    15654     15416     15304  13061   13041</span></span>
<span><span class="co">#&gt; 5           S102_its 33268    19467     19251     19097  18467   18467</span></span>
<span><span class="co">#&gt; 6           S103_its 31711    20252     19942     19738  19095   19095</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="step-5">Step 5<a class="anchor" aria-label="anchor" href="#step-5"></a>
</h3>
<p>Convert asv matrix to taxmap and phyloseq objects with one
function</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_asv_matrix_to_objs.html">convert_asv_matrix_to_objs</a></span><span class="op">(</span><span class="va">outputs</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2372</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">177</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>   (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (174): S1_its, S10_its, S100_its, S101_its, S102_its, S103_its, S104_its...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For its dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/benchmark_demulticoder/demulticoder/vignette_outputs/taxmap_obj_its.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/benchmark_demulticoder/demulticoder/vignette_outputs/phylo_obj_its.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">323</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">177</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span>   (3): asv_id, sequence, dada2_tax</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (174): S1_rps10, S10_rps10, S100_rps10, S101_rps10, S102_rps10, S103_rps...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; For rps10 dataset </span></span>
<span><span class="co">#&gt; Taxmap object saved in: ~/benchmark_demulticoder/demulticoder/vignette_outputs/taxmap_obj_rps10.RData </span></span>
<span><span class="co">#&gt; Phyloseq object saved in: ~/benchmark_demulticoder/demulticoder/vignette_outputs/phylo_obj_rps10.RData </span></span>
<span><span class="co">#&gt; ASVs filtered by minimum read depth: 0 </span></span>
<span><span class="co">#&gt; For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span></span>
<span><span class="co">#&gt; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-6-lets-make-some-quick-visuals-heat-trees-to-take-a-look-at-outputs-of-the-analyses">Step 6-let’s make some quick visuals (heat trees) to take a look at
outputs of the analyses<a class="anchor" aria-label="anchor" href="#step-6-lets-make-some-quick-visuals-heat-trees-to-take-a-look-at-outputs-of-the-analyses"></a>
</h3>
<p>At this point the rps10 and ITS1 analyses are separate. Given the
greater complexity of this dataset, more robust analyses are detailed in
the associated package manuscript and the analyses code is also
linked.</p>
<p>First we make a heat tree for our ITS-barcoded samples</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_bootstrap</span> <span class="op">&lt;-</span> <span class="fl">80</span></span>
<span></span>
<span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_its</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span><span class="op">$</span><span class="va">boot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_its</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span><span class="op">$</span><span class="va">boot</span><span class="op">)</span></span>
<span><span class="va">max_boot</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">taxmap_its</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">taxon_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">boot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">max_boot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">max_boot</span><span class="op">$</span><span class="va">max</span>, <span class="va">max_boot</span><span class="op">$</span><span class="va">taxon_id</span><span class="op">)</span></span>
<span><span class="va">obj_subset_its</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_its</span>, <span class="va">max_boot</span><span class="op">[</span><span class="va">taxon_ids</span><span class="op">]</span> <span class="op">&gt;=</span> <span class="va">min_bootstrap</span> <span class="op">|</span> <span class="va">taxon_ranks</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ASV"</span>, <span class="st">"Reference"</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>abund <span class="op">=</span> <span class="cn">TRUE</span>, score <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: There is no "taxon_id" column in the data set "4", so there are no</span></span>
<span><span class="co">#&gt; taxon IDs.</span></span>
<span></span>
<span><span class="va">obj_subset_its</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="va">is_stem</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"_sp$"</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"incertae_sedis"</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"NA"</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">metacoder</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span>node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>                       node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>                       node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>                       node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>                       node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>                       layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: There is no "taxon_id" column in the data set "4", so there are no</span></span>
<span><span class="co">#&gt; taxon IDs.</span></span></code></pre></div>
<p><img src="Heatree_demonstration_its-1.png" width="70%"></p>
<p>Now we make a heat tree for our rps10-barcoded samples</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_bootstrap</span> <span class="op">&lt;-</span> <span class="fl">80</span></span>
<span></span>
<span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_rps10</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span><span class="op">$</span><span class="va">boot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_rps10</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span><span class="op">$</span><span class="va">boot</span><span class="op">)</span></span>
<span><span class="va">max_boot</span> <span class="op">&lt;-</span> <span class="va">objs</span><span class="op">$</span><span class="va">taxmap_rps10</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">score</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">taxon_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">boot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">max_boot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">max_boot</span><span class="op">$</span><span class="va">max</span>, <span class="va">max_boot</span><span class="op">$</span><span class="va">taxon_id</span><span class="op">)</span></span>
<span><span class="va">obj_subset_rps10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="va">objs</span><span class="op">$</span><span class="va">taxmap_rps10</span>, <span class="va">max_boot</span><span class="op">[</span><span class="va">taxon_ids</span><span class="op">]</span> <span class="op">&gt;=</span> <span class="va">min_bootstrap</span> <span class="op">|</span> <span class="va">taxon_ranks</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ASV"</span>, <span class="st">"Reference"</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>abund <span class="op">=</span> <span class="cn">TRUE</span>, score <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: There is no "taxon_id" column in the data set "4", so there are no</span></span>
<span><span class="co">#&gt; taxon IDs.</span></span>
<span><span class="va">obj_subset_rps10</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"_sp$"</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"incertae_sedis"</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/filter_taxa.html" class="external-link">filter_taxa</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxon_names</span>, <span class="st">"NA"</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, reassign_obs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">metacoder</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/metacoder/man/heat_tree.html" class="external-link">heat_tree</a></span><span class="op">(</span>node_label <span class="op">=</span> <span class="va">taxon_names</span>,</span>
<span>                       node_size <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>                       node_color <span class="op">=</span> <span class="va">n_obs</span>,</span>
<span>                       node_color_axis_label <span class="op">=</span> <span class="st">"ASV count"</span>,</span>
<span>                       node_size_axis_label <span class="op">=</span> <span class="st">"Total Abundance of Taxa"</span>,</span>
<span>                       layout <span class="op">=</span> <span class="st">"da"</span>, initial_layout <span class="op">=</span> <span class="st">"re"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Heatree_demonstration_rps10-1.png" width="70%"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
