<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Assign taxonomy functions"><title>Assign taxonomy functions — assign_tax • demulticoder</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Assign taxonomy functions — assign_tax"><meta property="og:description" content="Assign taxonomy functions"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/mixed_barcode_example.html">Mixed barcode example</a>
    <a class="dropdown-item" href="../articles/package_workflow.html">Package Workflow</a>
    <a class="dropdown-item" href="../articles/pooled_barcode_example.html">Pooled barcode example</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Assign taxonomy functions</h1>
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/R/assign_taxonomy.R" class="external-link"><code>R/assign_taxonomy.R</code></a></small>
      <div class="d-none name"><code>assign_tax.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Assign taxonomy functions</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">assign_tax</span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  <span class="va">asv_abund_matrix</span>,</span>
<span>  tryRC <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  multithread <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  retrieve_files <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  db_rps10 <span class="op">=</span> <span class="st">"oomycetedb.fasta"</span>,</span>
<span>  db_its <span class="op">=</span> <span class="st">"fungidb.fasta"</span>,</span>
<span>  db_16s <span class="op">=</span> <span class="st">"bacteriadb.fasta"</span>,</span>
<span>  db_other1 <span class="op">=</span> <span class="st">"otherdb1.fasta"</span>,</span>
<span>  db_other2 <span class="op">=</span> <span class="st">"otherdb2.fasta"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>analysis_setup</dt>
<dd><p>An object containing directory paths and data tables,
produced by the <code>prepare_reads</code> function</p></dd>


<dt>asv_abund_matrix</dt>
<dd><p>ASV abundance matrix.</p></dd>


<dt>tryRC</dt>
<dd><p>Whether to try reverse complementing sequences during taxonomic
assignment</p></dd>


<dt>verbose</dt>
<dd><p>Logical, indicating whether to display verbose output</p></dd>


<dt>multithread</dt>
<dd><p>Logical, indicating whether to use multithreading</p></dd>


<dt>retrieve_files</dt>
<dd><p>Specify TRUE/FALSE whether to copy files from the temp
directory to the output directory</p></dd>


<dt>overwrite_existing</dt>
<dd><p>Logical, indicating whether to remove or overwrite
existing files and directories from previous runs. Default is <code>FALSE</code>.</p></dd>


<dt>db_rps10</dt>
<dd><p>The reference database for the rps10 locus</p></dd>


<dt>db_its</dt>
<dd><p>The reference database for the ITS locus</p></dd>


<dt>db_16s</dt>
<dd><p>The reference database for the 16S locus</p></dd>


<dt>db_other1</dt>
<dd><p>The reference database for different locus 1 (assumes format is like SILVA DB entries)</p></dd>


<dt>db_other2</dt>
<dd><p>The reference database for a different locus 2 (assumes format is like SILVA DB entries)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Taxonomic assignments of each unique ASV sequence</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Assign taxonomies to ASVs on a per barcode basis</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span><span class="op">&lt;-</span><span class="fu"><a href="prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>  output_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_id <span class="op">=</span> <span class="st">"demulticoder_run_temp"</span>, </span></span>
<span class="r-in"><span>  overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Existing analysis setup tables are not found. The 'prepare_reads' function was rerun</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">3</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">3</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">10</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">4</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (4): sample_name, primer_name, well, organism</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating output directory: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/prefiltered_sequences</span>
<span class="r-plt img"><img src="assign_tax-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="cut_trim.html">cut_trim</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>cutadapt_path<span class="op">=</span><span class="st">"/opt/homebrew/bin/cutadapt"</span>, </span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Existing analysis files not found. The 'cut_trim' function was rerun.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 4.1 for its sequence data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 4.1 for rps10 sequence data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 4.1 for sixteenS sequence data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_1_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_1_R2_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_2_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_2_R2_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_3_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/1_3_R2_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_1_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_1_R2_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_2_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_2_R2_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed forward reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_3_R1_sixteenS.fastq.gz </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Already trimmed reverse reads were appended to trimmed read directory, and they are located here: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/trimmed_sequences/2_3_R2_sixteenS.fastq.gz </span>
<span class="r-plt img"><img src="assign_tax-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>, </span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>No existing files found. The 'make_asv_abund_matrix' function will run.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48620525 total bases in 193819 reads from 6 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  9  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair sixteenS </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 31044 reads in 16523 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 33424 reads in 17777 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 31303 reads in 17153 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 32148 reads in 15953 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 33116 reads in 17060 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 32784 reads in 17064 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48621032 total bases in 193819 reads from 6 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  6  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair sixteenS </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 31044 reads in 20471 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 33424 reads in 22029 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 31303 reads in 21177 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 32148 reads in 20174 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 33116 reads in 21307 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 32784 reads in 21009 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 24286 paired-reads (in 566 unique pairings) successfully merged out of 27574 (in 1193 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 25900 paired-reads (in 606 unique pairings) successfully merged out of 29773 (in 1318 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 24504 paired-reads (in 552 unique pairings) successfully merged out of 27849 (in 1153 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 25377 paired-reads (in 603 unique pairings) successfully merged out of 28485 (in 1194 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 25790 paired-reads (in 584 unique pairings) successfully merged out of 29255 (in 1193 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 25743 paired-reads (in 587 unique pairings) successfully merged out of 28998 (in 1175 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 44 bimeras out of 2008 input sequences.</span>
<span class="r-plt img"><img src="assign_tax-3.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 713433 total bases in 2703 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1486 reads in 659 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1217 reads in 611 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 726880 total bases in 2703 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1486 reads in 1026 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1217 reads in 815 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-plt img"><img src="assign_tax-4.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-plt img"><img src="assign_tax-5.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 824778 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 933 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1018 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 821853 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 1044 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1284 unique sequences.</span>
<span class="r-plt img"><img src="assign_tax-6.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="assign_tax-7.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="assign_tax-8.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $sixteenS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/asvabund_matrixDADA2_sixteenS.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $its</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/asvabund_matrixDADA2_its.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rps10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpafaRoy/demulticoder_run_temp/asvabund_matrixDADA2_rps10.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu">assign_tax</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span><span class="va">asv_abund_matrix</span>, </span></span>
<span class="r-in"><span>retrieve_files<span class="op">=</span><span class="cn">FALSE</span>, </span></span>
<span class="r-in"><span>overwrite_existing<span class="op">=</span><span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>No existing files found. The analysis will be run.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       1_1_sixteenS 40437    31044     28831     28737  24286   23913</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       1_2_sixteenS 44735    33424     31246     30900  25900   25448</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3       1_3_sixteenS 40114    31303     28922     29133  24504   24172</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4       2_1_sixteenS 39984    32148     29659     29631  25377   24748</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5       2_2_sixteenS 43128    33116     30536     30410  25790   25130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6       2_3_sixteenS 43158    32784     30300     30131  25743   25121</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1             S1_its  2564     1486      1427      1433   1316    1316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2             S2_its  1996     1217      1145      1124   1065    1065</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1           S1_rps10  1830     1429      1429      1422   1420    1420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2           S2_rps10  2090     1506      1505      1505   1503    1503</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha Sudermann.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

