<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects — convert_asv_matrix_to_objs • demulticoder</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects — convert_asv_matrix_to_objs"><meta name="description" content="Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects"><meta property="og:description" content="Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes"><li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects</h1>
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/R/convert_asv_matrix.R" class="external-link"><code>R/convert_asv_matrix.R</code></a></small>
      <div class="d-none name"><code>convert_asv_matrix_to_objs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Filter ASV abundance matrix and convert to 'taxmap' and 'phyloseq' objects</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">convert_asv_matrix_to_objs</span><span class="op">(</span></span>
<span>  <span class="va">analysis_setup</span>,</span>
<span>  min_read_depth <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  minimum_bootstrap <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  save_outputs <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-analysis-setup">analysis_setup<a class="anchor" aria-label="anchor" href="#arg-analysis-setup"></a></dt>
<dd><p>An object containing directory paths and
data tables, produced by the <code>prepare_reads</code> function</p></dd>


<dt id="arg-min-read-depth">min_read_depth<a class="anchor" aria-label="anchor" href="#arg-min-read-depth"></a></dt>
<dd><p>ASV filter parameter. If mean read depth of across all
samples is less than this threshold, ASV will be filtered.</p></dd>


<dt id="arg-minimum-bootstrap">minimum_bootstrap<a class="anchor" aria-label="anchor" href="#arg-minimum-bootstrap"></a></dt>
<dd><p>Set threshold for bootstrap support value for taxonomic
assignments. Below designated minimum bootstrap threshold, taxonomic
assignments will be set to N/A</p></dd>


<dt id="arg-save-outputs">save_outputs<a class="anchor" aria-label="anchor" href="#arg-save-outputs"></a></dt>
<dd><p>Logical, indicating whether to save the resulting phyloseq
and 'taxmap' objects. If TRUE, the objects will be saved; if FALSE, they will
only be available in the global environment. Default is FALSE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>ASV matrix converted to 'taxmap' object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co"># Convert final matrix to 'taxmap' and phyloseq objects for downstream analysis steps</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span> <span class="op">&lt;-</span> <span class="fu"><a href="prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  output_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Existing files found in the output directory. Overwriting existing files.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">25</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (18): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): already_trimmed, count_all_samples, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">25</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (18): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): already_trimmed, count_all_samples, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">3</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): sample_name, primer_name, organism</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating output directory: /tmp/RtmpAtZc28/demulticoder_run/prefiltered_sequences</span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="cut_trim.html">cut_trim</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span>,</span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running cutadapt 3.5 for its sequence data</span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-2.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running cutadapt 3.5 for rps10 sequence data</span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="make_asv_abund_matrix.html">make_asv_abund_matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80608 total bases in 307 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Error rate plot for the  Forward  read of primer pair  its</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 163 reads in 84 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 144 reads in 96 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82114 total bases in 307 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Error rate plot for the  Reverse  read of primer pair  its</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 163 reads in 128 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 144 reads in 119 unique sequences.</span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-4.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-5.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91897 total bases in 327 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Error rate plot for the  Forward  read of primer pair  rps10</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 145 reads in 107 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 182 reads in 133 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91567 total bases in 327 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Error rate plot for the  Reverse  read of primer pair  rps10</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 145 reads in 114 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 182 reads in 170 unique sequences.</span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-6.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="convert_asv_matrix_to_objs-7.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $its</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpAtZc28/demulticoder_run/asvabund_matrixDADA2_its.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rps10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpAtZc28/demulticoder_run/asvabund_matrixDADA2_rps10.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu"><a href="assign_tax.html">assign_tax</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span><span class="va">asv_abund_matrix</span>,</span></span>
<span class="r-in"><span>retrieve_files<span class="op">=</span><span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>overwrite_existing<span class="op">=</span><span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Tracking read counts:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1             S1_its   299      163       146       141    132     132</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2             S2_its   235      144       113        99     99      99</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Tracking read counts:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode input filtered denoisedF denoisedR merged nonchim</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1           S1_rps10   196      145       145       145    145     145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2           S2_rps10   253      182       181       181    181     181</span>
<span class="r-in"><span><span class="va">objs</span><span class="op">&lt;-</span><span class="fu">convert_asv_matrix_to_objs</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">12</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">5</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): asv_id, sequence, dada2_tax</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (2): S1_its, S2_its</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For its dataset </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Taxmap object saved in: /tmp/RtmpAtZc28/taxmap_obj_its.RData </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Phyloseq object saved in: /tmp/RtmpAtZc28/phylo_obj_its.RData </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ASVs filtered by minimum read depth: 0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">5</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">5</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): asv_id, sequence, dada2_tax</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (2): S1_rps10, S2_rps10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For rps10 dataset </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Taxmap object saved in: /tmp/RtmpAtZc28/taxmap_obj_rps10.RData </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Phyloseq object saved in: /tmp/RtmpAtZc28/phylo_obj_rps10.RData </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ASVs filtered by minimum read depth: 0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For taxonomic assignments, if minimum bootstrap was set to: 0 assignments were set to 'Unsupported' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

