<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed"><title>Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed — cut_trim • demulticoder</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed — cut_trim"><meta property="og:description" content="Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/01_getting_started.html">Get Started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/02_documentation.html">Documentation</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-example-workflows">Example Workflows</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-example-workflows">
    <a class="dropdown-item" href="../articles/03_mixed_barcode_example.html">Mixed Barcode Example</a>
    <a class="dropdown-item" href="../articles/04_pooled_amplicon_example.html">Pooled Amplicon Example</a>
    <a class="dropdown-item" href="../articles/05_custom_database_example.html">Custom Database Example</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Main Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed</h1>
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/R/cut_primers_trim_reads.R" class="external-link"><code>R/cut_primers_trim_reads.R</code></a></small>
      <div class="d-none name"><code>cut_trim.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Main command to trim primers using Cutadapt and core DADA2 functions. If
samples contain pooled barcodes, reads will also be demultiplexed</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">cut_trim</span><span class="op">(</span><span class="va">analysis_setup</span>, <span class="va">cutadapt_path</span>, overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>analysis_setup</dt>
<dd><p>An object containing directory paths and data tables,
produced by the <code>prepare_reads</code> function</p></dd>


<dt>cutadapt_path</dt>
<dd><p>Path to the Cutadapt program.</p></dd>


<dt>overwrite_existing</dt>
<dd><p>Logical, indicating whether to remove or overwrite
existing files and directories from previous runs. Default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Trimmed reads, primer counts, quality plots, and ASV matrix.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Remove remaining primers from raw reads, demultiplex pooled barcoded samples, </span></span></span>
<span class="r-in"><span><span class="co"># and then trim reads based on specific DADA2 parameters</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span><span class="op">&lt;-</span><span class="fu"><a href="prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>  output_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_id <span class="op">=</span> <span class="st">"demulticoder_run_temp"</span>, </span></span>
<span class="r-in"><span>  overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Existing data detected: Primer counts and N's may have been removed from previous runs. Loading existing output. To perform a  new analysis, specify overwrite_existing = TRUE.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">4</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (4): sample_name, primer_name, well, organism</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">16</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">7</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): primer_name, orientation, sequence</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (4): S1_R1, S1_R2, S2_R1, S2_R2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cutadapt input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sample_name samplename_barcode primer_name well organism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          S1             S1_its         its  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2          S1             S1_its         its  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3          S1           S1_rps10       rps10  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4          S1           S1_rps10       rps10  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5          S2             S2_its         its  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6          S2             S2_its         its  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7          S2           S2_rps10       rps10  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8          S2           S2_rps10       rps10  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  forward                f_compt                  f_rev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     f_rc               reverse               r_compt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   r_rev                  r_rc file_id direction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S1_R1   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S1_R2   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S1_R1   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S1_R2   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S2_R1   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC   S2_R2   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S2_R1   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT   S2_R2   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                           directory_data_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         temp_data_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                             prefiltered_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                   trimmed_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S1_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S1_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S1_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S1_R2_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S2_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S2_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S2_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /tmp/Rtmp4WSdPi/demulticoder_run_temp/trimmed_sequences/S2_R2_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                   untrimmed_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S1_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S1_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S1_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S1_R2_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S2_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S2_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S2_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /tmp/Rtmp4WSdPi/demulticoder_run_temp/untrimmed_sequences/S2_R2_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                   filtered_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S1_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S1_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S1_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S1_R2_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S2_R1_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S2_R2_its.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7 /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S2_R1_rps10.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8 /tmp/Rtmp4WSdPi/demulticoder_run_temp/filtered_sequences/S2_R2_rps10.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Primer input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   primer_name                forward                f_compt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         its CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       rps10   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    f_rev                   f_rc               reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 AATGAAGGAGATTTACTGGTTC TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   TCAGAARAYGAGATTGGTTG   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 r_compt                 r_rev                  r_rc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  CGACGCAAGAAGTAGCTACG  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 TAYRRATCTTTCTRARCTTGA TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fastq input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   file_id sample_name direction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   S1_R1          S1   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   S1_R2          S1   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   S2_R1          S2   Forward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   S2_R2          S2   Reverse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                           directory_data_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /home/marthasudermann/R/x86_64-pc-linux-gnu-library/4.1/demulticoder/extdata/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         temp_data_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/S2_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                             prefiltered_path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S1_R2.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R1.fastq.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 /tmp/Rtmp4WSdPi/demulticoder_run_temp/prefiltered_sequences/S2_R2.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Parameters input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 20</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   primer_name already_trimmed minCutadaptlength multithread verbose  maxN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> rps10       FALSE                         150 TRUE        TRUE        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> its         FALSE                          50 TRUE        TRUE        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 14 more variables: maxEE_forward &lt;dbl&gt;, maxEE_reverse &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   truncLen_forward &lt;dbl&gt;, truncLen_reverse &lt;dbl&gt;, truncQ &lt;dbl&gt;, minLen &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   maxLen &lt;dbl&gt;, minQ &lt;dbl&gt;, trimLeft &lt;dbl&gt;, trimRight &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   rm.lowcomplex &lt;dbl&gt;, minOverlap &lt;dbl&gt;, maxMismatch &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   min_asv_length &lt;dbl&gt;</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Metadata input</span>
<span class="r-plt img"><img src="cut_trim-1.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   samplename_barcode primer_name sample_name well organism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1             S1_its         its          S1  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3           S1_rps10       rps10          S1  A01      Cry</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2             S2_its         its          S2  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4           S2_rps10       rps10          S2  B01      Cin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  forward                f_compt                  f_rev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 CTTGGTCATTTAGAGGAAGTAA GAACCAGTAAATCTCCTTCATT AATGAAGGAGATTTACTGGTTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   GTTGGTTAGAGYARAAGACT   CAACCAATCTCRTYTTCTGA   TCAGAARAYGAGATTGGTTG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     f_rc               reverse               r_compt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 TTACTTCCTCTAAATGACCAAG  GCTGCGTTCTTCATCGATGC  CGACGCAAGAAGTAGCTACG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   AGTCTTYTRCTCTAACCAAC ATRYYTAGAAAGAYTYGAACT TAYRRATCTTTCTRARCTTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   r_rev                  r_rc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  CGTAGCTACTTCTTGCGTCG  GCATCGATGAAGAACGCAGC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 TCAAGYTYAGAAAGATYYRTA AGTTCRARTCTTTCTARRYAT</span>
<span class="r-in"><span><span class="fu">cut_trim</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span>, </span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in list.files(path = directory_path, pattern = pattern, full.names = TRUE,     recursive = TRUE):</span> invalid 'path' argument</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martha A. Sudermann.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

