<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes — make_asv_abund_matrix • demulticoder</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes — make_asv_abund_matrix"><meta name="description" content="Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes"><meta property="og:description" content="Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/Getting_started.html">Getting Started</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Documentation.html">Documentation</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-example-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Example Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-example-vignettes"><li><a class="dropdown-item" href="../articles/DADA2_16S_mothur_validation.html">16S DADA2 Mothur SOP Validation</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Main Functions</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes</h1>
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/R/make_asv_abund_matrix.R" class="external-link"><code>R/make_asv_abund_matrix.R</code></a></small>
      <div class="d-none name"><code>make_asv_abund_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Make an amplified sequence variant (ASV) abundance matrix for each of the input barcodes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">make_asv_abund_matrix</span><span class="op">(</span><span class="va">analysis_setup</span>, overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-analysis-setup">analysis_setup<a class="anchor" aria-label="anchor" href="#arg-analysis-setup"></a></dt>
<dd><p>analysis_setup An object containing directory paths and
data tables, produced by the <code>prepare_reads</code> function</p></dd>


<dt id="arg-overwrite-existing">overwrite_existing<a class="anchor" aria-label="anchor" href="#arg-overwrite-existing"></a></dt>
<dd><p>Logical, indicating whether to overwrite existing
results. Default is FALSE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The ASV abundance matrix (<code>asv_abund_matrix</code>)</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function processes data for each unique barcode separately,
inferring ASVs, merging reads, and creating an ASV abundance matrix.
To do this, the DADA2 core denoising alogrithm is used to infer ASVs.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># The primary wrapper function for DADA2 ASV inference steps</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span> <span class="op">&lt;-</span> <span class="fu"><a href="prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  output_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_id <span class="op">=</span> <span class="st">"demulticoder_run_temp"</span>,</span></span>
<span class="r-in"><span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): already_trimmed, count_all_samples, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">23</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): already_trimmed, count_all_samples, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">3</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): sample_name, primer_name, organism</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating output directory: /var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpZA9S3T/demulticoder_run_temp/prefiltered_sequences</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="cut_trim.html">cut_trim</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>cutadapt_path<span class="op">=</span><span class="st">"/opt/homebrew/bin/cutadapt"</span>,</span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 4.1 for its sequence data </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 2564 paired-sequences, output 1479 (57.7%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 1996 paired-sequences, output 1215 (60.9%) filtered paired-sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 4.1 for rps10 sequence data </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 1830 paired-sequences, output 1429 (78.1%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 2090 paired-sequences, output 1506 (72.1%) filtered paired-sequences.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">make_asv_abund_matrix</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>, </span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 710847 total bases in 2694 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1479 reads in 654 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1215 reads in 610 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 724232 total bases in 2694 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1479 reads in 1019 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1215 reads in 814 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1315 paired-reads (in 21 unique pairings) successfully merged out of 1416 (in 32 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1063 paired-reads (in 25 unique pairings) successfully merged out of 1108 (in 28 pairings) input.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-3.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 38 input sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 824778 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 933 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1018 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 821853 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-4.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 1044 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1284 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1420 paired-reads (in 2 unique pairings) successfully merged out of 1422 (in 4 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1503 paired-reads (in 5 unique pairings) successfully merged out of 1504 (in 6 pairings) input.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-5.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 5 input sequences.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-6.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $its</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpZA9S3T/demulticoder_run_temp/asvabund_matrixDADA2_its.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rps10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/var/folders/59/9jp4sjjd00n2wp4kgqtvq7dh0000gn/T//RtmpZA9S3T/demulticoder_run_temp/asvabund_matrixDADA2_rps10.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martha A. Sudermann, Zachary S. L Foster, Samantha Dawson, Hung Phan, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

