<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs."><title>Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs. — make_asv_abund_matrix • demulticoder</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs. — make_asv_abund_matrix"><meta property="og:description" content="Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">demulticoder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/01_getting_started.html">Get Started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/02_documentation.html">Documentation</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-example-workflows">Example Workflows</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-example-workflows">
    <a class="dropdown-item" href="../articles/03_mixed_barcode_example.html">Mixed Barcode Example</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Main Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grunwaldlab/demulticoder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs.</h1>
      <small class="dont-index">Source: <a href="https://github.com/grunwaldlab/demulticoder/blob/HEAD/R/make_asv_abund_matrix.R" class="external-link"><code>R/make_asv_abund_matrix.R</code></a></small>
      <div class="d-none name"><code>make_asv_abund_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Make an amplified sequence variant (ASV) abundance matrix This function
generates an ASV abundance matrix using raw reads processed during previous
steps, including read preparation, removing primers, and using DADA2 core
denoising alogrithm to infer ASVs.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">make_asv_abund_matrix</span><span class="op">(</span><span class="va">analysis_setup</span>, overwrite_existing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>analysis_setup</dt>
<dd><p>analysis_setup An object containing directory paths and
data tables, produced by the <code>prepare_reads</code> function</p></dd>


<dt>overwrite_existing</dt>
<dd><p>Logical, indicating whether to overwrite existing
results. Default is FALSE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>The ASV abundance matrix (<code>asv_abund_matrix</code>)</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function processes data for each unique barcode separately,
inferring ASVs, merging reads, and creating an ASV abundance matrix</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># The primary wrapper function for DADA2 ASV inference steps</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span> <span class="op">&lt;-</span> <span class="fu"><a href="prepare_reads.html">prepare_reads</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"demulticoder"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  output_directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  tempdir_id <span class="op">=</span> <span class="st">"demulticoder_run_temp"</span>,</span></span>
<span class="r-in"><span>  overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">2</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">22</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span>  (3): primer_name, forward, reverse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (16): minCutadaptlength, maxN, maxEE_forward, maxEE_reverse, truncLen_fo...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): already_trimmed, multithread, verbose</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">3</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (3): sample_name, primer_name, organism</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating output directory: /tmp/Rtmp23cMn7/demulticoder_run_temp/prefiltered_sequences</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="cut_trim.html">cut_trim</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>,</span></span>
<span class="r-in"><span>cutadapt_path<span class="op">=</span><span class="st">"/usr/bin/cutadapt"</span>,</span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 3.5 for its sequence data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running Cutadapt 3.5 for rps10 sequence data </span>
<span class="r-plt img"><img src="make_asv_abund_matrix-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">make_asv_abund_matrix</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">analysis_setup</span>, </span></span>
<span class="r-in"><span>overwrite_existing <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 711900 total bases in 2698 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1481 reads in 660 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1217 reads in 614 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 725393 total bases in 2698 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair its </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1481 reads in 1021 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1217 reads in 815 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1316 paired-reads (in 21 unique pairings) successfully merged out of 1418 (in 32 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences in merged output.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1065 paired-reads (in 25 unique pairings) successfully merged out of 1110 (in 28 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Duplicate sequences detected and merged.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 38 input sequences.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-3.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 824778 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Forward read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 933 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1018 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 821851 total bases in 2935 reads from 2 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 ..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  3  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error rate plot for the Reverse read of primer pair rps10 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #00BB00;">log-10</span> transformation introduced infinite values.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 1429 reads in 1044 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 1506 reads in 1284 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1420 paired-reads (in 2 unique pairings) successfully merged out of 1422 (in 4 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1503 paired-reads (in 5 unique pairings) successfully merged out of 1504 (in 6 pairings) input.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-4.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 5 input sequences.</span>
<span class="r-plt img"><img src="make_asv_abund_matrix-5.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="make_asv_abund_matrix-6.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $its</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/Rtmp23cMn7/demulticoder_run_temp/asvabund_matrixDADA2_its.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rps10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/Rtmp23cMn7/demulticoder_run_temp/asvabund_matrixDADA2_rps10.RData"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martha A. Sudermann, Zachary S. L Foster, Hung Phan, Samantha Dawson, Jeff H. Chang, Niklaus Grünwald.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

